{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# QC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 81\n",
      "total bases: 6081\n",
      "Q20 bases: 5138(84.4927%)\n",
      "Q30 bases: 4471(73.5241%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 81\n",
      "total bases: 5326\n",
      "Q20 bases: 4805(90.2178%)\n",
      "Q30 bases: 4392(82.4634%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 5\n",
      "total bases: 510\n",
      "Q20 bases: 488(95.6863%)\n",
      "Q30 bases: 461(90.3922%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 10\n",
      "reads failed due to low quality: 28\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 124\n",
      "reads with adapter trimmed: 10\n",
      "bases trimmed due to adapters: 111\n",
      "reads corrected by overlap analysis: 3\n",
      "bases corrected by overlap analysis: 3\n",
      "\n",
      "Duplication rate: 0%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 5\n",
      "% of original read pairs: 6.17284%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/21AU_fastp.json\n",
      "HTML report: ./fastp_out/21AU_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/21AU.R1.fastq -I 2_demultiplexed/ADD/21AU.R2.fastq -o 3_qc/ADD/21AU.R1.fastq -O 3_qc/ADD/21AU.R2.fastq --unpaired1 3_qc/ADD/21AU_unpaired.R1.fastq --unpaired2 3_qc/ADD/21AU_unpaired.R2.fastq --failed_out 3_qc/ADD/21AU_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/21AU_merged.fastq -w 6 -h ./fastp_out/21AU_fastp.html -j ./fastp_out/21AU_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 1535\n",
      "total bases: 90946\n",
      "Q20 bases: 85410(93.9129%)\n",
      "Q30 bases: 79450(87.3595%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 1535\n",
      "total bases: 87335\n",
      "Q20 bases: 84386(96.6233%)\n",
      "Q30 bases: 80974(92.7166%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 1\n",
      "total bases: 104\n",
      "Q20 bases: 96(92.3077%)\n",
      "Q30 bases: 87(83.6538%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2\n",
      "reads failed due to low quality: 274\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2794\n",
      "reads with adapter trimmed: 2\n",
      "bases trimmed due to adapters: 20\n",
      "reads corrected by overlap analysis: 1\n",
      "bases corrected by overlap analysis: 1\n",
      "\n",
      "Duplication rate: 0%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 144\n",
      "\n",
      "Read pairs merged: 1\n",
      "% of original read pairs: 0.0651466%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/21SM_fastp.json\n",
      "HTML report: ./fastp_out/21SM_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/21SM.R1.fastq -I 2_demultiplexed/ADD/21SM.R2.fastq -o 3_qc/ADD/21SM.R1.fastq -O 3_qc/ADD/21SM.R2.fastq --unpaired1 3_qc/ADD/21SM_unpaired.R1.fastq --unpaired2 3_qc/ADD/21SM_unpaired.R2.fastq --failed_out 3_qc/ADD/21SM_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/21SM_merged.fastq -w 6 -h ./fastp_out/21SM_fastp.html -j ./fastp_out/21SM_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 39825\n",
      "total bases: 3652193\n",
      "Q20 bases: 3507078(96.0266%)\n",
      "Q30 bases: 3313609(90.7293%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 39825\n",
      "total bases: 3581878\n",
      "Q20 bases: 3469358(96.8586%)\n",
      "Q30 bases: 3301891(92.1832%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 12476\n",
      "total bases: 1284969\n",
      "Q20 bases: 1268950(98.7534%)\n",
      "Q30 bases: 1224682(95.3083%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 25118\n",
      "reads failed due to low quality: 5984\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 48548\n",
      "reads with adapter trimmed: 25310\n",
      "bases trimmed due to adapters: 317442\n",
      "reads corrected by overlap analysis: 2579\n",
      "bases corrected by overlap analysis: 4466\n",
      "\n",
      "Duplication rate: 27.2727%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 12476\n",
      "% of original read pairs: 31.3271%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A47S_fastp.json\n",
      "HTML report: ./fastp_out/A47S_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/A47S.R1.fastq -I 2_demultiplexed/ADD/A47S.R2.fastq -o 3_qc/ADD/A47S.R1.fastq -O 3_qc/ADD/A47S.R2.fastq --unpaired1 3_qc/ADD/A47S_unpaired.R1.fastq --unpaired2 3_qc/ADD/A47S_unpaired.R2.fastq --failed_out 3_qc/ADD/A47S_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/A47S_merged.fastq -w 6 -h ./fastp_out/A47S_fastp.html -j ./fastp_out/A47S_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 98932\n",
      "total bases: 15126507\n",
      "Q20 bases: 14832479(98.0562%)\n",
      "Q30 bases: 14265399(94.3073%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 98932\n",
      "total bases: 15112446\n",
      "Q20 bases: 14762368(97.6835%)\n",
      "Q30 bases: 14110991(93.3733%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 94669\n",
      "total bases: 9745277\n",
      "Q20 bases: 9644074(98.9615%)\n",
      "Q30 bases: 9348224(95.9257%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 189890\n",
      "reads failed due to low quality: 4160\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 3808\n",
      "reads with adapter trimmed: 191748\n",
      "bases trimmed due to adapters: 2233786\n",
      "reads corrected by overlap analysis: 16269\n",
      "bases corrected by overlap analysis: 27023\n",
      "\n",
      "Duplication rate: 27.7778%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 94669\n",
      "% of original read pairs: 95.691%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C10S_fastp.json\n",
      "HTML report: ./fastp_out/C10S_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/C10S.R1.fastq -I 2_demultiplexed/ADD/C10S.R2.fastq -o 3_qc/ADD/C10S.R1.fastq -O 3_qc/ADD/C10S.R2.fastq --unpaired1 3_qc/ADD/C10S_unpaired.R1.fastq --unpaired2 3_qc/ADD/C10S_unpaired.R2.fastq --failed_out 3_qc/ADD/C10S_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/C10S_merged.fastq -w 6 -h ./fastp_out/C10S_fastp.html -j ./fastp_out/C10S_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 102284\n",
      "total bases: 15362625\n",
      "Q20 bases: 15006588(97.6824%)\n",
      "Q30 bases: 14361560(93.4838%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 102284\n",
      "total bases: 15350153\n",
      "Q20 bases: 14961678(97.4692%)\n",
      "Q30 bases: 14259080(92.8921%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 94809\n",
      "total bases: 9725860\n",
      "Q20 bases: 9596765(98.6727%)\n",
      "Q30 bases: 9242574(95.0309%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 190112\n",
      "reads failed due to low quality: 5956\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 8498\n",
      "reads with adapter trimmed: 192338\n",
      "bases trimmed due to adapters: 2398024\n",
      "reads corrected by overlap analysis: 20060\n",
      "bases corrected by overlap analysis: 33622\n",
      "\n",
      "Duplication rate: 19.1176%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 94809\n",
      "% of original read pairs: 92.6919%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C1ot_fastp.json\n",
      "HTML report: ./fastp_out/C1ot_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/C1ot.R1.fastq -I 2_demultiplexed/ADD/C1ot.R2.fastq -o 3_qc/ADD/C1ot.R1.fastq -O 3_qc/ADD/C1ot.R2.fastq --unpaired1 3_qc/ADD/C1ot_unpaired.R1.fastq --unpaired2 3_qc/ADD/C1ot_unpaired.R2.fastq --failed_out 3_qc/ADD/C1ot_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/C1ot_merged.fastq -w 6 -h ./fastp_out/C1ot_fastp.html -j ./fastp_out/C1ot_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 72528\n",
      "total bases: 9258006\n",
      "Q20 bases: 9002547(97.2407%)\n",
      "Q30 bases: 8596404(92.8537%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 72528\n",
      "total bases: 9170933\n",
      "Q20 bases: 8961087(97.7118%)\n",
      "Q30 bases: 8588768(93.6521%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 48743\n",
      "total bases: 5019335\n",
      "Q20 bases: 4959731(98.8125%)\n",
      "Q30 bases: 4791006(95.451%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 107508\n",
      "reads failed due to low quality: 6902\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 30646\n",
      "reads with adapter trimmed: 99586\n",
      "bases trimmed due to adapters: 1152346\n",
      "reads corrected by overlap analysis: 9209\n",
      "bases corrected by overlap analysis: 15807\n",
      "\n",
      "Duplication rate: 27.7311%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 48743\n",
      "% of original read pairs: 67.2058%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C2ot_fastp.json\n",
      "HTML report: ./fastp_out/C2ot_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/C2ot.R1.fastq -I 2_demultiplexed/ADD/C2ot.R2.fastq -o 3_qc/ADD/C2ot.R1.fastq -O 3_qc/ADD/C2ot.R2.fastq --unpaired1 3_qc/ADD/C2ot_unpaired.R1.fastq --unpaired2 3_qc/ADD/C2ot_unpaired.R2.fastq --failed_out 3_qc/ADD/C2ot_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/C2ot_merged.fastq -w 6 -h ./fastp_out/C2ot_fastp.html -j ./fastp_out/C2ot_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 37664\n",
      "total bases: 4972421\n",
      "Q20 bases: 4843645(97.4102%)\n",
      "Q30 bases: 4630045(93.1145%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 37664\n",
      "total bases: 4959008\n",
      "Q20 bases: 4845215(97.7053%)\n",
      "Q30 bases: 4647623(93.7208%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 26864\n",
      "total bases: 2773017\n",
      "Q20 bases: 2740931(98.8429%)\n",
      "Q30 bases: 2649159(95.5335%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 58166\n",
      "reads failed due to low quality: 3218\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 13944\n",
      "reads with adapter trimmed: 55548\n",
      "bases trimmed due to adapters: 585245\n",
      "reads corrected by overlap analysis: 4561\n",
      "bases corrected by overlap analysis: 7805\n",
      "\n",
      "Duplication rate: 20.202%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 26864\n",
      "% of original read pairs: 71.3254%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C3ot_fastp.json\n",
      "HTML report: ./fastp_out/C3ot_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/C3ot.R1.fastq -I 2_demultiplexed/ADD/C3ot.R2.fastq -o 3_qc/ADD/C3ot.R1.fastq -O 3_qc/ADD/C3ot.R2.fastq --unpaired1 3_qc/ADD/C3ot_unpaired.R1.fastq --unpaired2 3_qc/ADD/C3ot_unpaired.R2.fastq --failed_out 3_qc/ADD/C3ot_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/C3ot_merged.fastq -w 6 -h ./fastp_out/C3ot_fastp.html -j ./fastp_out/C3ot_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 64656\n",
      "total bases: 5015511\n",
      "Q20 bases: 4763339(94.9722%)\n",
      "Q30 bases: 4467476(89.0732%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 64656\n",
      "total bases: 4853911\n",
      "Q20 bases: 4708943(97.0134%)\n",
      "Q30 bases: 4511010(92.9356%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 11324\n",
      "total bases: 1175413\n",
      "Q20 bases: 1161393(98.8072%)\n",
      "Q30 bases: 1121489(95.4123%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 22724\n",
      "reads failed due to low quality: 10220\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 96366\n",
      "reads with adapter trimmed: 22942\n",
      "bases trimmed due to adapters: 243619\n",
      "reads corrected by overlap analysis: 2169\n",
      "bases corrected by overlap analysis: 3690\n",
      "\n",
      "Duplication rate: 28.125%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 11324\n",
      "% of original read pairs: 17.5142%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C45S_fastp.json\n",
      "HTML report: ./fastp_out/C45S_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/C45S.R1.fastq -I 2_demultiplexed/ADD/C45S.R2.fastq -o 3_qc/ADD/C45S.R1.fastq -O 3_qc/ADD/C45S.R2.fastq --unpaired1 3_qc/ADD/C45S_unpaired.R1.fastq --unpaired2 3_qc/ADD/C45S_unpaired.R2.fastq --failed_out 3_qc/ADD/C45S_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/C45S_merged.fastq -w 6 -h ./fastp_out/C45S_fastp.html -j ./fastp_out/C45S_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 41216\n",
      "total bases: 4425559\n",
      "Q20 bases: 4213070(95.1986%)\n",
      "Q30 bases: 3969384(89.6923%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 41216\n",
      "total bases: 4251474\n",
      "Q20 bases: 4133315(97.2208%)\n",
      "Q30 bases: 3947255(92.8444%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 17439\n",
      "total bases: 1795907\n",
      "Q20 bases: 1773621(98.7591%)\n",
      "Q30 bases: 1711191(95.2828%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 35062\n",
      "reads failed due to low quality: 7422\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 39946\n",
      "reads with adapter trimmed: 35354\n",
      "bases trimmed due to adapters: 443233\n",
      "reads corrected by overlap analysis: 3488\n",
      "bases corrected by overlap analysis: 5777\n",
      "\n",
      "Duplication rate: 12.8205%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 144\n",
      "\n",
      "Read pairs merged: 17439\n",
      "% of original read pairs: 42.3112%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C47S_fastp.json\n",
      "HTML report: ./fastp_out/C47S_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/C47S.R1.fastq -I 2_demultiplexed/ADD/C47S.R2.fastq -o 3_qc/ADD/C47S.R1.fastq -O 3_qc/ADD/C47S.R2.fastq --unpaired1 3_qc/ADD/C47S_unpaired.R1.fastq --unpaired2 3_qc/ADD/C47S_unpaired.R2.fastq --failed_out 3_qc/ADD/C47S_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/C47S_merged.fastq -w 6 -h ./fastp_out/C47S_fastp.html -j ./fastp_out/C47S_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 47528\n",
      "total bases: 6759687\n",
      "Q20 bases: 6574897(97.2663%)\n",
      "Q30 bases: 6255027(92.5343%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 47528\n",
      "total bases: 6739182\n",
      "Q20 bases: 6564899(97.4139%)\n",
      "Q30 bases: 6255541(92.8234%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 30739\n",
      "total bases: 3128065\n",
      "Q20 bases: 3087778(98.7121%)\n",
      "Q30 bases: 2972188(95.0168%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 81616\n",
      "reads failed due to low quality: 4398\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 9040\n",
      "reads with adapter trimmed: 64150\n",
      "bases trimmed due to adapters: 810758\n",
      "reads corrected by overlap analysis: 6228\n",
      "bases corrected by overlap analysis: 10492\n",
      "\n",
      "Duplication rate: 26.3889%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 30739\n",
      "% of original read pairs: 64.6756%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C4ot_fastp.json\n",
      "HTML report: ./fastp_out/C4ot_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/C4ot.R1.fastq -I 2_demultiplexed/ADD/C4ot.R2.fastq -o 3_qc/ADD/C4ot.R1.fastq -O 3_qc/ADD/C4ot.R2.fastq --unpaired1 3_qc/ADD/C4ot_unpaired.R1.fastq --unpaired2 3_qc/ADD/C4ot_unpaired.R2.fastq --failed_out 3_qc/ADD/C4ot_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/C4ot_merged.fastq -w 6 -h ./fastp_out/C4ot_fastp.html -j ./fastp_out/C4ot_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 6376\n",
      "total bases: 340264\n",
      "Q20 bases: 321411(94.4593%)\n",
      "Q30 bases: 300408(88.2867%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 6376\n",
      "total bases: 325561\n",
      "Q20 bases: 313730(96.366%)\n",
      "Q30 bases: 299626(92.0338%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 43\n",
      "total bases: 4585\n",
      "Q20 bases: 4527(98.735%)\n",
      "Q30 bases: 4372(95.3544%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 584\n",
      "reads failed due to low quality: 1012\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 11156\n",
      "reads with adapter trimmed: 112\n",
      "bases trimmed due to adapters: 1266\n",
      "reads corrected by overlap analysis: 13\n",
      "bases corrected by overlap analysis: 18\n",
      "\n",
      "Duplication rate: 5.88235%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 156\n",
      "\n",
      "Read pairs merged: 43\n",
      "% of original read pairs: 0.674404%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C50S_fastp.json\n",
      "HTML report: ./fastp_out/C50S_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/C50S.R1.fastq -I 2_demultiplexed/ADD/C50S.R2.fastq -o 3_qc/ADD/C50S.R1.fastq -O 3_qc/ADD/C50S.R2.fastq --unpaired1 3_qc/ADD/C50S_unpaired.R1.fastq --unpaired2 3_qc/ADD/C50S_unpaired.R2.fastq --failed_out 3_qc/ADD/C50S_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/C50S_merged.fastq -w 6 -h ./fastp_out/C50S_fastp.html -j ./fastp_out/C50S_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 211593\n",
      "total bases: 32217540\n",
      "Q20 bases: 31637857(98.2007%)\n",
      "Q30 bases: 30498914(94.6656%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 211593\n",
      "total bases: 32205303\n",
      "Q20 bases: 31567359(98.0191%)\n",
      "Q30 bases: 30336073(94.1959%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 201168\n",
      "total bases: 20704823\n",
      "Q20 bases: 20480809(98.9181%)\n",
      "Q30 bases: 19839200(95.8192%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 405996\n",
      "reads failed due to low quality: 8106\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 9078\n",
      "reads with adapter trimmed: 406370\n",
      "bases trimmed due to adapters: 4714636\n",
      "reads corrected by overlap analysis: 34391\n",
      "bases corrected by overlap analysis: 56087\n",
      "\n",
      "Duplication rate: 49.5192%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 201168\n",
      "% of original read pairs: 95.0731%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C5ot_fastp.json\n",
      "HTML report: ./fastp_out/C5ot_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/C5ot.R1.fastq -I 2_demultiplexed/ADD/C5ot.R2.fastq -o 3_qc/ADD/C5ot.R1.fastq -O 3_qc/ADD/C5ot.R2.fastq --unpaired1 3_qc/ADD/C5ot_unpaired.R1.fastq --unpaired2 3_qc/ADD/C5ot_unpaired.R2.fastq --failed_out 3_qc/ADD/C5ot_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/C5ot_merged.fastq -w 6 -h ./fastp_out/C5ot_fastp.html -j ./fastp_out/C5ot_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 17633\n",
      "total bases: 1451194\n",
      "Q20 bases: 1336991(92.1304%)\n",
      "Q30 bases: 1232133(84.9048%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 17633\n",
      "total bases: 1342701\n",
      "Q20 bases: 1270805(94.6454%)\n",
      "Q30 bases: 1191357(88.7284%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 3144\n",
      "total bases: 318627\n",
      "Q20 bases: 314617(98.7415%)\n",
      "Q30 bases: 303267(95.1793%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 6318\n",
      "reads failed due to low quality: 5178\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 23770\n",
      "reads with adapter trimmed: 6356\n",
      "bases trimmed due to adapters: 79654\n",
      "reads corrected by overlap analysis: 740\n",
      "bases corrected by overlap analysis: 1243\n",
      "\n",
      "Duplication rate: 25.3333%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 3144\n",
      "% of original read pairs: 17.8302%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/N1ot_fastp.json\n",
      "HTML report: ./fastp_out/N1ot_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/N1ot.R1.fastq -I 2_demultiplexed/ADD/N1ot.R2.fastq -o 3_qc/ADD/N1ot.R1.fastq -O 3_qc/ADD/N1ot.R2.fastq --unpaired1 3_qc/ADD/N1ot_unpaired.R1.fastq --unpaired2 3_qc/ADD/N1ot_unpaired.R2.fastq --failed_out 3_qc/ADD/N1ot_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/N1ot_merged.fastq -w 6 -h ./fastp_out/N1ot_fastp.html -j ./fastp_out/N1ot_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 93945\n",
      "total bases: 14321443\n",
      "Q20 bases: 13988977(97.6785%)\n",
      "Q30 bases: 13375390(93.3942%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 93945\n",
      "total bases: 14338344\n",
      "Q20 bases: 13889530(96.8698%)\n",
      "Q30 bases: 13119050(91.4963%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 90573\n",
      "total bases: 9237848\n",
      "Q20 bases: 9119416(98.718%)\n",
      "Q30 bases: 8781201(95.0568%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 181574\n",
      "reads failed due to low quality: 5070\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1246\n",
      "reads with adapter trimmed: 183610\n",
      "bases trimmed due to adapters: 1981324\n",
      "reads corrected by overlap analysis: 18931\n",
      "bases corrected by overlap analysis: 31159\n",
      "\n",
      "Duplication rate: 81.6794%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 90573\n",
      "% of original read pairs: 96.4107%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/P1ot_fastp.json\n",
      "HTML report: ./fastp_out/P1ot_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/P1ot.R1.fastq -I 2_demultiplexed/ADD/P1ot.R2.fastq -o 3_qc/ADD/P1ot.R1.fastq -O 3_qc/ADD/P1ot.R2.fastq --unpaired1 3_qc/ADD/P1ot_unpaired.R1.fastq --unpaired2 3_qc/ADD/P1ot_unpaired.R2.fastq --failed_out 3_qc/ADD/P1ot_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/P1ot_merged.fastq -w 6 -h ./fastp_out/P1ot_fastp.html -j ./fastp_out/P1ot_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 345223\n",
      "total bases: 52881876\n",
      "Q20 bases: 51622597(97.6187%)\n",
      "Q30 bases: 49292367(93.2122%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 345223\n",
      "total bases: 52854182\n",
      "Q20 bases: 51346444(97.1474%)\n",
      "Q30 bases: 48633213(92.0139%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 335614\n",
      "total bases: 34230474\n",
      "Q20 bases: 33761896(98.6311%)\n",
      "Q30 bases: 32456463(94.8175%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 672508\n",
      "reads failed due to low quality: 17892\n",
      "reads failed due to too many N: 8\n",
      "reads failed due to too short: 38\n",
      "reads with adapter trimmed: 681328\n",
      "bases trimmed due to adapters: 7335833\n",
      "reads corrected by overlap analysis: 70030\n",
      "bases corrected by overlap analysis: 115773\n",
      "\n",
      "Duplication rate: 82.9582%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 335614\n",
      "% of original read pairs: 97.2166%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/POSI_fastp.json\n",
      "HTML report: ./fastp_out/POSI_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/POSI.R1.fastq -I 2_demultiplexed/ADD/POSI.R2.fastq -o 3_qc/ADD/POSI.R1.fastq -O 3_qc/ADD/POSI.R2.fastq --unpaired1 3_qc/ADD/POSI_unpaired.R1.fastq --unpaired2 3_qc/ADD/POSI_unpaired.R2.fastq --failed_out 3_qc/ADD/POSI_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/POSI_merged.fastq -w 6 -h ./fastp_out/POSI_fastp.html -j ./fastp_out/POSI_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 39786\n",
      "total bases: 4243225\n",
      "Q20 bases: 4096388(96.5395%)\n",
      "Q30 bases: 3900358(91.9197%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 39786\n",
      "total bases: 4222005\n",
      "Q20 bases: 4055912(96.066%)\n",
      "Q30 bases: 3847263(91.1241%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 20986\n",
      "total bases: 2161102\n",
      "Q20 bases: 2137599(98.9125%)\n",
      "Q30 bases: 2070681(95.816%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 42398\n",
      "reads failed due to low quality: 6512\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 30662\n",
      "reads with adapter trimmed: 42402\n",
      "bases trimmed due to adapters: 494318\n",
      "reads corrected by overlap analysis: 3994\n",
      "bases corrected by overlap analysis: 6570\n",
      "\n",
      "Duplication rate: 27.7778%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 20986\n",
      "% of original read pairs: 52.7472%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/RS11_fastp.json\n",
      "HTML report: ./fastp_out/RS11_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/RS11.R1.fastq -I 2_demultiplexed/ADD/RS11.R2.fastq -o 3_qc/ADD/RS11.R1.fastq -O 3_qc/ADD/RS11.R2.fastq --unpaired1 3_qc/ADD/RS11_unpaired.R1.fastq --unpaired2 3_qc/ADD/RS11_unpaired.R2.fastq --failed_out 3_qc/ADD/RS11_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/RS11_merged.fastq -w 6 -h ./fastp_out/RS11_fastp.html -j ./fastp_out/RS11_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 20443\n",
      "total bases: 1756854\n",
      "Q20 bases: 1680188(95.6362%)\n",
      "Q30 bases: 1581125(89.9975%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 20443\n",
      "total bases: 1717480\n",
      "Q20 bases: 1657203(96.4904%)\n",
      "Q30 bases: 1571733(91.5139%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 7198\n",
      "total bases: 747386\n",
      "Q20 bases: 737020(98.613%)\n",
      "Q30 bases: 709186(94.8889%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 15198\n",
      "reads failed due to low quality: 2876\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 22812\n",
      "reads with adapter trimmed: 14704\n",
      "bases trimmed due to adapters: 158367\n",
      "reads corrected by overlap analysis: 1590\n",
      "bases corrected by overlap analysis: 2734\n",
      "\n",
      "Duplication rate: 25.6098%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 7198\n",
      "% of original read pairs: 35.2101%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/RS15_fastp.json\n",
      "HTML report: ./fastp_out/RS15_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/RS15.R1.fastq -I 2_demultiplexed/ADD/RS15.R2.fastq -o 3_qc/ADD/RS15.R1.fastq -O 3_qc/ADD/RS15.R2.fastq --unpaired1 3_qc/ADD/RS15_unpaired.R1.fastq --unpaired2 3_qc/ADD/RS15_unpaired.R2.fastq --failed_out 3_qc/ADD/RS15_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/RS15_merged.fastq -w 6 -h ./fastp_out/RS15_fastp.html -j ./fastp_out/RS15_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 19286\n",
      "total bases: 1841423\n",
      "Q20 bases: 1774497(96.3655%)\n",
      "Q30 bases: 1687796(91.6572%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 19286\n",
      "total bases: 1864195\n",
      "Q20 bases: 1769786(94.9357%)\n",
      "Q30 bases: 1670097(89.5881%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 8060\n",
      "total bases: 830072\n",
      "Q20 bases: 820542(98.8519%)\n",
      "Q30 bases: 794072(95.663%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 16178\n",
      "reads failed due to low quality: 3670\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 18724\n",
      "reads with adapter trimmed: 16250\n",
      "bases trimmed due to adapters: 205160\n",
      "reads corrected by overlap analysis: 1596\n",
      "bases corrected by overlap analysis: 2579\n",
      "\n",
      "Duplication rate: 27.2727%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 8060\n",
      "% of original read pairs: 41.792%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/RS19_fastp.json\n",
      "HTML report: ./fastp_out/RS19_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/RS19.R1.fastq -I 2_demultiplexed/ADD/RS19.R2.fastq -o 3_qc/ADD/RS19.R1.fastq -O 3_qc/ADD/RS19.R2.fastq --unpaired1 3_qc/ADD/RS19_unpaired.R1.fastq --unpaired2 3_qc/ADD/RS19_unpaired.R2.fastq --failed_out 3_qc/ADD/RS19_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/RS19_merged.fastq -w 6 -h ./fastp_out/RS19_fastp.html -j ./fastp_out/RS19_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 79728\n",
      "total bases: 6875497\n",
      "Q20 bases: 6603787(96.0481%)\n",
      "Q30 bases: 6249913(90.9013%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 79728\n",
      "total bases: 6735175\n",
      "Q20 bases: 6530683(96.9638%)\n",
      "Q30 bases: 6234131(92.5608%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 20327\n",
      "total bases: 2093074\n",
      "Q20 bases: 2071301(98.9598%)\n",
      "Q30 bases: 2006891(95.8825%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 41076\n",
      "reads failed due to low quality: 11664\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 106716\n",
      "reads with adapter trimmed: 41040\n",
      "bases trimmed due to adapters: 477184\n",
      "reads corrected by overlap analysis: 3915\n",
      "bases corrected by overlap analysis: 6348\n",
      "\n",
      "Duplication rate: 13.7931%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 20327\n",
      "% of original read pairs: 25.4954%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/RS31_fastp.json\n",
      "HTML report: ./fastp_out/RS31_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/RS31.R1.fastq -I 2_demultiplexed/ADD/RS31.R2.fastq -o 3_qc/ADD/RS31.R1.fastq -O 3_qc/ADD/RS31.R2.fastq --unpaired1 3_qc/ADD/RS31_unpaired.R1.fastq --unpaired2 3_qc/ADD/RS31_unpaired.R2.fastq --failed_out 3_qc/ADD/RS31_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/RS31_merged.fastq -w 6 -h ./fastp_out/RS31_fastp.html -j ./fastp_out/RS31_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 29447\n",
      "total bases: 3543799\n",
      "Q20 bases: 3354679(94.6634%)\n",
      "Q30 bases: 3145692(88.7661%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 29447\n",
      "total bases: 3334324\n",
      "Q20 bases: 3232476(96.9455%)\n",
      "Q30 bases: 3069659(92.0624%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 15223\n",
      "total bases: 1570287\n",
      "Q20 bases: 1551394(98.7968%)\n",
      "Q30 bases: 1498167(95.4072%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 34746\n",
      "reads failed due to low quality: 5920\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 18224\n",
      "reads with adapter trimmed: 35020\n",
      "bases trimmed due to adapters: 349943\n",
      "reads corrected by overlap analysis: 3768\n",
      "bases corrected by overlap analysis: 6353\n",
      "\n",
      "Duplication rate: 32.967%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 15223\n",
      "% of original read pairs: 51.6963%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/RS49_fastp.json\n",
      "HTML report: ./fastp_out/RS49_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/RS49.R1.fastq -I 2_demultiplexed/ADD/RS49.R2.fastq -o 3_qc/ADD/RS49.R1.fastq -O 3_qc/ADD/RS49.R2.fastq --unpaired1 3_qc/ADD/RS49_unpaired.R1.fastq --unpaired2 3_qc/ADD/RS49_unpaired.R2.fastq --failed_out 3_qc/ADD/RS49_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/RS49_merged.fastq -w 6 -h ./fastp_out/RS49_fastp.html -j ./fastp_out/RS49_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 93345\n",
      "total bases: 8226268\n",
      "Q20 bases: 7559461(91.8942%)\n",
      "Q30 bases: 6951558(84.5044%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 93345\n",
      "total bases: 7530856\n",
      "Q20 bases: 7275883(96.6143%)\n",
      "Q30 bases: 6940021(92.1545%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 19448\n",
      "total bases: 2003011\n",
      "Q20 bases: 1979807(98.8415%)\n",
      "Q30 bases: 1912791(95.4958%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 39732\n",
      "reads failed due to low quality: 24304\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 122654\n",
      "reads with adapter trimmed: 39604\n",
      "bases trimmed due to adapters: 500441\n",
      "reads corrected by overlap analysis: 3985\n",
      "bases corrected by overlap analysis: 6645\n",
      "\n",
      "Duplication rate: 15%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 19448\n",
      "% of original read pairs: 20.8345%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/RS51_fastp.json\n",
      "HTML report: ./fastp_out/RS51_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/RS51.R1.fastq -I 2_demultiplexed/ADD/RS51.R2.fastq -o 3_qc/ADD/RS51.R1.fastq -O 3_qc/ADD/RS51.R2.fastq --unpaired1 3_qc/ADD/RS51_unpaired.R1.fastq --unpaired2 3_qc/ADD/RS51_unpaired.R2.fastq --failed_out 3_qc/ADD/RS51_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/RS51_merged.fastq -w 6 -h ./fastp_out/RS51_fastp.html -j ./fastp_out/RS51_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 139473\n",
      "total bases: 21079426\n",
      "Q20 bases: 20597396(97.7133%)\n",
      "Q30 bases: 19737321(93.6331%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 139473\n",
      "total bases: 21030712\n",
      "Q20 bases: 20479843(97.3806%)\n",
      "Q30 bases: 19503128(92.7364%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 130824\n",
      "total bases: 13585204\n",
      "Q20 bases: 13417439(98.7651%)\n",
      "Q30 bases: 12949274(95.319%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 262912\n",
      "reads failed due to low quality: 7042\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 8986\n",
      "reads with adapter trimmed: 264858\n",
      "bases trimmed due to adapters: 2839847\n",
      "reads corrected by overlap analysis: 26746\n",
      "bases corrected by overlap analysis: 45957\n",
      "\n",
      "Duplication rate: 13.6752%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 144\n",
      "\n",
      "Read pairs merged: 130824\n",
      "% of original read pairs: 93.7988%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/S1ot_fastp.json\n",
      "HTML report: ./fastp_out/S1ot_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/S1ot.R1.fastq -I 2_demultiplexed/ADD/S1ot.R2.fastq -o 3_qc/ADD/S1ot.R1.fastq -O 3_qc/ADD/S1ot.R2.fastq --unpaired1 3_qc/ADD/S1ot_unpaired.R1.fastq --unpaired2 3_qc/ADD/S1ot_unpaired.R2.fastq --failed_out 3_qc/ADD/S1ot_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/S1ot_merged.fastq -w 6 -h ./fastp_out/S1ot_fastp.html -j ./fastp_out/S1ot_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 55578\n",
      "total bases: 7570838\n",
      "Q20 bases: 7280026(96.1588%)\n",
      "Q30 bases: 6888387(90.9858%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 55578\n",
      "total bases: 7356101\n",
      "Q20 bases: 7173969(97.5241%)\n",
      "Q30 bases: 6846652(93.0745%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 41383\n",
      "total bases: 4298015\n",
      "Q20 bases: 4238730(98.6206%)\n",
      "Q30 bases: 4076115(94.8372%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 83780\n",
      "reads failed due to low quality: 7082\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 20292\n",
      "reads with adapter trimmed: 84526\n",
      "bases trimmed due to adapters: 1074849\n",
      "reads corrected by overlap analysis: 8867\n",
      "bases corrected by overlap analysis: 15073\n",
      "\n",
      "Duplication rate: 31.5315%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 144\n",
      "\n",
      "Read pairs merged: 41383\n",
      "% of original read pairs: 74.4593%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/S3ot_fastp.json\n",
      "HTML report: ./fastp_out/S3ot_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/S3ot.R1.fastq -I 2_demultiplexed/ADD/S3ot.R2.fastq -o 3_qc/ADD/S3ot.R1.fastq -O 3_qc/ADD/S3ot.R2.fastq --unpaired1 3_qc/ADD/S3ot_unpaired.R1.fastq --unpaired2 3_qc/ADD/S3ot_unpaired.R2.fastq --failed_out 3_qc/ADD/S3ot_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/S3ot_merged.fastq -w 6 -h ./fastp_out/S3ot_fastp.html -j ./fastp_out/S3ot_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 53082\n",
      "total bases: 6831407\n",
      "Q20 bases: 6610346(96.764%)\n",
      "Q30 bases: 6261826(91.6623%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 53082\n",
      "total bases: 6771465\n",
      "Q20 bases: 6567760(96.9917%)\n",
      "Q30 bases: 6220377(91.8616%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 34342\n",
      "total bases: 3570128\n",
      "Q20 bases: 3517886(98.5367%)\n",
      "Q30 bases: 3377873(94.6149%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 79604\n",
      "reads failed due to low quality: 5428\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 21130\n",
      "reads with adapter trimmed: 70342\n",
      "bases trimmed due to adapters: 830280\n",
      "reads corrected by overlap analysis: 7967\n",
      "bases corrected by overlap analysis: 13725\n",
      "\n",
      "Duplication rate: 31.7308%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 144\n",
      "\n",
      "Read pairs merged: 34342\n",
      "% of original read pairs: 64.6961%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/S5ot_fastp.json\n",
      "HTML report: ./fastp_out/S5ot_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/ADD/S5ot.R1.fastq -I 2_demultiplexed/ADD/S5ot.R2.fastq -o 3_qc/ADD/S5ot.R1.fastq -O 3_qc/ADD/S5ot.R2.fastq --unpaired1 3_qc/ADD/S5ot_unpaired.R1.fastq --unpaired2 3_qc/ADD/S5ot_unpaired.R2.fastq --failed_out 3_qc/ADD/S5ot_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/ADD/S5ot_merged.fastq -w 6 -h ./fastp_out/S5ot_fastp.html -j ./fastp_out/S5ot_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 77605\n",
      "total bases: 11991458\n",
      "Q20 bases: 11722883(97.7603%)\n",
      "Q30 bases: 11230012(93.6501%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 77605\n",
      "total bases: 11994898\n",
      "Q20 bases: 11715398(97.6698%)\n",
      "Q30 bases: 11206978(93.4312%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 74397\n",
      "total bases: 7631187\n",
      "Q20 bases: 7535470(98.7457%)\n",
      "Q30 bases: 7268571(95.2482%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 149648\n",
      "reads failed due to low quality: 4178\n",
      "reads failed due to too many N: 8\n",
      "reads failed due to too short: 1376\n",
      "reads with adapter trimmed: 151768\n",
      "bases trimmed due to adapters: 1891927\n",
      "reads corrected by overlap analysis: 14044\n",
      "bases corrected by overlap analysis: 23270\n",
      "\n",
      "Duplication rate: 25.5814%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 74397\n",
      "% of original read pairs: 95.8662%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A01W_fastp.json\n",
      "HTML report: ./fastp_out/A01W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A01W.R1.fastq -I 2_demultiplexed/AW1/A01W.R2.fastq -o 3_qc/AW1/A01W.R1.fastq -O 3_qc/AW1/A01W.R2.fastq --unpaired1 3_qc/AW1/A01W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A01W_unpaired.R2.fastq --failed_out 3_qc/AW1/A01W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A01W_merged.fastq -w 6 -h ./fastp_out/A01W_fastp.html -j ./fastp_out/A01W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 168890\n",
      "total bases: 25977656\n",
      "Q20 bases: 25412061(97.8228%)\n",
      "Q30 bases: 24373767(93.8259%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 168890\n",
      "total bases: 25956997\n",
      "Q20 bases: 25349078(97.658%)\n",
      "Q30 bases: 24252783(93.4345%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 161386\n",
      "total bases: 16546526\n",
      "Q20 bases: 16348755(98.8048%)\n",
      "Q30 bases: 15790150(95.4288%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 326090\n",
      "reads failed due to low quality: 9120\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 2564\n",
      "reads with adapter trimmed: 329868\n",
      "bases trimmed due to adapters: 3835240\n",
      "reads corrected by overlap analysis: 28512\n",
      "bases corrected by overlap analysis: 48219\n",
      "\n",
      "Duplication rate: 48.2143%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 161386\n",
      "% of original read pairs: 95.5569%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A02W_fastp.json\n",
      "HTML report: ./fastp_out/A02W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A02W.R1.fastq -I 2_demultiplexed/AW1/A02W.R2.fastq -o 3_qc/AW1/A02W.R1.fastq -O 3_qc/AW1/A02W.R2.fastq --unpaired1 3_qc/AW1/A02W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A02W_unpaired.R2.fastq --failed_out 3_qc/AW1/A02W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A02W_merged.fastq -w 6 -h ./fastp_out/A02W_fastp.html -j ./fastp_out/A02W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 108337\n",
      "total bases: 16569772\n",
      "Q20 bases: 16229849(97.9485%)\n",
      "Q30 bases: 15559695(93.9041%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 108337\n",
      "total bases: 16561960\n",
      "Q20 bases: 16165507(97.6062%)\n",
      "Q30 bases: 15400864(92.9894%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 104963\n",
      "total bases: 10800106\n",
      "Q20 bases: 10677101(98.8611%)\n",
      "Q30 bases: 10319146(95.5467%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 210116\n",
      "reads failed due to low quality: 4506\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 2050\n",
      "reads with adapter trimmed: 212620\n",
      "bases trimmed due to adapters: 2263167\n",
      "reads corrected by overlap analysis: 17222\n",
      "bases corrected by overlap analysis: 28398\n",
      "\n",
      "Duplication rate: 39.3548%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 104963\n",
      "% of original read pairs: 96.8856%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A03W_fastp.json\n",
      "HTML report: ./fastp_out/A03W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A03W.R1.fastq -I 2_demultiplexed/AW1/A03W.R2.fastq -o 3_qc/AW1/A03W.R1.fastq -O 3_qc/AW1/A03W.R2.fastq --unpaired1 3_qc/AW1/A03W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A03W_unpaired.R2.fastq --failed_out 3_qc/AW1/A03W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A03W_merged.fastq -w 6 -h ./fastp_out/A03W_fastp.html -j ./fastp_out/A03W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 97383\n",
      "total bases: 15063982\n",
      "Q20 bases: 14730419(97.7857%)\n",
      "Q30 bases: 14090624(93.5385%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 97383\n",
      "total bases: 15060405\n",
      "Q20 bases: 14669716(97.4059%)\n",
      "Q30 bases: 13935956(92.5337%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 93495\n",
      "total bases: 9621699\n",
      "Q20 bases: 9508801(98.8266%)\n",
      "Q30 bases: 9183966(95.4506%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 187810\n",
      "reads failed due to low quality: 4776\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 2176\n",
      "reads with adapter trimmed: 189966\n",
      "bases trimmed due to adapters: 2413035\n",
      "reads corrected by overlap analysis: 16292\n",
      "bases corrected by overlap analysis: 27147\n",
      "\n",
      "Duplication rate: 55.8935%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 93495\n",
      "% of original read pairs: 96.0075%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A04W_fastp.json\n",
      "HTML report: ./fastp_out/A04W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A04W.R1.fastq -I 2_demultiplexed/AW1/A04W.R2.fastq -o 3_qc/AW1/A04W.R1.fastq -O 3_qc/AW1/A04W.R2.fastq --unpaired1 3_qc/AW1/A04W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A04W_unpaired.R2.fastq --failed_out 3_qc/AW1/A04W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A04W_merged.fastq -w 6 -h ./fastp_out/A04W_fastp.html -j ./fastp_out/A04W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 179756\n",
      "total bases: 27263949\n",
      "Q20 bases: 26685215(97.8773%)\n",
      "Q30 bases: 25627954(93.9994%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 179756\n",
      "total bases: 27226711\n",
      "Q20 bases: 26594344(97.6774%)\n",
      "Q30 bases: 25439971(93.4375%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 168359\n",
      "total bases: 17278552\n",
      "Q20 bases: 17086025(98.8857%)\n",
      "Q30 bases: 16535170(95.6977%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 339396\n",
      "reads failed due to low quality: 9136\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 10976\n",
      "reads with adapter trimmed: 342660\n",
      "bases trimmed due to adapters: 3991346\n",
      "reads corrected by overlap analysis: 27493\n",
      "bases corrected by overlap analysis: 45575\n",
      "\n",
      "Duplication rate: 45.614%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 168359\n",
      "% of original read pairs: 93.6597%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A05W_fastp.json\n",
      "HTML report: ./fastp_out/A05W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A05W.R1.fastq -I 2_demultiplexed/AW1/A05W.R2.fastq -o 3_qc/AW1/A05W.R1.fastq -O 3_qc/AW1/A05W.R2.fastq --unpaired1 3_qc/AW1/A05W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A05W_unpaired.R2.fastq --failed_out 3_qc/AW1/A05W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A05W_merged.fastq -w 6 -h ./fastp_out/A05W_fastp.html -j ./fastp_out/A05W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 130906\n",
      "total bases: 20119761\n",
      "Q20 bases: 19733975(98.0826%)\n",
      "Q30 bases: 18999084(94.43%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 130906\n",
      "total bases: 20114249\n",
      "Q20 bases: 19605547(97.4709%)\n",
      "Q30 bases: 18680555(92.8722%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 127717\n",
      "total bases: 13148932\n",
      "Q20 bases: 13012058(98.959%)\n",
      "Q30 bases: 12613924(95.9312%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 255788\n",
      "reads failed due to low quality: 5428\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 594\n",
      "reads with adapter trimmed: 258744\n",
      "bases trimmed due to adapters: 2743559\n",
      "reads corrected by overlap analysis: 20206\n",
      "bases corrected by overlap analysis: 33673\n",
      "\n",
      "Duplication rate: 43.0233%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 127717\n",
      "% of original read pairs: 97.5639%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A06W_fastp.json\n",
      "HTML report: ./fastp_out/A06W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A06W.R1.fastq -I 2_demultiplexed/AW1/A06W.R2.fastq -o 3_qc/AW1/A06W.R1.fastq -O 3_qc/AW1/A06W.R2.fastq --unpaired1 3_qc/AW1/A06W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A06W_unpaired.R2.fastq --failed_out 3_qc/AW1/A06W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A06W_merged.fastq -w 6 -h ./fastp_out/A06W_fastp.html -j ./fastp_out/A06W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 180288\n",
      "total bases: 27888930\n",
      "Q20 bases: 27386658(98.199%)\n",
      "Q30 bases: 26404473(94.6773%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 180288\n",
      "total bases: 27888997\n",
      "Q20 bases: 27361203(98.1075%)\n",
      "Q30 bases: 26331512(94.4154%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 174733\n",
      "total bases: 17974030\n",
      "Q20 bases: 17788629(98.9685%)\n",
      "Q30 bases: 17253056(95.9888%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 349828\n",
      "reads failed due to low quality: 7234\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 3510\n",
      "reads with adapter trimmed: 353904\n",
      "bases trimmed due to adapters: 4442828\n",
      "reads corrected by overlap analysis: 25206\n",
      "bases corrected by overlap analysis: 41395\n",
      "\n",
      "Duplication rate: 51.9337%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 174733\n",
      "% of original read pairs: 96.9188%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A07W_fastp.json\n",
      "HTML report: ./fastp_out/A07W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A07W.R1.fastq -I 2_demultiplexed/AW1/A07W.R2.fastq -o 3_qc/AW1/A07W.R1.fastq -O 3_qc/AW1/A07W.R2.fastq --unpaired1 3_qc/AW1/A07W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A07W_unpaired.R2.fastq --failed_out 3_qc/AW1/A07W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A07W_merged.fastq -w 6 -h ./fastp_out/A07W_fastp.html -j ./fastp_out/A07W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 97166\n",
      "total bases: 14907144\n",
      "Q20 bases: 14633815(98.1665%)\n",
      "Q30 bases: 14104904(94.6184%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 97166\n",
      "total bases: 14900569\n",
      "Q20 bases: 14562873(97.7337%)\n",
      "Q30 bases: 13937148(93.5343%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 94611\n",
      "total bases: 9740590\n",
      "Q20 bases: 9641301(98.9807%)\n",
      "Q30 bases: 9349410(95.984%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 189510\n",
      "reads failed due to low quality: 3858\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 964\n",
      "reads with adapter trimmed: 191422\n",
      "bases trimmed due to adapters: 2023650\n",
      "reads corrected by overlap analysis: 14138\n",
      "bases corrected by overlap analysis: 23456\n",
      "\n",
      "Duplication rate: 43.2749%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 94611\n",
      "% of original read pairs: 97.3705%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A08W_fastp.json\n",
      "HTML report: ./fastp_out/A08W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A08W.R1.fastq -I 2_demultiplexed/AW1/A08W.R2.fastq -o 3_qc/AW1/A08W.R1.fastq -O 3_qc/AW1/A08W.R2.fastq --unpaired1 3_qc/AW1/A08W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A08W_unpaired.R2.fastq --failed_out 3_qc/AW1/A08W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A08W_merged.fastq -w 6 -h ./fastp_out/A08W_fastp.html -j ./fastp_out/A08W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 122727\n",
      "total bases: 18988678\n",
      "Q20 bases: 18628731(98.1044%)\n",
      "Q30 bases: 17903426(94.2847%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 122727\n",
      "total bases: 19001079\n",
      "Q20 bases: 18567745(97.7194%)\n",
      "Q30 bases: 17726041(93.2897%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 119022\n",
      "total bases: 12252247\n",
      "Q20 bases: 12112919(98.8628%)\n",
      "Q30 bases: 11714768(95.6132%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 238446\n",
      "reads failed due to low quality: 4762\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 2240\n",
      "reads with adapter trimmed: 240808\n",
      "bases trimmed due to adapters: 3013774\n",
      "reads corrected by overlap analysis: 20924\n",
      "bases corrected by overlap analysis: 34278\n",
      "\n",
      "Duplication rate: 61.8812%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 119022\n",
      "% of original read pairs: 96.9811%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A09W_fastp.json\n",
      "HTML report: ./fastp_out/A09W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A09W.R1.fastq -I 2_demultiplexed/AW1/A09W.R2.fastq -o 3_qc/AW1/A09W.R1.fastq -O 3_qc/AW1/A09W.R2.fastq --unpaired1 3_qc/AW1/A09W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A09W_unpaired.R2.fastq --failed_out 3_qc/AW1/A09W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A09W_merged.fastq -w 6 -h ./fastp_out/A09W_fastp.html -j ./fastp_out/A09W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 51674\n",
      "total bases: 7952006\n",
      "Q20 bases: 7795473(98.0315%)\n",
      "Q30 bases: 7493936(94.2396%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 51674\n",
      "total bases: 7948828\n",
      "Q20 bases: 7771040(97.7633%)\n",
      "Q30 bases: 7439521(93.5927%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 49976\n",
      "total bases: 5144658\n",
      "Q20 bases: 5087990(98.8985%)\n",
      "Q30 bases: 4926406(95.7577%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 100116\n",
      "reads failed due to low quality: 2300\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 932\n",
      "reads with adapter trimmed: 101256\n",
      "bases trimmed due to adapters: 1176823\n",
      "reads corrected by overlap analysis: 8442\n",
      "bases corrected by overlap analysis: 13747\n",
      "\n",
      "Duplication rate: 40.708%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 49976\n",
      "% of original read pairs: 96.714%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A10W_fastp.json\n",
      "HTML report: ./fastp_out/A10W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A10W.R1.fastq -I 2_demultiplexed/AW1/A10W.R2.fastq -o 3_qc/AW1/A10W.R1.fastq -O 3_qc/AW1/A10W.R2.fastq --unpaired1 3_qc/AW1/A10W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A10W_unpaired.R2.fastq --failed_out 3_qc/AW1/A10W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A10W_merged.fastq -w 6 -h ./fastp_out/A10W_fastp.html -j ./fastp_out/A10W_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 90349\n",
      "total bases: 13804712\n",
      "Q20 bases: 13487507(97.7022%)\n",
      "Q30 bases: 12897950(93.4315%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 90349\n",
      "total bases: 13789899\n",
      "Q20 bases: 13427840(97.3745%)\n",
      "Q30 bases: 12765374(92.5705%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 86960\n",
      "total bases: 8949763\n",
      "Q20 bases: 8840043(98.774%)\n",
      "Q30 bases: 8529211(95.301%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 174132\n",
      "reads failed due to low quality: 4660\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 1904\n",
      "reads with adapter trimmed: 176600\n",
      "bases trimmed due to adapters: 1882046\n",
      "reads corrected by overlap analysis: 16034\n",
      "bases corrected by overlap analysis: 27098\n",
      "\n",
      "Duplication rate: 15.3846%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 86960\n",
      "% of original read pairs: 96.249%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A11W_fastp.json\n",
      "HTML report: ./fastp_out/A11W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A11W.R1.fastq -I 2_demultiplexed/AW1/A11W.R2.fastq -o 3_qc/AW1/A11W.R1.fastq -O 3_qc/AW1/A11W.R2.fastq --unpaired1 3_qc/AW1/A11W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A11W_unpaired.R2.fastq --failed_out 3_qc/AW1/A11W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A11W_merged.fastq -w 6 -h ./fastp_out/A11W_fastp.html -j ./fastp_out/A11W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 121136\n",
      "total bases: 17238880\n",
      "Q20 bases: 16878844(97.9115%)\n",
      "Q30 bases: 16231017(94.1535%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 121136\n",
      "total bases: 17198262\n",
      "Q20 bases: 16804933(97.713%)\n",
      "Q30 bases: 16112290(93.6856%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 102409\n",
      "total bases: 10535015\n",
      "Q20 bases: 10422847(98.9353%)\n",
      "Q30 bases: 10104841(95.9167%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 205218\n",
      "reads failed due to low quality: 7074\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 29976\n",
      "reads with adapter trimmed: 207462\n",
      "bases trimmed due to adapters: 2399760\n",
      "reads corrected by overlap analysis: 16780\n",
      "bases corrected by overlap analysis: 28340\n",
      "\n",
      "Duplication rate: 19.6078%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 102409\n",
      "% of original read pairs: 84.5405%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A12W_fastp.json\n",
      "HTML report: ./fastp_out/A12W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A12W.R1.fastq -I 2_demultiplexed/AW1/A12W.R2.fastq -o 3_qc/AW1/A12W.R1.fastq -O 3_qc/AW1/A12W.R2.fastq --unpaired1 3_qc/AW1/A12W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A12W_unpaired.R2.fastq --failed_out 3_qc/AW1/A12W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A12W_merged.fastq -w 6 -h ./fastp_out/A12W_fastp.html -j ./fastp_out/A12W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 77025\n",
      "total bases: 11613772\n",
      "Q20 bases: 11344140(97.6783%)\n",
      "Q30 bases: 10870637(93.6013%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 77025\n",
      "total bases: 11573289\n",
      "Q20 bases: 11313833(97.7581%)\n",
      "Q30 bases: 10850826(93.7575%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 72084\n",
      "total bases: 7405394\n",
      "Q20 bases: 7318354(98.8246%)\n",
      "Q30 bases: 7071674(95.4936%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 145414\n",
      "reads failed due to low quality: 4638\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 3994\n",
      "reads with adapter trimmed: 146898\n",
      "bases trimmed due to adapters: 1565164\n",
      "reads corrected by overlap analysis: 11803\n",
      "bases corrected by overlap analysis: 19562\n",
      "\n",
      "Duplication rate: 28.3019%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 72084\n",
      "% of original read pairs: 93.5852%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A13W_fastp.json\n",
      "HTML report: ./fastp_out/A13W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A13W.R1.fastq -I 2_demultiplexed/AW1/A13W.R2.fastq -o 3_qc/AW1/A13W.R1.fastq -O 3_qc/AW1/A13W.R2.fastq --unpaired1 3_qc/AW1/A13W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A13W_unpaired.R2.fastq --failed_out 3_qc/AW1/A13W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A13W_merged.fastq -w 6 -h ./fastp_out/A13W_fastp.html -j ./fastp_out/A13W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 95847\n",
      "total bases: 13292934\n",
      "Q20 bases: 12935381(97.3102%)\n",
      "Q30 bases: 12347784(92.8898%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 95847\n",
      "total bases: 13216510\n",
      "Q20 bases: 12909607(97.6779%)\n",
      "Q30 bases: 12354918(93.4809%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 75163\n",
      "total bases: 7723885\n",
      "Q20 bases: 7632075(98.8113%)\n",
      "Q30 bases: 7372904(95.4559%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 151474\n",
      "reads failed due to low quality: 7730\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 32486\n",
      "reads with adapter trimmed: 153494\n",
      "bases trimmed due to adapters: 1951375\n",
      "reads corrected by overlap analysis: 12761\n",
      "bases corrected by overlap analysis: 21275\n",
      "\n",
      "Duplication rate: 23.0769%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 75163\n",
      "% of original read pairs: 78.4198%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A14W_fastp.json\n",
      "HTML report: ./fastp_out/A14W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A14W.R1.fastq -I 2_demultiplexed/AW1/A14W.R2.fastq -o 3_qc/AW1/A14W.R1.fastq -O 3_qc/AW1/A14W.R2.fastq --unpaired1 3_qc/AW1/A14W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A14W_unpaired.R2.fastq --failed_out 3_qc/AW1/A14W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A14W_merged.fastq -w 6 -h ./fastp_out/A14W_fastp.html -j ./fastp_out/A14W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 98272\n",
      "total bases: 12022713\n",
      "Q20 bases: 11610181(96.5687%)\n",
      "Q30 bases: 11071812(92.0908%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 98272\n",
      "total bases: 11751957\n",
      "Q20 bases: 11486213(97.7387%)\n",
      "Q30 bases: 11033728(93.8884%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 60461\n",
      "total bases: 6227189\n",
      "Q20 bases: 6161596(98.9467%)\n",
      "Q30 bases: 5972703(95.9133%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 122156\n",
      "reads failed due to low quality: 12324\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 62060\n",
      "reads with adapter trimmed: 123566\n",
      "bases trimmed due to adapters: 1305187\n",
      "reads corrected by overlap analysis: 8740\n",
      "bases corrected by overlap analysis: 14533\n",
      "\n",
      "Duplication rate: 22.449%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 60461\n",
      "% of original read pairs: 61.5241%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A15W_fastp.json\n",
      "HTML report: ./fastp_out/A15W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A15W.R1.fastq -I 2_demultiplexed/AW1/A15W.R2.fastq -o 3_qc/AW1/A15W.R1.fastq -O 3_qc/AW1/A15W.R2.fastq --unpaired1 3_qc/AW1/A15W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A15W_unpaired.R2.fastq --failed_out 3_qc/AW1/A15W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A15W_merged.fastq -w 6 -h ./fastp_out/A15W_fastp.html -j ./fastp_out/A15W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 118389\n",
      "total bases: 17907967\n",
      "Q20 bases: 17539399(97.9419%)\n",
      "Q30 bases: 16837318(94.0214%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 118389\n",
      "total bases: 17893110\n",
      "Q20 bases: 17518306(97.9053%)\n",
      "Q30 bases: 16804521(93.9162%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 110066\n",
      "total bases: 11330107\n",
      "Q20 bases: 11200948(98.86%)\n",
      "Q30 bases: 10834526(95.626%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 220916\n",
      "reads failed due to low quality: 5940\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 9920\n",
      "reads with adapter trimmed: 223342\n",
      "bases trimmed due to adapters: 2795028\n",
      "reads corrected by overlap analysis: 17646\n",
      "bases corrected by overlap analysis: 29629\n",
      "\n",
      "Duplication rate: 20.1835%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 110066\n",
      "% of original read pairs: 92.9698%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A16W_fastp.json\n",
      "HTML report: ./fastp_out/A16W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A16W.R1.fastq -I 2_demultiplexed/AW1/A16W.R2.fastq -o 3_qc/AW1/A16W.R1.fastq -O 3_qc/AW1/A16W.R2.fastq --unpaired1 3_qc/AW1/A16W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A16W_unpaired.R2.fastq --failed_out 3_qc/AW1/A16W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A16W_merged.fastq -w 6 -h ./fastp_out/A16W_fastp.html -j ./fastp_out/A16W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 91049\n",
      "total bases: 14008765\n",
      "Q20 bases: 13718113(97.9252%)\n",
      "Q30 bases: 13166915(93.9905%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 91049\n",
      "total bases: 13990469\n",
      "Q20 bases: 13667203(97.6894%)\n",
      "Q30 bases: 13077364(93.4734%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 86674\n",
      "total bases: 8916553\n",
      "Q20 bases: 8818530(98.9007%)\n",
      "Q30 bases: 8537715(95.7513%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 175462\n",
      "reads failed due to low quality: 4228\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2408\n",
      "reads with adapter trimmed: 177382\n",
      "bases trimmed due to adapters: 2058062\n",
      "reads corrected by overlap analysis: 14500\n",
      "bases corrected by overlap analysis: 24337\n",
      "\n",
      "Duplication rate: 32.7586%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 86674\n",
      "% of original read pairs: 95.1949%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A17W_fastp.json\n",
      "HTML report: ./fastp_out/A17W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A17W.R1.fastq -I 2_demultiplexed/AW1/A17W.R2.fastq -o 3_qc/AW1/A17W.R1.fastq -O 3_qc/AW1/A17W.R2.fastq --unpaired1 3_qc/AW1/A17W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A17W_unpaired.R2.fastq --failed_out 3_qc/AW1/A17W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A17W_merged.fastq -w 6 -h ./fastp_out/A17W_fastp.html -j ./fastp_out/A17W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 141963\n",
      "total bases: 20600939\n",
      "Q20 bases: 20141253(97.7686%)\n",
      "Q30 bases: 19328260(93.8222%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 141963\n",
      "total bases: 20548040\n",
      "Q20 bases: 20011736(97.39%)\n",
      "Q30 bases: 19076501(92.8385%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 125368\n",
      "total bases: 12904810\n",
      "Q20 bases: 12760413(98.8811%)\n",
      "Q30 bases: 12349701(95.6984%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 251058\n",
      "reads failed due to low quality: 8364\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 24500\n",
      "reads with adapter trimmed: 254030\n",
      "bases trimmed due to adapters: 2713955\n",
      "reads corrected by overlap analysis: 20977\n",
      "bases corrected by overlap analysis: 35181\n",
      "\n",
      "Duplication rate: 22.695%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 125368\n",
      "% of original read pairs: 88.3103%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A18W_fastp.json\n",
      "HTML report: ./fastp_out/A18W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A18W.R1.fastq -I 2_demultiplexed/AW1/A18W.R2.fastq -o 3_qc/AW1/A18W.R1.fastq -O 3_qc/AW1/A18W.R2.fastq --unpaired1 3_qc/AW1/A18W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A18W_unpaired.R2.fastq --failed_out 3_qc/AW1/A18W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A18W_merged.fastq -w 6 -h ./fastp_out/A18W_fastp.html -j ./fastp_out/A18W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 122600\n",
      "total bases: 18726289\n",
      "Q20 bases: 18349087(97.9857%)\n",
      "Q30 bases: 17637053(94.1834%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 122600\n",
      "total bases: 18680259\n",
      "Q20 bases: 18313786(98.0382%)\n",
      "Q30 bases: 17596782(94.1999%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 115741\n",
      "total bases: 11917720\n",
      "Q20 bases: 11784904(98.8856%)\n",
      "Q30 bases: 11405423(95.7014%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 231620\n",
      "reads failed due to low quality: 6046\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 7528\n",
      "reads with adapter trimmed: 234530\n",
      "bases trimmed due to adapters: 2914732\n",
      "reads corrected by overlap analysis: 18420\n",
      "bases corrected by overlap analysis: 30089\n",
      "\n",
      "Duplication rate: 19.2982%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 115741\n",
      "% of original read pairs: 94.4054%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A19W_fastp.json\n",
      "HTML report: ./fastp_out/A19W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A19W.R1.fastq -I 2_demultiplexed/AW1/A19W.R2.fastq -o 3_qc/AW1/A19W.R1.fastq -O 3_qc/AW1/A19W.R2.fastq --unpaired1 3_qc/AW1/A19W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A19W_unpaired.R2.fastq --failed_out 3_qc/AW1/A19W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A19W_merged.fastq -w 6 -h ./fastp_out/A19W_fastp.html -j ./fastp_out/A19W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 117525\n",
      "total bases: 17916936\n",
      "Q20 bases: 17643470(98.4737%)\n",
      "Q30 bases: 17054511(95.1865%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 117525\n",
      "total bases: 17918041\n",
      "Q20 bases: 17575336(98.0874%)\n",
      "Q30 bases: 16900696(94.3222%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 112286\n",
      "total bases: 11562427\n",
      "Q20 bases: 11456714(99.0857%)\n",
      "Q30 bases: 11140908(96.3544%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 226532\n",
      "reads failed due to low quality: 3614\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 4902\n",
      "reads with adapter trimmed: 227686\n",
      "bases trimmed due to adapters: 2603553\n",
      "reads corrected by overlap analysis: 15883\n",
      "bases corrected by overlap analysis: 25956\n",
      "\n",
      "Duplication rate: 26.6667%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 112286\n",
      "% of original read pairs: 95.5422%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A20W_fastp.json\n",
      "HTML report: ./fastp_out/A20W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A20W.R1.fastq -I 2_demultiplexed/AW1/A20W.R2.fastq -o 3_qc/AW1/A20W.R1.fastq -O 3_qc/AW1/A20W.R2.fastq --unpaired1 3_qc/AW1/A20W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A20W_unpaired.R2.fastq --failed_out 3_qc/AW1/A20W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A20W_merged.fastq -w 6 -h ./fastp_out/A20W_fastp.html -j ./fastp_out/A20W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 84372\n",
      "total bases: 12759514\n",
      "Q20 bases: 12466612(97.7044%)\n",
      "Q30 bases: 11967215(93.7905%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 84372\n",
      "total bases: 12688666\n",
      "Q20 bases: 12434551(97.9973%)\n",
      "Q30 bases: 11959173(94.2508%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 78954\n",
      "total bases: 8130445\n",
      "Q20 bases: 8038504(98.8692%)\n",
      "Q30 bases: 7780082(95.6907%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 159022\n",
      "reads failed due to low quality: 4900\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 4822\n",
      "reads with adapter trimmed: 160450\n",
      "bases trimmed due to adapters: 1692835\n",
      "reads corrected by overlap analysis: 12026\n",
      "bases corrected by overlap analysis: 20117\n",
      "\n",
      "Duplication rate: 16.5138%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 78954\n",
      "% of original read pairs: 93.5784%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A21W_fastp.json\n",
      "HTML report: ./fastp_out/A21W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/A21W.R1.fastq -I 2_demultiplexed/AW1/A21W.R2.fastq -o 3_qc/AW1/A21W.R1.fastq -O 3_qc/AW1/A21W.R2.fastq --unpaired1 3_qc/AW1/A21W_unpaired.R1.fastq --unpaired2 3_qc/AW1/A21W_unpaired.R2.fastq --failed_out 3_qc/AW1/A21W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/A21W_merged.fastq -w 6 -h ./fastp_out/A21W_fastp.html -j ./fastp_out/A21W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 5198\n",
      "total bases: 289732\n",
      "Q20 bases: 265443(91.6167%)\n",
      "Q30 bases: 242788(83.7974%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 5198\n",
      "total bases: 268929\n",
      "Q20 bases: 253470(94.2516%)\n",
      "Q30 bases: 237030(88.1385%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 2\n",
      "total bases: 206\n",
      "Q20 bases: 203(98.5437%)\n",
      "Q30 bases: 193(93.6893%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 4\n",
      "reads failed due to low quality: 1296\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 9096\n",
      "reads with adapter trimmed: 4\n",
      "bases trimmed due to adapters: 79\n",
      "reads corrected by overlap analysis: 25\n",
      "bases corrected by overlap analysis: 38\n",
      "\n",
      "Duplication rate: 16.129%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 202\n",
      "\n",
      "Read pairs merged: 2\n",
      "% of original read pairs: 0.0384763%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/AF3W_fastp.json\n",
      "HTML report: ./fastp_out/AF3W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/AF3W.R1.fastq -I 2_demultiplexed/AW1/AF3W.R2.fastq -o 3_qc/AW1/AF3W.R1.fastq -O 3_qc/AW1/AF3W.R2.fastq --unpaired1 3_qc/AW1/AF3W_unpaired.R1.fastq --unpaired2 3_qc/AW1/AF3W_unpaired.R2.fastq --failed_out 3_qc/AW1/AF3W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/AF3W_merged.fastq -w 6 -h ./fastp_out/AF3W_fastp.html -j ./fastp_out/AF3W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 41877\n",
      "total bases: 3010825\n",
      "Q20 bases: 2692853(89.439%)\n",
      "Q30 bases: 2431777(80.7678%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 41877\n",
      "total bases: 2622621\n",
      "Q20 bases: 2498707(95.2752%)\n",
      "Q30 bases: 2355898(89.8299%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 0\n",
      "total bases: 0\n",
      "Q20 bases: 0(-nan%)\n",
      "Q30 bases: 0(-nan%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 108\n",
      "reads failed due to low quality: 13184\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 70462\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "reads corrected by overlap analysis: 274\n",
      "bases corrected by overlap analysis: 489\n",
      "\n",
      "Duplication rate: 13.4328%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 210\n",
      "\n",
      "Read pairs merged: 0\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/AN1W_fastp.json\n",
      "HTML report: ./fastp_out/AN1W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/AN1W.R1.fastq -I 2_demultiplexed/AW1/AN1W.R2.fastq -o 3_qc/AW1/AN1W.R1.fastq -O 3_qc/AW1/AN1W.R2.fastq --unpaired1 3_qc/AW1/AN1W_unpaired.R1.fastq --unpaired2 3_qc/AW1/AN1W_unpaired.R2.fastq --failed_out 3_qc/AW1/AN1W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/AN1W_merged.fastq -w 6 -h ./fastp_out/AN1W_fastp.html -j ./fastp_out/AN1W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 87622\n",
      "total bases: 13538572\n",
      "Q20 bases: 13263428(97.9677%)\n",
      "Q30 bases: 12725087(93.9914%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 87622\n",
      "total bases: 13558993\n",
      "Q20 bases: 13232998(97.5957%)\n",
      "Q30 bases: 12611076(93.0089%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 85547\n",
      "total bases: 8724673\n",
      "Q20 bases: 8615451(98.7481%)\n",
      "Q30 bases: 8310071(95.2479%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 171308\n",
      "reads failed due to low quality: 3502\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 430\n",
      "reads with adapter trimmed: 173168\n",
      "bases trimmed due to adapters: 2195358\n",
      "reads corrected by overlap analysis: 17800\n",
      "bases corrected by overlap analysis: 28936\n",
      "\n",
      "Duplication rate: 80.1339%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 85547\n",
      "% of original read pairs: 97.6319%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/AP1W_fastp.json\n",
      "HTML report: ./fastp_out/AP1W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW1/AP1W.R1.fastq -I 2_demultiplexed/AW1/AP1W.R2.fastq -o 3_qc/AW1/AP1W.R1.fastq -O 3_qc/AW1/AP1W.R2.fastq --unpaired1 3_qc/AW1/AP1W_unpaired.R1.fastq --unpaired2 3_qc/AW1/AP1W_unpaired.R2.fastq --failed_out 3_qc/AW1/AP1W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW1/AP1W_merged.fastq -w 6 -h ./fastp_out/AP1W_fastp.html -j ./fastp_out/AP1W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 108291\n",
      "total bases: 16494627\n",
      "Q20 bases: 16285208(98.7304%)\n",
      "Q30 bases: 15840817(96.0362%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 108291\n",
      "total bases: 16567292\n",
      "Q20 bases: 16008184(96.6252%)\n",
      "Q30 bases: 15068153(90.9512%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 102301\n",
      "total bases: 10550940\n",
      "Q20 bases: 10477338(99.3024%)\n",
      "Q30 bases: 10244011(97.091%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 206310\n",
      "reads failed due to low quality: 4298\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 5968\n",
      "reads with adapter trimmed: 206896\n",
      "bases trimmed due to adapters: 2573835\n",
      "reads corrected by overlap analysis: 17482\n",
      "bases corrected by overlap analysis: 30852\n",
      "\n",
      "Duplication rate: 20.904%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 102301\n",
      "% of original read pairs: 94.4686%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A22W_fastp.json\n",
      "HTML report: ./fastp_out/A22W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A22W.R1.fastq -I 2_demultiplexed/AW2/A22W.R2.fastq -o 3_qc/AW2/A22W.R1.fastq -O 3_qc/AW2/A22W.R2.fastq --unpaired1 3_qc/AW2/A22W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A22W_unpaired.R2.fastq --failed_out 3_qc/AW2/A22W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A22W_merged.fastq -w 6 -h ./fastp_out/A22W_fastp.html -j ./fastp_out/A22W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 48429\n",
      "total bases: 7004539\n",
      "Q20 bases: 6908235(98.6251%)\n",
      "Q30 bases: 6714599(95.8607%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 48429\n",
      "total bases: 7042837\n",
      "Q20 bases: 6795888(96.4936%)\n",
      "Q30 bases: 6406919(90.9707%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 41054\n",
      "total bases: 4221956\n",
      "Q20 bases: 4193132(99.3173%)\n",
      "Q30 bases: 4101675(97.1511%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 85872\n",
      "reads failed due to low quality: 2766\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 8218\n",
      "reads with adapter trimmed: 85172\n",
      "bases trimmed due to adapters: 989165\n",
      "reads corrected by overlap analysis: 7378\n",
      "bases corrected by overlap analysis: 13100\n",
      "\n",
      "Duplication rate: 21.4286%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 41054\n",
      "% of original read pairs: 84.7715%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A23W_fastp.json\n",
      "HTML report: ./fastp_out/A23W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A23W.R1.fastq -I 2_demultiplexed/AW2/A23W.R2.fastq -o 3_qc/AW2/A23W.R1.fastq -O 3_qc/AW2/A23W.R2.fastq --unpaired1 3_qc/AW2/A23W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A23W_unpaired.R2.fastq --failed_out 3_qc/AW2/A23W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A23W_merged.fastq -w 6 -h ./fastp_out/A23W_fastp.html -j ./fastp_out/A23W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 85628\n",
      "total bases: 12954311\n",
      "Q20 bases: 12795009(98.7703%)\n",
      "Q30 bases: 12446464(96.0797%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 85628\n",
      "total bases: 13024121\n",
      "Q20 bases: 12554547(96.3946%)\n",
      "Q30 bases: 11773998(90.4015%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 81866\n",
      "total bases: 8426521\n",
      "Q20 bases: 8370590(99.3363%)\n",
      "Q30 bases: 8188627(97.1768%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 164462\n",
      "reads failed due to low quality: 3098\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 3696\n",
      "reads with adapter trimmed: 165060\n",
      "bases trimmed due to adapters: 1757039\n",
      "reads corrected by overlap analysis: 13994\n",
      "bases corrected by overlap analysis: 24782\n",
      "\n",
      "Duplication rate: 15.3061%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 81866\n",
      "% of original read pairs: 95.6066%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A24W_fastp.json\n",
      "HTML report: ./fastp_out/A24W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A24W.R1.fastq -I 2_demultiplexed/AW2/A24W.R2.fastq -o 3_qc/AW2/A24W.R1.fastq -O 3_qc/AW2/A24W.R2.fastq --unpaired1 3_qc/AW2/A24W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A24W_unpaired.R2.fastq --failed_out 3_qc/AW2/A24W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A24W_merged.fastq -w 6 -h ./fastp_out/A24W_fastp.html -j ./fastp_out/A24W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 41545\n",
      "total bases: 6212583\n",
      "Q20 bases: 6129296(98.6594%)\n",
      "Q30 bases: 5952889(95.8199%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 41545\n",
      "total bases: 6252539\n",
      "Q20 bases: 6021460(96.3042%)\n",
      "Q30 bases: 5638161(90.1739%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 37760\n",
      "total bases: 3889616\n",
      "Q20 bases: 3862930(99.3139%)\n",
      "Q30 bases: 3777335(97.1133%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 76432\n",
      "reads failed due to low quality: 1910\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 4744\n",
      "reads with adapter trimmed: 76594\n",
      "bases trimmed due to adapters: 969953\n",
      "reads corrected by overlap analysis: 6960\n",
      "bases corrected by overlap analysis: 12357\n",
      "\n",
      "Duplication rate: 19.5876%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 37760\n",
      "% of original read pairs: 90.8894%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A25W_fastp.json\n",
      "HTML report: ./fastp_out/A25W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A25W.R1.fastq -I 2_demultiplexed/AW2/A25W.R2.fastq -o 3_qc/AW2/A25W.R1.fastq -O 3_qc/AW2/A25W.R2.fastq --unpaired1 3_qc/AW2/A25W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A25W_unpaired.R2.fastq --failed_out 3_qc/AW2/A25W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A25W_merged.fastq -w 6 -h ./fastp_out/A25W_fastp.html -j ./fastp_out/A25W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 130254\n",
      "total bases: 19590943\n",
      "Q20 bases: 19346468(98.7521%)\n",
      "Q30 bases: 18843730(96.1859%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 130254\n",
      "total bases: 19677519\n",
      "Q20 bases: 19037591(96.7479%)\n",
      "Q30 bases: 17991396(91.4312%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 120878\n",
      "total bases: 12432052\n",
      "Q20 bases: 12351547(99.3524%)\n",
      "Q30 bases: 12095421(97.2922%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 244972\n",
      "reads failed due to low quality: 4970\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 10566\n",
      "reads with adapter trimmed: 244788\n",
      "bases trimmed due to adapters: 2855444\n",
      "reads corrected by overlap analysis: 19953\n",
      "bases corrected by overlap analysis: 34746\n",
      "\n",
      "Duplication rate: 23.8095%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 120878\n",
      "% of original read pairs: 92.8018%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A26W_fastp.json\n",
      "HTML report: ./fastp_out/A26W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A26W.R1.fastq -I 2_demultiplexed/AW2/A26W.R2.fastq -o 3_qc/AW2/A26W.R1.fastq -O 3_qc/AW2/A26W.R2.fastq --unpaired1 3_qc/AW2/A26W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A26W_unpaired.R2.fastq --failed_out 3_qc/AW2/A26W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A26W_merged.fastq -w 6 -h ./fastp_out/A26W_fastp.html -j ./fastp_out/A26W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 101678\n",
      "total bases: 15426595\n",
      "Q20 bases: 15253201(98.876%)\n",
      "Q30 bases: 14879386(96.4528%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 101678\n",
      "total bases: 15497543\n",
      "Q20 bases: 14969537(96.593%)\n",
      "Q30 bases: 14099732(90.9804%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 97825\n",
      "total bases: 10072245\n",
      "Q20 bases: 10010590(99.3879%)\n",
      "Q30 bases: 9808977(97.3862%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 196988\n",
      "reads failed due to low quality: 3110\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 3254\n",
      "reads with adapter trimmed: 197248\n",
      "bases trimmed due to adapters: 2093732\n",
      "reads corrected by overlap analysis: 15342\n",
      "bases corrected by overlap analysis: 27676\n",
      "\n",
      "Duplication rate: 8.33333%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 97825\n",
      "% of original read pairs: 96.2106%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A27W_fastp.json\n",
      "HTML report: ./fastp_out/A27W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A27W.R1.fastq -I 2_demultiplexed/AW2/A27W.R2.fastq -o 3_qc/AW2/A27W.R1.fastq -O 3_qc/AW2/A27W.R2.fastq --unpaired1 3_qc/AW2/A27W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A27W_unpaired.R2.fastq --failed_out 3_qc/AW2/A27W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A27W_merged.fastq -w 6 -h ./fastp_out/A27W_fastp.html -j ./fastp_out/A27W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 169451\n",
      "total bases: 25934408\n",
      "Q20 bases: 25632288(98.8351%)\n",
      "Q30 bases: 24998655(96.3918%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 169451\n",
      "total bases: 26030611\n",
      "Q20 bases: 25274445(97.0951%)\n",
      "Q30 bases: 23986005(92.1454%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 162160\n",
      "total bases: 16652266\n",
      "Q20 bases: 16546331(99.3638%)\n",
      "Q30 bases: 16206195(97.3213%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 326006\n",
      "reads failed due to low quality: 5418\n",
      "reads failed due to too many N: 8\n",
      "reads failed due to too short: 7470\n",
      "reads with adapter trimmed: 326638\n",
      "bases trimmed due to adapters: 4115499\n",
      "reads corrected by overlap analysis: 24718\n",
      "bases corrected by overlap analysis: 42922\n",
      "\n",
      "Duplication rate: 23.2759%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 162160\n",
      "% of original read pairs: 95.6973%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A28W_fastp.json\n",
      "HTML report: ./fastp_out/A28W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A28W.R1.fastq -I 2_demultiplexed/AW2/A28W.R2.fastq -o 3_qc/AW2/A28W.R1.fastq -O 3_qc/AW2/A28W.R2.fastq --unpaired1 3_qc/AW2/A28W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A28W_unpaired.R2.fastq --failed_out 3_qc/AW2/A28W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A28W_merged.fastq -w 6 -h ./fastp_out/A28W_fastp.html -j ./fastp_out/A28W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 143008\n",
      "total bases: 21903921\n",
      "Q20 bases: 21653637(98.8574%)\n",
      "Q30 bases: 21117510(96.4097%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 143008\n",
      "total bases: 22008493\n",
      "Q20 bases: 21230784(96.4663%)\n",
      "Q30 bases: 19965604(90.7177%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 140049\n",
      "total bases: 14408433\n",
      "Q20 bases: 14317953(99.372%)\n",
      "Q30 bases: 14021683(97.3158%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 280464\n",
      "reads failed due to low quality: 4558\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 994\n",
      "reads with adapter trimmed: 281714\n",
      "bases trimmed due to adapters: 2994462\n",
      "reads corrected by overlap analysis: 22594\n",
      "bases corrected by overlap analysis: 40241\n",
      "\n",
      "Duplication rate: 14.7541%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 140049\n",
      "% of original read pairs: 97.9309%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A29W_fastp.json\n",
      "HTML report: ./fastp_out/A29W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A29W.R1.fastq -I 2_demultiplexed/AW2/A29W.R2.fastq -o 3_qc/AW2/A29W.R1.fastq -O 3_qc/AW2/A29W.R2.fastq --unpaired1 3_qc/AW2/A29W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A29W_unpaired.R2.fastq --failed_out 3_qc/AW2/A29W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A29W_merged.fastq -w 6 -h ./fastp_out/A29W_fastp.html -j ./fastp_out/A29W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 98443\n",
      "total bases: 15053224\n",
      "Q20 bases: 14891209(98.9237%)\n",
      "Q30 bases: 14530396(96.5268%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 98443\n",
      "total bases: 15123382\n",
      "Q20 bases: 14655909(96.9089%)\n",
      "Q30 bases: 13853310(91.6019%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 94407\n",
      "total bases: 9720766\n",
      "Q20 bases: 9662695(99.4026%)\n",
      "Q30 bases: 9473795(97.4593%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 189284\n",
      "reads failed due to low quality: 3122\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 4474\n",
      "reads with adapter trimmed: 189960\n",
      "bases trimmed due to adapters: 2368370\n",
      "reads corrected by overlap analysis: 14185\n",
      "bases corrected by overlap analysis: 24321\n",
      "\n",
      "Duplication rate: 37.6147%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 94407\n",
      "% of original read pairs: 95.9002%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A30W_fastp.json\n",
      "HTML report: ./fastp_out/A30W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A30W.R1.fastq -I 2_demultiplexed/AW2/A30W.R2.fastq -o 3_qc/AW2/A30W.R1.fastq -O 3_qc/AW2/A30W.R2.fastq --unpaired1 3_qc/AW2/A30W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A30W_unpaired.R2.fastq --failed_out 3_qc/AW2/A30W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A30W_merged.fastq -w 6 -h ./fastp_out/A30W_fastp.html -j ./fastp_out/A30W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 108241\n",
      "total bases: 16614603\n",
      "Q20 bases: 16434678(98.9171%)\n",
      "Q30 bases: 16034855(96.5106%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 108241\n",
      "total bases: 16689893\n",
      "Q20 bases: 16145793(96.7399%)\n",
      "Q30 bases: 15226741(91.2333%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 105373\n",
      "total bases: 10845741\n",
      "Q20 bases: 10778016(99.3756%)\n",
      "Q30 bases: 10561445(97.3787%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 211266\n",
      "reads failed due to low quality: 3318\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1898\n",
      "reads with adapter trimmed: 211910\n",
      "bases trimmed due to adapters: 2461384\n",
      "reads corrected by overlap analysis: 18621\n",
      "bases corrected by overlap analysis: 32440\n",
      "\n",
      "Duplication rate: 28.5714%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 105373\n",
      "% of original read pairs: 97.3504%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A31W_fastp.json\n",
      "HTML report: ./fastp_out/A31W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A31W.R1.fastq -I 2_demultiplexed/AW2/A31W.R2.fastq -o 3_qc/AW2/A31W.R1.fastq -O 3_qc/AW2/A31W.R2.fastq --unpaired1 3_qc/AW2/A31W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A31W_unpaired.R2.fastq --failed_out 3_qc/AW2/A31W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A31W_merged.fastq -w 6 -h ./fastp_out/A31W_fastp.html -j ./fastp_out/A31W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 30328\n",
      "total bases: 3732288\n",
      "Q20 bases: 3659990(98.0629%)\n",
      "Q30 bases: 3531152(94.6109%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 30328\n",
      "total bases: 3748453\n",
      "Q20 bases: 3587615(95.7092%)\n",
      "Q30 bases: 3341303(89.1382%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 20399\n",
      "total bases: 2098876\n",
      "Q20 bases: 2082560(99.2226%)\n",
      "Q30 bases: 2031327(96.7817%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 42974\n",
      "reads failed due to low quality: 2742\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 14938\n",
      "reads with adapter trimmed: 41932\n",
      "bases trimmed due to adapters: 447484\n",
      "reads corrected by overlap analysis: 3949\n",
      "bases corrected by overlap analysis: 7024\n",
      "\n",
      "Duplication rate: 9.45946%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 20399\n",
      "% of original read pairs: 67.2613%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A32W_fastp.json\n",
      "HTML report: ./fastp_out/A32W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A32W.R1.fastq -I 2_demultiplexed/AW2/A32W.R2.fastq -o 3_qc/AW2/A32W.R1.fastq -O 3_qc/AW2/A32W.R2.fastq --unpaired1 3_qc/AW2/A32W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A32W_unpaired.R2.fastq --failed_out 3_qc/AW2/A32W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A32W_merged.fastq -w 6 -h ./fastp_out/A32W_fastp.html -j ./fastp_out/A32W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 187507\n",
      "total bases: 27497878\n",
      "Q20 bases: 27170971(98.8112%)\n",
      "Q30 bases: 26488397(96.3289%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 187507\n",
      "total bases: 27614331\n",
      "Q20 bases: 26663850(96.558%)\n",
      "Q30 bases: 25135098(91.0219%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 168942\n",
      "total bases: 17393422\n",
      "Q20 bases: 17286559(99.3856%)\n",
      "Q30 bases: 16941987(97.4046%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 338328\n",
      "reads failed due to low quality: 7920\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 28764\n",
      "reads with adapter trimmed: 339514\n",
      "bases trimmed due to adapters: 3940892\n",
      "reads corrected by overlap analysis: 28804\n",
      "bases corrected by overlap analysis: 51114\n",
      "\n",
      "Duplication rate: 20.5357%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 168942\n",
      "% of original read pairs: 90.099%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A33W_fastp.json\n",
      "HTML report: ./fastp_out/A33W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A33W.R1.fastq -I 2_demultiplexed/AW2/A33W.R2.fastq -o 3_qc/AW2/A33W.R1.fastq -O 3_qc/AW2/A33W.R2.fastq --unpaired1 3_qc/AW2/A33W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A33W_unpaired.R2.fastq --failed_out 3_qc/AW2/A33W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A33W_merged.fastq -w 6 -h ./fastp_out/A33W_fastp.html -j ./fastp_out/A33W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 120349\n",
      "total bases: 18356172\n",
      "Q20 bases: 18147228(98.8617%)\n",
      "Q30 bases: 17694330(96.3944%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 120349\n",
      "total bases: 18419819\n",
      "Q20 bases: 17869056(97.0099%)\n",
      "Q30 bases: 16929536(91.9094%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 117268\n",
      "total bases: 12074319\n",
      "Q20 bases: 11996575(99.3561%)\n",
      "Q30 bases: 11744158(97.2656%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 235000\n",
      "reads failed due to low quality: 3624\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 2072\n",
      "reads with adapter trimmed: 236232\n",
      "bases trimmed due to adapters: 2505564\n",
      "reads corrected by overlap analysis: 16783\n",
      "bases corrected by overlap analysis: 29183\n",
      "\n",
      "Duplication rate: 56.5445%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 117268\n",
      "% of original read pairs: 97.4399%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A34W_fastp.json\n",
      "HTML report: ./fastp_out/A34W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A34W.R1.fastq -I 2_demultiplexed/AW2/A34W.R2.fastq -o 3_qc/AW2/A34W.R1.fastq -O 3_qc/AW2/A34W.R2.fastq --unpaired1 3_qc/AW2/A34W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A34W_unpaired.R2.fastq --failed_out 3_qc/AW2/A34W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A34W_merged.fastq -w 6 -h ./fastp_out/A34W_fastp.html -j ./fastp_out/A34W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 166620\n",
      "total bases: 24559358\n",
      "Q20 bases: 24226103(98.6431%)\n",
      "Q30 bases: 23554050(95.9066%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 166620\n",
      "total bases: 24663540\n",
      "Q20 bases: 23906796(96.9317%)\n",
      "Q30 bases: 22624944(91.7344%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 149441\n",
      "total bases: 15384685\n",
      "Q20 bases: 15281358(99.3284%)\n",
      "Q30 bases: 14952989(97.194%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 299632\n",
      "reads failed due to low quality: 7356\n",
      "reads failed due to too many N: 8\n",
      "reads failed due to too short: 26244\n",
      "reads with adapter trimmed: 300888\n",
      "bases trimmed due to adapters: 3804161\n",
      "reads corrected by overlap analysis: 22807\n",
      "bases corrected by overlap analysis: 38683\n",
      "\n",
      "Duplication rate: 20.3883%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 149441\n",
      "% of original read pairs: 89.6897%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A35W_fastp.json\n",
      "HTML report: ./fastp_out/A35W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A35W.R1.fastq -I 2_demultiplexed/AW2/A35W.R2.fastq -o 3_qc/AW2/A35W.R1.fastq -O 3_qc/AW2/A35W.R2.fastq --unpaired1 3_qc/AW2/A35W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A35W_unpaired.R2.fastq --failed_out 3_qc/AW2/A35W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A35W_merged.fastq -w 6 -h ./fastp_out/A35W_fastp.html -j ./fastp_out/A35W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 153689\n",
      "total bases: 21876445\n",
      "Q20 bases: 21590062(98.6909%)\n",
      "Q30 bases: 21025487(96.1102%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 153689\n",
      "total bases: 21947563\n",
      "Q20 bases: 21213704(96.6563%)\n",
      "Q30 bases: 20021626(91.2248%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 133062\n",
      "total bases: 13694297\n",
      "Q20 bases: 13605079(99.3485%)\n",
      "Q30 bases: 13318655(97.2569%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 266532\n",
      "reads failed due to low quality: 7228\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 33614\n",
      "reads with adapter trimmed: 267700\n",
      "bases trimmed due to adapters: 2840139\n",
      "reads corrected by overlap analysis: 20420\n",
      "bases corrected by overlap analysis: 36414\n",
      "\n",
      "Duplication rate: 26.4%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 133062\n",
      "% of original read pairs: 86.5787%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A36W_fastp.json\n",
      "HTML report: ./fastp_out/A36W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A36W.R1.fastq -I 2_demultiplexed/AW2/A36W.R2.fastq -o 3_qc/AW2/A36W.R1.fastq -O 3_qc/AW2/A36W.R2.fastq --unpaired1 3_qc/AW2/A36W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A36W_unpaired.R2.fastq --failed_out 3_qc/AW2/A36W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A36W_merged.fastq -w 6 -h ./fastp_out/A36W_fastp.html -j ./fastp_out/A36W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 149091\n",
      "total bases: 22497847\n",
      "Q20 bases: 22207553(98.7097%)\n",
      "Q30 bases: 21590681(95.9678%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 149091\n",
      "total bases: 22602475\n",
      "Q20 bases: 21807391(96.4823%)\n",
      "Q30 bases: 20480453(90.6116%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 136428\n",
      "total bases: 13956894\n",
      "Q20 bases: 13859287(99.3007%)\n",
      "Q30 bases: 13546681(97.0609%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 283728\n",
      "reads failed due to low quality: 6292\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 8156\n",
      "reads with adapter trimmed: 274542\n",
      "bases trimmed due to adapters: 3453301\n",
      "reads corrected by overlap analysis: 24427\n",
      "bases corrected by overlap analysis: 42535\n",
      "\n",
      "Duplication rate: 27.9279%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 136428\n",
      "% of original read pairs: 91.5065%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A37W_fastp.json\n",
      "HTML report: ./fastp_out/A37W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A37W.R1.fastq -I 2_demultiplexed/AW2/A37W.R2.fastq -o 3_qc/AW2/A37W.R1.fastq -O 3_qc/AW2/A37W.R2.fastq --unpaired1 3_qc/AW2/A37W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A37W_unpaired.R2.fastq --failed_out 3_qc/AW2/A37W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A37W_merged.fastq -w 6 -h ./fastp_out/A37W_fastp.html -j ./fastp_out/A37W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 198153\n",
      "total bases: 30538978\n",
      "Q20 bases: 30166468(98.7802%)\n",
      "Q30 bases: 29369548(96.1707%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 198153\n",
      "total bases: 30694192\n",
      "Q20 bases: 29599375(96.4331%)\n",
      "Q30 bases: 27800493(90.5725%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 193638\n",
      "total bases: 19935514\n",
      "Q20 bases: 19807119(99.3559%)\n",
      "Q30 bases: 19393027(97.2788%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 388050\n",
      "reads failed due to low quality: 6560\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1696\n",
      "reads with adapter trimmed: 389698\n",
      "bases trimmed due to adapters: 4557692\n",
      "reads corrected by overlap analysis: 34782\n",
      "bases corrected by overlap analysis: 60232\n",
      "\n",
      "Duplication rate: 42.8571%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 193638\n",
      "% of original read pairs: 97.7215%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A38W_fastp.json\n",
      "HTML report: ./fastp_out/A38W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A38W.R1.fastq -I 2_demultiplexed/AW2/A38W.R2.fastq -o 3_qc/AW2/A38W.R1.fastq -O 3_qc/AW2/A38W.R2.fastq --unpaired1 3_qc/AW2/A38W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A38W_unpaired.R2.fastq --failed_out 3_qc/AW2/A38W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A38W_merged.fastq -w 6 -h ./fastp_out/A38W_fastp.html -j ./fastp_out/A38W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 176518\n",
      "total bases: 24982637\n",
      "Q20 bases: 24607370(98.4979%)\n",
      "Q30 bases: 23892253(95.6354%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 176518\n",
      "total bases: 25110771\n",
      "Q20 bases: 24115099(96.0349%)\n",
      "Q30 bases: 22590436(89.9631%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 151402\n",
      "total bases: 15560598\n",
      "Q20 bases: 15454100(99.3156%)\n",
      "Q30 bases: 15111161(97.1117%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 303388\n",
      "reads failed due to low quality: 10118\n",
      "reads failed due to too many N: 10\n",
      "reads failed due to too short: 39520\n",
      "reads with adapter trimmed: 304914\n",
      "bases trimmed due to adapters: 3281575\n",
      "reads corrected by overlap analysis: 26576\n",
      "bases corrected by overlap analysis: 48602\n",
      "\n",
      "Duplication rate: 10.687%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 151402\n",
      "% of original read pairs: 85.7714%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A39W_fastp.json\n",
      "HTML report: ./fastp_out/A39W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A39W.R1.fastq -I 2_demultiplexed/AW2/A39W.R2.fastq -o 3_qc/AW2/A39W.R1.fastq -O 3_qc/AW2/A39W.R2.fastq --unpaired1 3_qc/AW2/A39W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A39W_unpaired.R2.fastq --failed_out 3_qc/AW2/A39W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A39W_merged.fastq -w 6 -h ./fastp_out/A39W_fastp.html -j ./fastp_out/A39W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 296763\n",
      "total bases: 45129430\n",
      "Q20 bases: 44559278(98.7366%)\n",
      "Q30 bases: 43354137(96.0662%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 296763\n",
      "total bases: 45269118\n",
      "Q20 bases: 43804800(96.7653%)\n",
      "Q30 bases: 41315318(91.266%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 281267\n",
      "total bases: 28933629\n",
      "Q20 bases: 28717991(99.2547%)\n",
      "Q30 bases: 28038423(96.906%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 563532\n",
      "reads failed due to low quality: 11288\n",
      "reads failed due to too many N: 12\n",
      "reads failed due to too short: 18694\n",
      "reads with adapter trimmed: 565998\n",
      "bases trimmed due to adapters: 7077539\n",
      "reads corrected by overlap analysis: 55321\n",
      "bases corrected by overlap analysis: 98412\n",
      "\n",
      "Duplication rate: 27.8409%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 281267\n",
      "% of original read pairs: 94.7783%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A40W_fastp.json\n",
      "HTML report: ./fastp_out/A40W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/A40W.R1.fastq -I 2_demultiplexed/AW2/A40W.R2.fastq -o 3_qc/AW2/A40W.R1.fastq -O 3_qc/AW2/A40W.R2.fastq --unpaired1 3_qc/AW2/A40W_unpaired.R1.fastq --unpaired2 3_qc/AW2/A40W_unpaired.R2.fastq --failed_out 3_qc/AW2/A40W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/A40W_merged.fastq -w 6 -h ./fastp_out/A40W_fastp.html -j ./fastp_out/A40W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 6793\n",
      "total bases: 370081\n",
      "Q20 bases: 343541(92.8286%)\n",
      "Q30 bases: 317246(85.7234%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 6793\n",
      "total bases: 366664\n",
      "Q20 bases: 333918(91.0692%)\n",
      "Q30 bases: 302523(82.5069%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 275\n",
      "total bases: 28320\n",
      "Q20 bases: 28100(99.2232%)\n",
      "Q30 bases: 27362(96.6172%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 552\n",
      "reads failed due to low quality: 1868\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 11166\n",
      "reads with adapter trimmed: 552\n",
      "bases trimmed due to adapters: 6431\n",
      "reads corrected by overlap analysis: 72\n",
      "bases corrected by overlap analysis: 138\n",
      "\n",
      "Duplication rate: 16.6667%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 275\n",
      "% of original read pairs: 4.04828%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/AF2W_fastp.json\n",
      "HTML report: ./fastp_out/AF2W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/AF2W.R1.fastq -I 2_demultiplexed/AW2/AF2W.R2.fastq -o 3_qc/AW2/AF2W.R1.fastq -O 3_qc/AW2/AF2W.R2.fastq --unpaired1 3_qc/AW2/AF2W_unpaired.R1.fastq --unpaired2 3_qc/AW2/AF2W_unpaired.R2.fastq --failed_out 3_qc/AW2/AF2W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/AF2W_merged.fastq -w 6 -h ./fastp_out/AF2W_fastp.html -j ./fastp_out/AF2W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 33296\n",
      "total bases: 2136464\n",
      "Q20 bases: 1980219(92.6867%)\n",
      "Q30 bases: 1836191(85.9453%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 33296\n",
      "total bases: 2012003\n",
      "Q20 bases: 1910361(94.9482%)\n",
      "Q30 bases: 1794170(89.1733%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 1\n",
      "total bases: 103\n",
      "Q20 bases: 103(100%)\n",
      "Q30 bases: 103(100%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 50\n",
      "reads failed due to low quality: 7642\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 58900\n",
      "reads with adapter trimmed: 2\n",
      "bases trimmed due to adapters: 22\n",
      "reads corrected by overlap analysis: 64\n",
      "bases corrected by overlap analysis: 102\n",
      "\n",
      "Duplication rate: 22.5%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 187\n",
      "\n",
      "Read pairs merged: 1\n",
      "% of original read pairs: 0.00300336%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/AN2W_fastp.json\n",
      "HTML report: ./fastp_out/AN2W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/AN2W.R1.fastq -I 2_demultiplexed/AW2/AN2W.R2.fastq -o 3_qc/AW2/AN2W.R1.fastq -O 3_qc/AW2/AN2W.R2.fastq --unpaired1 3_qc/AW2/AN2W_unpaired.R1.fastq --unpaired2 3_qc/AW2/AN2W_unpaired.R2.fastq --failed_out 3_qc/AW2/AN2W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/AN2W_merged.fastq -w 6 -h ./fastp_out/AN2W_fastp.html -j ./fastp_out/AN2W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 67995\n",
      "total bases: 10490272\n",
      "Q20 bases: 10370233(98.8557%)\n",
      "Q30 bases: 10101169(96.2908%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 67995\n",
      "total bases: 10573026\n",
      "Q20 bases: 10190894(96.3858%)\n",
      "Q30 bases: 9559231(90.4115%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 66845\n",
      "total bases: 6817333\n",
      "Q20 bases: 6770037(99.3062%)\n",
      "Q30 bases: 6616569(97.0551%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 133958\n",
      "reads failed due to low quality: 1820\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 212\n",
      "reads with adapter trimmed: 134312\n",
      "bases trimmed due to adapters: 1718396\n",
      "reads corrected by overlap analysis: 14032\n",
      "bases corrected by overlap analysis: 24766\n",
      "\n",
      "Duplication rate: 76.8328%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 66845\n",
      "% of original read pairs: 98.3087%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/AP2W_fastp.json\n",
      "HTML report: ./fastp_out/AP2W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW2/AP2W.R1.fastq -I 2_demultiplexed/AW2/AP2W.R2.fastq -o 3_qc/AW2/AP2W.R1.fastq -O 3_qc/AW2/AP2W.R2.fastq --unpaired1 3_qc/AW2/AP2W_unpaired.R1.fastq --unpaired2 3_qc/AW2/AP2W_unpaired.R2.fastq --failed_out 3_qc/AW2/AP2W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW2/AP2W_merged.fastq -w 6 -h ./fastp_out/AP2W_fastp.html -j ./fastp_out/AP2W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 45161\n",
      "total bases: 6165521\n",
      "Q20 bases: 6015963(97.5743%)\n",
      "Q30 bases: 5777719(93.7102%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 45161\n",
      "total bases: 6122668\n",
      "Q20 bases: 5929945(96.8523%)\n",
      "Q30 bases: 5614470(91.6997%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 34834\n",
      "total bases: 3590277\n",
      "Q20 bases: 3556259(99.0525%)\n",
      "Q30 bases: 3456708(96.2797%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 71354\n",
      "reads failed due to low quality: 4264\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 14702\n",
      "reads with adapter trimmed: 71742\n",
      "bases trimmed due to adapters: 883798\n",
      "reads corrected by overlap analysis: 6105\n",
      "bases corrected by overlap analysis: 10243\n",
      "\n",
      "Duplication rate: 27.7419%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 34834\n",
      "% of original read pairs: 77.1329%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A41W_fastp.json\n",
      "HTML report: ./fastp_out/A41W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A41W.R1.fastq -I 2_demultiplexed/AW3/A41W.R2.fastq -o 3_qc/AW3/A41W.R1.fastq -O 3_qc/AW3/A41W.R2.fastq --unpaired1 3_qc/AW3/A41W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A41W_unpaired.R2.fastq --failed_out 3_qc/AW3/A41W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A41W_merged.fastq -w 6 -h ./fastp_out/A41W_fastp.html -j ./fastp_out/A41W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 43914\n",
      "total bases: 5975441\n",
      "Q20 bases: 5860357(98.0741%)\n",
      "Q30 bases: 5655972(94.6536%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 43914\n",
      "total bases: 5951026\n",
      "Q20 bases: 5791447(97.3185%)\n",
      "Q30 bases: 5516404(92.6967%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 34071\n",
      "total bases: 3504953\n",
      "Q20 bases: 3474364(99.1273%)\n",
      "Q30 bases: 3382944(96.519%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 71140\n",
      "reads failed due to low quality: 3108\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 13580\n",
      "reads with adapter trimmed: 69904\n",
      "bases trimmed due to adapters: 806623\n",
      "reads corrected by overlap analysis: 5561\n",
      "bases corrected by overlap analysis: 9113\n",
      "\n",
      "Duplication rate: 21.6867%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 34071\n",
      "% of original read pairs: 77.5857%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A42W_fastp.json\n",
      "HTML report: ./fastp_out/A42W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A42W.R1.fastq -I 2_demultiplexed/AW3/A42W.R2.fastq -o 3_qc/AW3/A42W.R1.fastq -O 3_qc/AW3/A42W.R2.fastq --unpaired1 3_qc/AW3/A42W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A42W_unpaired.R2.fastq --failed_out 3_qc/AW3/A42W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A42W_merged.fastq -w 6 -h ./fastp_out/A42W_fastp.html -j ./fastp_out/A42W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 26766\n",
      "total bases: 3582636\n",
      "Q20 bases: 3508635(97.9345%)\n",
      "Q30 bases: 3376827(94.2554%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 26766\n",
      "total bases: 3592579\n",
      "Q20 bases: 3473603(96.6883%)\n",
      "Q30 bases: 3283168(91.3875%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 20381\n",
      "total bases: 2100681\n",
      "Q20 bases: 2081147(99.0701%)\n",
      "Q30 bases: 2023427(96.3224%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 41242\n",
      "reads failed due to low quality: 2216\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 10074\n",
      "reads with adapter trimmed: 41216\n",
      "bases trimmed due to adapters: 435105\n",
      "reads corrected by overlap analysis: 3376\n",
      "bases corrected by overlap analysis: 5599\n",
      "\n",
      "Duplication rate: 23.913%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 20381\n",
      "% of original read pairs: 76.1451%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A43W_fastp.json\n",
      "HTML report: ./fastp_out/A43W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A43W.R1.fastq -I 2_demultiplexed/AW3/A43W.R2.fastq -o 3_qc/AW3/A43W.R1.fastq -O 3_qc/AW3/A43W.R2.fastq --unpaired1 3_qc/AW3/A43W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A43W_unpaired.R2.fastq --failed_out 3_qc/AW3/A43W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A43W_merged.fastq -w 6 -h ./fastp_out/A43W_fastp.html -j ./fastp_out/A43W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 88285\n",
      "total bases: 13025346\n",
      "Q20 bases: 12771944(98.0545%)\n",
      "Q30 bases: 12312232(94.5252%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 88285\n",
      "total bases: 13007508\n",
      "Q20 bases: 12658032(97.3133%)\n",
      "Q30 bases: 12035215(92.5251%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 77975\n",
      "total bases: 8020226\n",
      "Q20 bases: 7948249(99.1026%)\n",
      "Q30 bases: 7733339(96.423%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 158396\n",
      "reads failed due to low quality: 5258\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 12912\n",
      "reads with adapter trimmed: 159072\n",
      "bases trimmed due to adapters: 2007443\n",
      "reads corrected by overlap analysis: 12397\n",
      "bases corrected by overlap analysis: 20450\n",
      "\n",
      "Duplication rate: 22.6667%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 77975\n",
      "% of original read pairs: 88.3219%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A44W_fastp.json\n",
      "HTML report: ./fastp_out/A44W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A44W.R1.fastq -I 2_demultiplexed/AW3/A44W.R2.fastq -o 3_qc/AW3/A44W.R1.fastq -O 3_qc/AW3/A44W.R2.fastq --unpaired1 3_qc/AW3/A44W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A44W_unpaired.R2.fastq --failed_out 3_qc/AW3/A44W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A44W_merged.fastq -w 6 -h ./fastp_out/A44W_fastp.html -j ./fastp_out/A44W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 106902\n",
      "total bases: 15142612\n",
      "Q20 bases: 14761024(97.48%)\n",
      "Q30 bases: 14184255(93.6711%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 106902\n",
      "total bases: 14986512\n",
      "Q20 bases: 14573446(97.2437%)\n",
      "Q30 bases: 13876055(92.5903%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 83304\n",
      "total bases: 8565437\n",
      "Q20 bases: 8487840(99.0941%)\n",
      "Q30 bases: 8258400(96.4154%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 175896\n",
      "reads failed due to low quality: 9872\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 28034\n",
      "reads with adapter trimmed: 175206\n",
      "bases trimmed due to adapters: 2023529\n",
      "reads corrected by overlap analysis: 14209\n",
      "bases corrected by overlap analysis: 23703\n",
      "\n",
      "Duplication rate: 14.9425%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 83304\n",
      "% of original read pairs: 77.9256%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A45W_fastp.json\n",
      "HTML report: ./fastp_out/A45W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A45W.R1.fastq -I 2_demultiplexed/AW3/A45W.R2.fastq -o 3_qc/AW3/A45W.R1.fastq -O 3_qc/AW3/A45W.R2.fastq --unpaired1 3_qc/AW3/A45W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A45W_unpaired.R2.fastq --failed_out 3_qc/AW3/A45W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A45W_merged.fastq -w 6 -h ./fastp_out/A45W_fastp.html -j ./fastp_out/A45W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 139006\n",
      "total bases: 20970194\n",
      "Q20 bases: 20660873(98.5249%)\n",
      "Q30 bases: 20036543(95.5477%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 139006\n",
      "total bases: 20986478\n",
      "Q20 bases: 20473325(97.5548%)\n",
      "Q30 bases: 19545203(93.1324%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 130747\n",
      "total bases: 13474905\n",
      "Q20 bases: 13365469(99.1879%)\n",
      "Q30 bases: 13033536(96.7245%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 265466\n",
      "reads failed due to low quality: 4718\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 7824\n",
      "reads with adapter trimmed: 266792\n",
      "bases trimmed due to adapters: 2804576\n",
      "reads corrected by overlap analysis: 18388\n",
      "bases corrected by overlap analysis: 30485\n",
      "\n",
      "Duplication rate: 12.2449%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 130747\n",
      "% of original read pairs: 94.0585%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A46W_fastp.json\n",
      "HTML report: ./fastp_out/A46W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A46W.R1.fastq -I 2_demultiplexed/AW3/A46W.R2.fastq -o 3_qc/AW3/A46W.R1.fastq -O 3_qc/AW3/A46W.R2.fastq --unpaired1 3_qc/AW3/A46W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A46W_unpaired.R2.fastq --failed_out 3_qc/AW3/A46W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A46W_merged.fastq -w 6 -h ./fastp_out/A46W_fastp.html -j ./fastp_out/A46W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 224338\n",
      "total bases: 34251459\n",
      "Q20 bases: 33714098(98.4311%)\n",
      "Q30 bases: 32671397(95.3869%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 224338\n",
      "total bases: 34242630\n",
      "Q20 bases: 33485770(97.7897%)\n",
      "Q30 bases: 32066259(93.6443%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 209780\n",
      "total bases: 21626218\n",
      "Q20 bases: 21445780(99.1657%)\n",
      "Q30 bases: 20902983(96.6557%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 426152\n",
      "reads failed due to low quality: 9490\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 13028\n",
      "reads with adapter trimmed: 425598\n",
      "bases trimmed due to adapters: 5331014\n",
      "reads corrected by overlap analysis: 30166\n",
      "bases corrected by overlap analysis: 48565\n",
      "\n",
      "Duplication rate: 20.6349%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 209780\n",
      "% of original read pairs: 93.5107%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A47W_fastp.json\n",
      "HTML report: ./fastp_out/A47W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A47W.R1.fastq -I 2_demultiplexed/AW3/A47W.R2.fastq -o 3_qc/AW3/A47W.R1.fastq -O 3_qc/AW3/A47W.R2.fastq --unpaired1 3_qc/AW3/A47W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A47W_unpaired.R2.fastq --failed_out 3_qc/AW3/A47W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A47W_merged.fastq -w 6 -h ./fastp_out/A47W_fastp.html -j ./fastp_out/A47W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 208410\n",
      "total bases: 31914185\n",
      "Q20 bases: 31439159(98.5116%)\n",
      "Q30 bases: 30481347(95.5103%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 208410\n",
      "total bases: 31949218\n",
      "Q20 bases: 31136447(97.4561%)\n",
      "Q30 bases: 29659789(92.8342%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 203355\n",
      "total bases: 20936267\n",
      "Q20 bases: 20761450(99.165%)\n",
      "Q30 bases: 20227592(96.6151%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 407210\n",
      "reads failed due to low quality: 7098\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 2506\n",
      "reads with adapter trimmed: 410296\n",
      "bases trimmed due to adapters: 4323753\n",
      "reads corrected by overlap analysis: 29487\n",
      "bases corrected by overlap analysis: 47824\n",
      "\n",
      "Duplication rate: 27.2152%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 203355\n",
      "% of original read pairs: 97.5745%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A48W_fastp.json\n",
      "HTML report: ./fastp_out/A48W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A48W.R1.fastq -I 2_demultiplexed/AW3/A48W.R2.fastq -o 3_qc/AW3/A48W.R1.fastq -O 3_qc/AW3/A48W.R2.fastq --unpaired1 3_qc/AW3/A48W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A48W_unpaired.R2.fastq --failed_out 3_qc/AW3/A48W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A48W_merged.fastq -w 6 -h ./fastp_out/A48W_fastp.html -j ./fastp_out/A48W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 189747\n",
      "total bases: 29118587\n",
      "Q20 bases: 28700575(98.5644%)\n",
      "Q30 bases: 27822684(95.5496%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 189747\n",
      "total bases: 29157528\n",
      "Q20 bases: 28495478(97.7294%)\n",
      "Q30 bases: 27228491(93.3841%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 183112\n",
      "total bases: 18836109\n",
      "Q20 bases: 18681732(99.1804%)\n",
      "Q30 bases: 18209091(96.6712%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 366942\n",
      "reads failed due to low quality: 6002\n",
      "reads failed due to too many N: 8\n",
      "reads failed due to too short: 6542\n",
      "reads with adapter trimmed: 369254\n",
      "bases trimmed due to adapters: 4600563\n",
      "reads corrected by overlap analysis: 26118\n",
      "bases corrected by overlap analysis: 42304\n",
      "\n",
      "Duplication rate: 13.5417%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 183112\n",
      "% of original read pairs: 96.5032%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A49W_fastp.json\n",
      "HTML report: ./fastp_out/A49W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A49W.R1.fastq -I 2_demultiplexed/AW3/A49W.R2.fastq -o 3_qc/AW3/A49W.R1.fastq -O 3_qc/AW3/A49W.R2.fastq --unpaired1 3_qc/AW3/A49W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A49W_unpaired.R2.fastq --failed_out 3_qc/AW3/A49W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A49W_merged.fastq -w 6 -h ./fastp_out/A49W_fastp.html -j ./fastp_out/A49W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 243468\n",
      "total bases: 37548688\n",
      "Q20 bases: 37019243(98.59%)\n",
      "Q30 bases: 35914550(95.6479%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 243468\n",
      "total bases: 37600382\n",
      "Q20 bases: 36659589(97.4979%)\n",
      "Q30 bases: 34939148(92.9223%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 238318\n",
      "total bases: 24522907\n",
      "Q20 bases: 24321837(99.1801%)\n",
      "Q30 bases: 23715367(96.707%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 477618\n",
      "reads failed due to low quality: 7686\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 1630\n",
      "reads with adapter trimmed: 480870\n",
      "bases trimmed due to adapters: 5557659\n",
      "reads corrected by overlap analysis: 36096\n",
      "bases corrected by overlap analysis: 59405\n",
      "\n",
      "Duplication rate: 10.7143%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 238318\n",
      "% of original read pairs: 97.8847%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A50W_fastp.json\n",
      "HTML report: ./fastp_out/A50W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A50W.R1.fastq -I 2_demultiplexed/AW3/A50W.R2.fastq -o 3_qc/AW3/A50W.R1.fastq -O 3_qc/AW3/A50W.R2.fastq --unpaired1 3_qc/AW3/A50W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A50W_unpaired.R2.fastq --failed_out 3_qc/AW3/A50W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A50W_merged.fastq -w 6 -h ./fastp_out/A50W_fastp.html -j ./fastp_out/A50W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 128633\n",
      "total bases: 19251145\n",
      "Q20 bases: 18894036(98.145%)\n",
      "Q30 bases: 18195991(94.519%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 128633\n",
      "total bases: 19263206\n",
      "Q20 bases: 18696473(97.058%)\n",
      "Q30 bases: 17689951(91.8328%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 119635\n",
      "total bases: 12304250\n",
      "Q20 bases: 12184317(99.0253%)\n",
      "Q30 bases: 11825263(96.1071%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 242206\n",
      "reads failed due to low quality: 6040\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 9018\n",
      "reads with adapter trimmed: 243788\n",
      "bases trimmed due to adapters: 2590482\n",
      "reads corrected by overlap analysis: 20711\n",
      "bases corrected by overlap analysis: 34598\n",
      "\n",
      "Duplication rate: 24.7863%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 119635\n",
      "% of original read pairs: 93.0049%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A51W_fastp.json\n",
      "HTML report: ./fastp_out/A51W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A51W.R1.fastq -I 2_demultiplexed/AW3/A51W.R2.fastq -o 3_qc/AW3/A51W.R1.fastq -O 3_qc/AW3/A51W.R2.fastq --unpaired1 3_qc/AW3/A51W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A51W_unpaired.R2.fastq --failed_out 3_qc/AW3/A51W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A51W_merged.fastq -w 6 -h ./fastp_out/A51W_fastp.html -j ./fastp_out/A51W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 116324\n",
      "total bases: 15469073\n",
      "Q20 bases: 15179816(98.1301%)\n",
      "Q30 bases: 14662852(94.7882%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 116324\n",
      "total bases: 15445597\n",
      "Q20 bases: 15052226(97.4532%)\n",
      "Q30 bases: 14376767(93.08%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 87195\n",
      "total bases: 8968772\n",
      "Q20 bases: 8892452(99.149%)\n",
      "Q30 bases: 8664227(96.6044%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 175140\n",
      "reads failed due to low quality: 7652\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 49854\n",
      "reads with adapter trimmed: 176190\n",
      "bases trimmed due to adapters: 2040670\n",
      "reads corrected by overlap analysis: 13246\n",
      "bases corrected by overlap analysis: 21888\n",
      "\n",
      "Duplication rate: 20.1835%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 87195\n",
      "% of original read pairs: 74.9587%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A52W_fastp.json\n",
      "HTML report: ./fastp_out/A52W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A52W.R1.fastq -I 2_demultiplexed/AW3/A52W.R2.fastq -o 3_qc/AW3/A52W.R1.fastq -O 3_qc/AW3/A52W.R2.fastq --unpaired1 3_qc/AW3/A52W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A52W_unpaired.R2.fastq --failed_out 3_qc/AW3/A52W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A52W_merged.fastq -w 6 -h ./fastp_out/A52W_fastp.html -j ./fastp_out/A52W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 117181\n",
      "total bases: 17872472\n",
      "Q20 bases: 17601310(98.4828%)\n",
      "Q30 bases: 17047075(95.3817%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 117181\n",
      "total bases: 17884577\n",
      "Q20 bases: 17458116(97.6155%)\n",
      "Q30 bases: 16670117(93.2095%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 111864\n",
      "total bases: 11511596\n",
      "Q20 bases: 11410690(99.1234%)\n",
      "Q30 bases: 11105150(96.4692%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 229054\n",
      "reads failed due to low quality: 4194\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 1110\n",
      "reads with adapter trimmed: 226162\n",
      "bases trimmed due to adapters: 2384034\n",
      "reads corrected by overlap analysis: 15944\n",
      "bases corrected by overlap analysis: 25774\n",
      "\n",
      "Duplication rate: 20.339%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 111864\n",
      "% of original read pairs: 95.4626%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A53W_fastp.json\n",
      "HTML report: ./fastp_out/A53W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A53W.R1.fastq -I 2_demultiplexed/AW3/A53W.R2.fastq -o 3_qc/AW3/A53W.R1.fastq -O 3_qc/AW3/A53W.R2.fastq --unpaired1 3_qc/AW3/A53W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A53W_unpaired.R2.fastq --failed_out 3_qc/AW3/A53W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A53W_merged.fastq -w 6 -h ./fastp_out/A53W_fastp.html -j ./fastp_out/A53W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 125571\n",
      "total bases: 18237437\n",
      "Q20 bases: 17887795(98.0828%)\n",
      "Q30 bases: 17258497(94.6323%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 125571\n",
      "total bases: 18202090\n",
      "Q20 bases: 17787528(97.7224%)\n",
      "Q30 bases: 17023405(93.5245%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 108601\n",
      "total bases: 11172449\n",
      "Q20 bases: 11072525(99.1056%)\n",
      "Q30 bases: 10775599(96.448%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 218368\n",
      "reads failed due to low quality: 7056\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 25716\n",
      "reads with adapter trimmed: 219826\n",
      "bases trimmed due to adapters: 2782244\n",
      "reads corrected by overlap analysis: 15659\n",
      "bases corrected by overlap analysis: 25072\n",
      "\n",
      "Duplication rate: 25.2427%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 108601\n",
      "% of original read pairs: 86.4857%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A54W_fastp.json\n",
      "HTML report: ./fastp_out/A54W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A54W.R1.fastq -I 2_demultiplexed/AW3/A54W.R2.fastq -o 3_qc/AW3/A54W.R1.fastq -O 3_qc/AW3/A54W.R2.fastq --unpaired1 3_qc/AW3/A54W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A54W_unpaired.R2.fastq --failed_out 3_qc/AW3/A54W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A54W_merged.fastq -w 6 -h ./fastp_out/A54W_fastp.html -j ./fastp_out/A54W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 110640\n",
      "total bases: 16921760\n",
      "Q20 bases: 16631453(98.2844%)\n",
      "Q30 bases: 16079165(95.0206%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 110640\n",
      "total bases: 16893226\n",
      "Q20 bases: 16504435(97.6985%)\n",
      "Q30 bases: 15781662(93.4201%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 107122\n",
      "total bases: 11019980\n",
      "Q20 bases: 10917163(99.067%)\n",
      "Q30 bases: 10614568(96.3211%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 215160\n",
      "reads failed due to low quality: 4474\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 1642\n",
      "reads with adapter trimmed: 216754\n",
      "bases trimmed due to adapters: 2283312\n",
      "reads corrected by overlap analysis: 15722\n",
      "bases corrected by overlap analysis: 25428\n",
      "\n",
      "Duplication rate: 12.381%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 107122\n",
      "% of original read pairs: 96.8203%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/A55W_fastp.json\n",
      "HTML report: ./fastp_out/A55W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/A55W.R1.fastq -I 2_demultiplexed/AW3/A55W.R2.fastq -o 3_qc/AW3/A55W.R1.fastq -O 3_qc/AW3/A55W.R2.fastq --unpaired1 3_qc/AW3/A55W_unpaired.R1.fastq --unpaired2 3_qc/AW3/A55W_unpaired.R2.fastq --failed_out 3_qc/AW3/A55W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/A55W_merged.fastq -w 6 -h ./fastp_out/A55W_fastp.html -j ./fastp_out/A55W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 5112\n",
      "total bases: 288868\n",
      "Q20 bases: 261399(90.4908%)\n",
      "Q30 bases: 237317(82.1541%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 5112\n",
      "total bases: 258724\n",
      "Q20 bases: 243364(94.0632%)\n",
      "Q30 bases: 227135(87.7905%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 0\n",
      "total bases: 0\n",
      "Q20 bases: 0(-nan%)\n",
      "Q30 bases: 0(-nan%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 0\n",
      "reads failed due to low quality: 1318\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 8906\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "reads corrected by overlap analysis: 20\n",
      "bases corrected by overlap analysis: 36\n",
      "\n",
      "Duplication rate: 6.89655%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 193\n",
      "\n",
      "Read pairs merged: 0\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/AF1W_fastp.json\n",
      "HTML report: ./fastp_out/AF1W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/AF1W.R1.fastq -I 2_demultiplexed/AW3/AF1W.R2.fastq -o 3_qc/AW3/AF1W.R1.fastq -O 3_qc/AW3/AF1W.R2.fastq --unpaired1 3_qc/AW3/AF1W_unpaired.R1.fastq --unpaired2 3_qc/AW3/AF1W_unpaired.R2.fastq --failed_out 3_qc/AW3/AF1W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/AF1W_merged.fastq -w 6 -h ./fastp_out/AF1W_fastp.html -j ./fastp_out/AF1W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 3266\n",
      "total bases: 151888\n",
      "Q20 bases: 143394(94.4077%)\n",
      "Q30 bases: 133822(88.1057%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 3266\n",
      "total bases: 152353\n",
      "Q20 bases: 141210(92.6861%)\n",
      "Q30 bases: 131831(86.53%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 1\n",
      "total bases: 103\n",
      "Q20 bases: 103(100%)\n",
      "Q30 bases: 101(98.0583%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2\n",
      "reads failed due to low quality: 632\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 5898\n",
      "reads with adapter trimmed: 2\n",
      "bases trimmed due to adapters: 22\n",
      "reads corrected by overlap analysis: 8\n",
      "bases corrected by overlap analysis: 16\n",
      "\n",
      "Duplication rate: 34.0909%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 1\n",
      "% of original read pairs: 0.0306185%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/AN3W_fastp.json\n",
      "HTML report: ./fastp_out/AN3W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/AN3W.R1.fastq -I 2_demultiplexed/AW3/AN3W.R2.fastq -o 3_qc/AW3/AN3W.R1.fastq -O 3_qc/AW3/AN3W.R2.fastq --unpaired1 3_qc/AW3/AN3W_unpaired.R1.fastq --unpaired2 3_qc/AW3/AN3W_unpaired.R2.fastq --failed_out 3_qc/AW3/AN3W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/AN3W_merged.fastq -w 6 -h ./fastp_out/AN3W_fastp.html -j ./fastp_out/AN3W_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 75454\n",
      "total bases: 11642399\n",
      "Q20 bases: 11464949(98.4758%)\n",
      "Q30 bases: 11092973(95.2808%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 75454\n",
      "total bases: 11685078\n",
      "Q20 bases: 11369653(97.3006%)\n",
      "Q30 bases: 10790903(92.3477%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 73944\n",
      "total bases: 7541335\n",
      "Q20 bases: 7470448(99.06%)\n",
      "Q30 bases: 7257914(96.2418%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 148144\n",
      "reads failed due to low quality: 2358\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 404\n",
      "reads with adapter trimmed: 149104\n",
      "bases trimmed due to adapters: 1885867\n",
      "reads corrected by overlap analysis: 14605\n",
      "bases corrected by overlap analysis: 23533\n",
      "\n",
      "Duplication rate: 78.7037%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 73944\n",
      "% of original read pairs: 97.9988%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/AP3W_fastp.json\n",
      "HTML report: ./fastp_out/AP3W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/AW3/AP3W.R1.fastq -I 2_demultiplexed/AW3/AP3W.R2.fastq -o 3_qc/AW3/AP3W.R1.fastq -O 3_qc/AW3/AP3W.R2.fastq --unpaired1 3_qc/AW3/AP3W_unpaired.R1.fastq --unpaired2 3_qc/AW3/AP3W_unpaired.R2.fastq --failed_out 3_qc/AW3/AP3W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/AW3/AP3W_merged.fastq -w 6 -h ./fastp_out/AP3W_fastp.html -j ./fastp_out/AP3W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 914117\n",
      "total bases: 142034625\n",
      "Q20 bases: 140349070(98.8133%)\n",
      "Q30 bases: 136732850(96.2673%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 914117\n",
      "total bases: 142534199\n",
      "Q20 bases: 138306892(97.0342%)\n",
      "Q30 bases: 131009145(91.9142%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 898582\n",
      "total bases: 92570138\n",
      "Q20 bases: 91949941(99.33%)\n",
      "Q30 bases: 89981945(97.2041%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1799148\n",
      "reads failed due to low quality: 28418\n",
      "reads failed due to too many N: 32\n",
      "reads failed due to too short: 636\n",
      "reads with adapter trimmed: 1809886\n",
      "bases trimmed due to adapters: 22435587\n",
      "reads corrected by overlap analysis: 136132\n",
      "bases corrected by overlap analysis: 235887\n",
      "\n",
      "Duplication rate: 34.1365%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 898582\n",
      "% of original read pairs: 98.3005%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B01W_fastp.json\n",
      "HTML report: ./fastp_out/B01W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B01W.R1.fastq -I 2_demultiplexed/BW1/B01W.R2.fastq -o 3_qc/BW1/B01W.R1.fastq -O 3_qc/BW1/B01W.R2.fastq --unpaired1 3_qc/BW1/B01W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B01W_unpaired.R2.fastq --failed_out 3_qc/BW1/B01W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B01W_merged.fastq -w 6 -h ./fastp_out/B01W_fastp.html -j ./fastp_out/B01W_fastp.json \n",
      "fastp v0.20.0, time used: 6 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 122565\n",
      "total bases: 18774520\n",
      "Q20 bases: 18553340(98.8219%)\n",
      "Q30 bases: 18069322(96.2439%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 122565\n",
      "total bases: 18849272\n",
      "Q20 bases: 18266885(96.9103%)\n",
      "Q30 bases: 17285081(91.7016%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 117873\n",
      "total bases: 12101761\n",
      "Q20 bases: 12017439(99.3032%)\n",
      "Q30 bases: 11747559(97.0731%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 237982\n",
      "reads failed due to low quality: 3932\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 3216\n",
      "reads with adapter trimmed: 238752\n",
      "bases trimmed due to adapters: 2756355\n",
      "reads corrected by overlap analysis: 20698\n",
      "bases corrected by overlap analysis: 36860\n",
      "\n",
      "Duplication rate: 18.8119%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 117873\n",
      "% of original read pairs: 96.1718%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B02W_fastp.json\n",
      "HTML report: ./fastp_out/B02W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B02W.R1.fastq -I 2_demultiplexed/BW1/B02W.R2.fastq -o 3_qc/BW1/B02W.R1.fastq -O 3_qc/BW1/B02W.R2.fastq --unpaired1 3_qc/BW1/B02W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B02W_unpaired.R2.fastq --failed_out 3_qc/BW1/B02W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B02W_merged.fastq -w 6 -h ./fastp_out/B02W_fastp.html -j ./fastp_out/B02W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 31782\n",
      "total bases: 4709730\n",
      "Q20 bases: 4652970(98.7948%)\n",
      "Q30 bases: 4528956(96.1617%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 31782\n",
      "total bases: 4730443\n",
      "Q20 bases: 4583667(96.8972%)\n",
      "Q30 bases: 4333758(91.6142%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 29508\n",
      "total bases: 3037498\n",
      "Q20 bases: 3017613(99.3453%)\n",
      "Q30 bases: 2952307(97.1954%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 59186\n",
      "reads failed due to low quality: 1110\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 3268\n",
      "reads with adapter trimmed: 59346\n",
      "bases trimmed due to adapters: 626810\n",
      "reads corrected by overlap analysis: 4503\n",
      "bases corrected by overlap analysis: 7720\n",
      "\n",
      "Duplication rate: 11.4286%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 29508\n",
      "% of original read pairs: 92.845%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B03W_fastp.json\n",
      "HTML report: ./fastp_out/B03W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B03W.R1.fastq -I 2_demultiplexed/BW1/B03W.R2.fastq -o 3_qc/BW1/B03W.R1.fastq -O 3_qc/BW1/B03W.R2.fastq --unpaired1 3_qc/BW1/B03W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B03W_unpaired.R2.fastq --failed_out 3_qc/BW1/B03W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B03W_merged.fastq -w 6 -h ./fastp_out/B03W_fastp.html -j ./fastp_out/B03W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 87348\n",
      "total bases: 13309526\n",
      "Q20 bases: 13136312(98.6986%)\n",
      "Q30 bases: 12765927(95.9157%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 87348\n",
      "total bases: 13381001\n",
      "Q20 bases: 12935226(96.6686%)\n",
      "Q30 bases: 12187508(91.0807%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 80982\n",
      "total bases: 8322276\n",
      "Q20 bases: 8265471(99.3174%)\n",
      "Q30 bases: 8081300(97.1044%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 165864\n",
      "reads failed due to low quality: 3336\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 5494\n",
      "reads with adapter trimmed: 164432\n",
      "bases trimmed due to adapters: 2083186\n",
      "reads corrected by overlap analysis: 13793\n",
      "bases corrected by overlap analysis: 23808\n",
      "\n",
      "Duplication rate: 27.5591%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 80982\n",
      "% of original read pairs: 92.7119%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B04W_fastp.json\n",
      "HTML report: ./fastp_out/B04W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B04W.R1.fastq -I 2_demultiplexed/BW1/B04W.R2.fastq -o 3_qc/BW1/B04W.R1.fastq -O 3_qc/BW1/B04W.R2.fastq --unpaired1 3_qc/BW1/B04W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B04W_unpaired.R2.fastq --failed_out 3_qc/BW1/B04W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B04W_merged.fastq -w 6 -h ./fastp_out/B04W_fastp.html -j ./fastp_out/B04W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 156205\n",
      "total bases: 22953943\n",
      "Q20 bases: 22626229(98.5723%)\n",
      "Q30 bases: 21998240(95.8364%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 156205\n",
      "total bases: 23046308\n",
      "Q20 bases: 22283418(96.6898%)\n",
      "Q30 bases: 21066418(91.4091%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 139222\n",
      "total bases: 14247280\n",
      "Q20 bases: 14151050(99.3246%)\n",
      "Q30 bases: 13845387(97.1792%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 281914\n",
      "reads failed due to low quality: 7354\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 23142\n",
      "reads with adapter trimmed: 280674\n",
      "bases trimmed due to adapters: 3282856\n",
      "reads corrected by overlap analysis: 22440\n",
      "bases corrected by overlap analysis: 39021\n",
      "\n",
      "Duplication rate: 37.8378%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 139222\n",
      "% of original read pairs: 89.1277%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B05W_fastp.json\n",
      "HTML report: ./fastp_out/B05W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B05W.R1.fastq -I 2_demultiplexed/BW1/B05W.R2.fastq -o 3_qc/BW1/B05W.R1.fastq -O 3_qc/BW1/B05W.R2.fastq --unpaired1 3_qc/BW1/B05W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B05W_unpaired.R2.fastq --failed_out 3_qc/BW1/B05W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B05W_merged.fastq -w 6 -h ./fastp_out/B05W_fastp.html -j ./fastp_out/B05W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 71257\n",
      "total bases: 10886527\n",
      "Q20 bases: 10724174(98.5087%)\n",
      "Q30 bases: 10431165(95.8172%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 71257\n",
      "total bases: 10930180\n",
      "Q20 bases: 10563991(96.6497%)\n",
      "Q30 bases: 9980081(91.3076%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 68447\n",
      "total bases: 7035067\n",
      "Q20 bases: 6986904(99.3154%)\n",
      "Q30 bases: 6834792(97.1532%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 138638\n",
      "reads failed due to low quality: 2796\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 1078\n",
      "reads with adapter trimmed: 138048\n",
      "bases trimmed due to adapters: 1459052\n",
      "reads corrected by overlap analysis: 10415\n",
      "bases corrected by overlap analysis: 18402\n",
      "\n",
      "Duplication rate: 13.5802%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 68447\n",
      "% of original read pairs: 96.0565%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B06W_fastp.json\n",
      "HTML report: ./fastp_out/B06W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B06W.R1.fastq -I 2_demultiplexed/BW1/B06W.R2.fastq -o 3_qc/BW1/B06W.R1.fastq -O 3_qc/BW1/B06W.R2.fastq --unpaired1 3_qc/BW1/B06W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B06W_unpaired.R2.fastq --failed_out 3_qc/BW1/B06W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B06W_merged.fastq -w 6 -h ./fastp_out/B06W_fastp.html -j ./fastp_out/B06W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 38927\n",
      "total bases: 5682776\n",
      "Q20 bases: 5610052(98.7203%)\n",
      "Q30 bases: 5461787(96.1112%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 38927\n",
      "total bases: 5702044\n",
      "Q20 bases: 5543885(97.2263%)\n",
      "Q30 bases: 5270298(92.4282%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 32935\n",
      "total bases: 3409121\n",
      "Q20 bases: 3386169(99.3267%)\n",
      "Q30 bases: 3313692(97.2008%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 69248\n",
      "reads failed due to low quality: 1550\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 7056\n",
      "reads with adapter trimmed: 66326\n",
      "bases trimmed due to adapters: 832248\n",
      "reads corrected by overlap analysis: 5009\n",
      "bases corrected by overlap analysis: 8563\n",
      "\n",
      "Duplication rate: 20%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 32935\n",
      "% of original read pairs: 84.6071%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B07W_fastp.json\n",
      "HTML report: ./fastp_out/B07W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B07W.R1.fastq -I 2_demultiplexed/BW1/B07W.R2.fastq -o 3_qc/BW1/B07W.R1.fastq -O 3_qc/BW1/B07W.R2.fastq --unpaired1 3_qc/BW1/B07W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B07W_unpaired.R2.fastq --failed_out 3_qc/BW1/B07W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B07W_merged.fastq -w 6 -h ./fastp_out/B07W_fastp.html -j ./fastp_out/B07W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 20553\n",
      "total bases: 3127829\n",
      "Q20 bases: 3095708(98.9731%)\n",
      "Q30 bases: 3023856(96.6759%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 20553\n",
      "total bases: 3140000\n",
      "Q20 bases: 3050695(97.1559%)\n",
      "Q30 bases: 2896054(92.231%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 20007\n",
      "total bases: 2058757\n",
      "Q20 bases: 2046309(99.3954%)\n",
      "Q30 bases: 2005606(97.4183%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 40066\n",
      "reads failed due to low quality: 512\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 528\n",
      "reads with adapter trimmed: 40222\n",
      "bases trimmed due to adapters: 424150\n",
      "reads corrected by overlap analysis: 2798\n",
      "bases corrected by overlap analysis: 4896\n",
      "\n",
      "Duplication rate: 29.2683%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 20007\n",
      "% of original read pairs: 97.3435%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B08W_fastp.json\n",
      "HTML report: ./fastp_out/B08W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B08W.R1.fastq -I 2_demultiplexed/BW1/B08W.R2.fastq -o 3_qc/BW1/B08W.R1.fastq -O 3_qc/BW1/B08W.R2.fastq --unpaired1 3_qc/BW1/B08W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B08W_unpaired.R2.fastq --failed_out 3_qc/BW1/B08W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B08W_merged.fastq -w 6 -h ./fastp_out/B08W_fastp.html -j ./fastp_out/B08W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 32609\n",
      "total bases: 4995388\n",
      "Q20 bases: 4937192(98.835%)\n",
      "Q30 bases: 4811461(96.3181%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 32609\n",
      "total bases: 5020346\n",
      "Q20 bases: 4856950(96.7453%)\n",
      "Q30 bases: 4580391(91.2366%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 31435\n",
      "total bases: 3222903\n",
      "Q20 bases: 3202249(99.3591%)\n",
      "Q30 bases: 3135956(97.3022%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 62948\n",
      "reads failed due to low quality: 1022\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1248\n",
      "reads with adapter trimmed: 63226\n",
      "bases trimmed due to adapters: 795107\n",
      "reads corrected by overlap analysis: 5248\n",
      "bases corrected by overlap analysis: 9016\n",
      "\n",
      "Duplication rate: 60.8%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 31435\n",
      "% of original read pairs: 96.3998%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B09W_fastp.json\n",
      "HTML report: ./fastp_out/B09W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B09W.R1.fastq -I 2_demultiplexed/BW1/B09W.R2.fastq -o 3_qc/BW1/B09W.R1.fastq -O 3_qc/BW1/B09W.R2.fastq --unpaired1 3_qc/BW1/B09W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B09W_unpaired.R2.fastq --failed_out 3_qc/BW1/B09W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B09W_merged.fastq -w 6 -h ./fastp_out/B09W_fastp.html -j ./fastp_out/B09W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 44957\n",
      "total bases: 6802461\n",
      "Q20 bases: 6724492(98.8538%)\n",
      "Q30 bases: 6557060(96.3925%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 44957\n",
      "total bases: 6829936\n",
      "Q20 bases: 6626005(97.0142%)\n",
      "Q30 bases: 6284955(92.0207%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 42914\n",
      "total bases: 4405639\n",
      "Q20 bases: 4377102(99.3523%)\n",
      "Q30 bases: 4286938(97.3057%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 85988\n",
      "reads failed due to low quality: 1384\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2542\n",
      "reads with adapter trimmed: 86286\n",
      "bases trimmed due to adapters: 1000528\n",
      "reads corrected by overlap analysis: 6699\n",
      "bases corrected by overlap analysis: 11424\n",
      "\n",
      "Duplication rate: 30.4878%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 42914\n",
      "% of original read pairs: 95.4557%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B10W_fastp.json\n",
      "HTML report: ./fastp_out/B10W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B10W.R1.fastq -I 2_demultiplexed/BW1/B10W.R2.fastq -o 3_qc/BW1/B10W.R1.fastq -O 3_qc/BW1/B10W.R2.fastq --unpaired1 3_qc/BW1/B10W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B10W_unpaired.R2.fastq --failed_out 3_qc/BW1/B10W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B10W_merged.fastq -w 6 -h ./fastp_out/B10W_fastp.html -j ./fastp_out/B10W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 23947\n",
      "total bases: 3614578\n",
      "Q20 bases: 3563574(98.5889%)\n",
      "Q30 bases: 3456001(95.6128%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 23947\n",
      "total bases: 3631240\n",
      "Q20 bases: 3502210(96.4467%)\n",
      "Q30 bases: 3288670(90.566%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 22768\n",
      "total bases: 2338473\n",
      "Q20 bases: 2321206(99.2616%)\n",
      "Q30 bases: 2264985(96.8574%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 45778\n",
      "reads failed due to low quality: 922\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1194\n",
      "reads with adapter trimmed: 46060\n",
      "bases trimmed due to adapters: 488449\n",
      "reads corrected by overlap analysis: 4038\n",
      "bases corrected by overlap analysis: 7112\n",
      "\n",
      "Duplication rate: 9.83607%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 22768\n",
      "% of original read pairs: 95.0766%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B11W_fastp.json\n",
      "HTML report: ./fastp_out/B11W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B11W.R1.fastq -I 2_demultiplexed/BW1/B11W.R2.fastq -o 3_qc/BW1/B11W.R1.fastq -O 3_qc/BW1/B11W.R2.fastq --unpaired1 3_qc/BW1/B11W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B11W_unpaired.R2.fastq --failed_out 3_qc/BW1/B11W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B11W_merged.fastq -w 6 -h ./fastp_out/B11W_fastp.html -j ./fastp_out/B11W_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 16547\n",
      "total bases: 1018952\n",
      "Q20 bases: 980761(96.2519%)\n",
      "Q30 bases: 929933(91.2637%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 16547\n",
      "total bases: 1008998\n",
      "Q20 bases: 972822(96.4147%)\n",
      "Q30 bases: 924988(91.6739%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 197\n",
      "total bases: 20094\n",
      "Q20 bases: 19961(99.3381%)\n",
      "Q30 bases: 19517(97.1285%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 398\n",
      "reads failed due to low quality: 2292\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 30404\n",
      "reads with adapter trimmed: 396\n",
      "bases trimmed due to adapters: 4526\n",
      "reads corrected by overlap analysis: 33\n",
      "bases corrected by overlap analysis: 52\n",
      "\n",
      "Duplication rate: 22.6415%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 197\n",
      "% of original read pairs: 1.19055%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B12W_fastp.json\n",
      "HTML report: ./fastp_out/B12W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B12W.R1.fastq -I 2_demultiplexed/BW1/B12W.R2.fastq -o 3_qc/BW1/B12W.R1.fastq -O 3_qc/BW1/B12W.R2.fastq --unpaired1 3_qc/BW1/B12W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B12W_unpaired.R2.fastq --failed_out 3_qc/BW1/B12W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B12W_merged.fastq -w 6 -h ./fastp_out/B12W_fastp.html -j ./fastp_out/B12W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 46913\n",
      "total bases: 7127952\n",
      "Q20 bases: 7016313(98.4338%)\n",
      "Q30 bases: 6816039(95.6241%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 46913\n",
      "total bases: 7154103\n",
      "Q20 bases: 6914304(96.6481%)\n",
      "Q30 bases: 6529293(91.2664%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 45160\n",
      "total bases: 4623907\n",
      "Q20 bases: 4590019(99.2671%)\n",
      "Q30 bases: 4482019(96.9314%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 90728\n",
      "reads failed due to low quality: 1924\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 1174\n",
      "reads with adapter trimmed: 91330\n",
      "bases trimmed due to adapters: 969897\n",
      "reads corrected by overlap analysis: 7334\n",
      "bases corrected by overlap analysis: 12578\n",
      "\n",
      "Duplication rate: 31.5217%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 45160\n",
      "% of original read pairs: 96.2633%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B13W_fastp.json\n",
      "HTML report: ./fastp_out/B13W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B13W.R1.fastq -I 2_demultiplexed/BW1/B13W.R2.fastq -o 3_qc/BW1/B13W.R1.fastq -O 3_qc/BW1/B13W.R2.fastq --unpaired1 3_qc/BW1/B13W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B13W_unpaired.R2.fastq --failed_out 3_qc/BW1/B13W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B13W_merged.fastq -w 6 -h ./fastp_out/B13W_fastp.html -j ./fastp_out/B13W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 88401\n",
      "total bases: 9047515\n",
      "Q20 bases: 8834115(97.6413%)\n",
      "Q30 bases: 8507962(94.0365%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 88401\n",
      "total bases: 9076891\n",
      "Q20 bases: 8766180(96.5769%)\n",
      "Q30 bases: 8308187(91.5312%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 37620\n",
      "total bases: 3848304\n",
      "Q20 bases: 3821485(99.3031%)\n",
      "Q30 bases: 3736638(97.0983%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 76502\n",
      "reads failed due to low quality: 9326\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 90972\n",
      "reads with adapter trimmed: 75760\n",
      "bases trimmed due to adapters: 960163\n",
      "reads corrected by overlap analysis: 5834\n",
      "bases corrected by overlap analysis: 9923\n",
      "\n",
      "Duplication rate: 19.4805%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 37620\n",
      "% of original read pairs: 42.5561%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B14W_fastp.json\n",
      "HTML report: ./fastp_out/B14W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B14W.R1.fastq -I 2_demultiplexed/BW1/B14W.R2.fastq -o 3_qc/BW1/B14W.R1.fastq -O 3_qc/BW1/B14W.R2.fastq --unpaired1 3_qc/BW1/B14W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B14W_unpaired.R2.fastq --failed_out 3_qc/BW1/B14W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B14W_merged.fastq -w 6 -h ./fastp_out/B14W_fastp.html -j ./fastp_out/B14W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 43870\n",
      "total bases: 3184723\n",
      "Q20 bases: 3076951(96.616%)\n",
      "Q30 bases: 2939590(92.3028%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 43870\n",
      "total bases: 3158302\n",
      "Q20 bases: 3052430(96.6478%)\n",
      "Q30 bases: 2908277(92.0836%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 5317\n",
      "total bases: 544344\n",
      "Q20 bases: 540242(99.2464%)\n",
      "Q30 bases: 527154(96.8421%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 10696\n",
      "reads failed due to low quality: 5388\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 71656\n",
      "reads with adapter trimmed: 10786\n",
      "bases trimmed due to adapters: 114861\n",
      "reads corrected by overlap analysis: 829\n",
      "bases corrected by overlap analysis: 1456\n",
      "\n",
      "Duplication rate: 28.2051%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 5317\n",
      "% of original read pairs: 12.1199%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B15W_fastp.json\n",
      "HTML report: ./fastp_out/B15W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B15W.R1.fastq -I 2_demultiplexed/BW1/B15W.R2.fastq -o 3_qc/BW1/B15W.R1.fastq -O 3_qc/BW1/B15W.R2.fastq --unpaired1 3_qc/BW1/B15W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B15W_unpaired.R2.fastq --failed_out 3_qc/BW1/B15W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B15W_merged.fastq -w 6 -h ./fastp_out/B15W_fastp.html -j ./fastp_out/B15W_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 62733\n",
      "total bases: 7846809\n",
      "Q20 bases: 7718789(98.3685%)\n",
      "Q30 bases: 7484394(95.3814%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 62733\n",
      "total bases: 7868041\n",
      "Q20 bases: 7617380(96.8142%)\n",
      "Q30 bases: 7214337(91.6917%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 41631\n",
      "total bases: 4263166\n",
      "Q20 bases: 4234080(99.3177%)\n",
      "Q30 bases: 4141676(97.1502%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 83594\n",
      "reads failed due to low quality: 4354\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 37518\n",
      "reads with adapter trimmed: 83778\n",
      "bases trimmed due to adapters: 1052684\n",
      "reads corrected by overlap analysis: 6489\n",
      "bases corrected by overlap analysis: 11239\n",
      "\n",
      "Duplication rate: 12.1212%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 41631\n",
      "% of original read pairs: 66.3622%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B16W_fastp.json\n",
      "HTML report: ./fastp_out/B16W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B16W.R1.fastq -I 2_demultiplexed/BW1/B16W.R2.fastq -o 3_qc/BW1/B16W.R1.fastq -O 3_qc/BW1/B16W.R2.fastq --unpaired1 3_qc/BW1/B16W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B16W_unpaired.R2.fastq --failed_out 3_qc/BW1/B16W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B16W_merged.fastq -w 6 -h ./fastp_out/B16W_fastp.html -j ./fastp_out/B16W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 165035\n",
      "total bases: 25161648\n",
      "Q20 bases: 24840865(98.7251%)\n",
      "Q30 bases: 24172587(96.0692%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 165035\n",
      "total bases: 25256975\n",
      "Q20 bases: 24451996(96.8128%)\n",
      "Q30 bases: 23109271(91.4966%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 158811\n",
      "total bases: 16275931\n",
      "Q20 bases: 16170352(99.3513%)\n",
      "Q30 bases: 15825595(97.2331%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 318634\n",
      "reads failed due to low quality: 5546\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 5888\n",
      "reads with adapter trimmed: 320418\n",
      "bases trimmed due to adapters: 3736840\n",
      "reads corrected by overlap analysis: 24617\n",
      "bases corrected by overlap analysis: 42684\n",
      "\n",
      "Duplication rate: 37.4194%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 158811\n",
      "% of original read pairs: 96.2287%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B17W_fastp.json\n",
      "HTML report: ./fastp_out/B17W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B17W.R1.fastq -I 2_demultiplexed/BW1/B17W.R2.fastq -o 3_qc/BW1/B17W.R1.fastq -O 3_qc/BW1/B17W.R2.fastq --unpaired1 3_qc/BW1/B17W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B17W_unpaired.R2.fastq --failed_out 3_qc/BW1/B17W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B17W_merged.fastq -w 6 -h ./fastp_out/B17W_fastp.html -j ./fastp_out/B17W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 164957\n",
      "total bases: 19767394\n",
      "Q20 bases: 19359501(97.9365%)\n",
      "Q30 bases: 18686842(94.5337%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 164957\n",
      "total bases: 19832664\n",
      "Q20 bases: 19042174(96.0142%)\n",
      "Q30 bases: 17872592(90.117%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 103811\n",
      "total bases: 10625784\n",
      "Q20 bases: 10543411(99.2248%)\n",
      "Q30 bases: 10286316(96.8052%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 208918\n",
      "reads failed due to low quality: 13714\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 107276\n",
      "reads with adapter trimmed: 210400\n",
      "bases trimmed due to adapters: 2259224\n",
      "reads corrected by overlap analysis: 18213\n",
      "bases corrected by overlap analysis: 33033\n",
      "\n",
      "Duplication rate: 16.5414%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 103811\n",
      "% of original read pairs: 62.9322%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B18W_fastp.json\n",
      "HTML report: ./fastp_out/B18W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B18W.R1.fastq -I 2_demultiplexed/BW1/B18W.R2.fastq -o 3_qc/BW1/B18W.R1.fastq -O 3_qc/BW1/B18W.R2.fastq --unpaired1 3_qc/BW1/B18W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B18W_unpaired.R2.fastq --failed_out 3_qc/BW1/B18W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B18W_merged.fastq -w 6 -h ./fastp_out/B18W_fastp.html -j ./fastp_out/B18W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 77552\n",
      "total bases: 9805180\n",
      "Q20 bases: 9606114(97.9698%)\n",
      "Q30 bases: 9302145(94.8697%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 77552\n",
      "total bases: 9738402\n",
      "Q20 bases: 9451296(97.0518%)\n",
      "Q30 bases: 8972858(92.1389%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 51791\n",
      "total bases: 5314314\n",
      "Q20 bases: 5279542(99.3457%)\n",
      "Q30 bases: 5167814(97.2433%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 104154\n",
      "reads failed due to low quality: 6458\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 44490\n",
      "reads with adapter trimmed: 104290\n",
      "bases trimmed due to adapters: 1299593\n",
      "reads corrected by overlap analysis: 8079\n",
      "bases corrected by overlap analysis: 13866\n",
      "\n",
      "Duplication rate: 17.5824%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 51791\n",
      "% of original read pairs: 66.7823%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B19W_fastp.json\n",
      "HTML report: ./fastp_out/B19W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B19W.R1.fastq -I 2_demultiplexed/BW1/B19W.R2.fastq -o 3_qc/BW1/B19W.R1.fastq -O 3_qc/BW1/B19W.R2.fastq --unpaired1 3_qc/BW1/B19W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B19W_unpaired.R2.fastq --failed_out 3_qc/BW1/B19W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B19W_merged.fastq -w 6 -h ./fastp_out/B19W_fastp.html -j ./fastp_out/B19W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 58163\n",
      "total bases: 8854928\n",
      "Q20 bases: 8769989(99.0408%)\n",
      "Q30 bases: 8567880(96.7583%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 58163\n",
      "total bases: 8888492\n",
      "Q20 bases: 8645142(97.2622%)\n",
      "Q30 bases: 8210425(92.3714%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 55989\n",
      "total bases: 5751089\n",
      "Q20 bases: 5718410(99.4318%)\n",
      "Q30 bases: 5609400(97.5363%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 112508\n",
      "reads failed due to low quality: 1304\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 2512\n",
      "reads with adapter trimmed: 112536\n",
      "bases trimmed due to adapters: 1294084\n",
      "reads corrected by overlap analysis: 8132\n",
      "bases corrected by overlap analysis: 14030\n",
      "\n",
      "Duplication rate: 29.2683%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 55989\n",
      "% of original read pairs: 96.2622%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B20W_fastp.json\n",
      "HTML report: ./fastp_out/B20W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B20W.R1.fastq -I 2_demultiplexed/BW1/B20W.R2.fastq -o 3_qc/BW1/B20W.R1.fastq -O 3_qc/BW1/B20W.R2.fastq --unpaired1 3_qc/BW1/B20W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B20W_unpaired.R2.fastq --failed_out 3_qc/BW1/B20W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B20W_merged.fastq -w 6 -h ./fastp_out/B20W_fastp.html -j ./fastp_out/B20W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 78078\n",
      "total bases: 11554288\n",
      "Q20 bases: 11366471(98.3745%)\n",
      "Q30 bases: 11019878(95.3748%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 78078\n",
      "total bases: 11551260\n",
      "Q20 bases: 11198552(96.9466%)\n",
      "Q30 bases: 10597873(91.7465%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 71797\n",
      "total bases: 7332157\n",
      "Q20 bases: 7274680(99.2161%)\n",
      "Q30 bases: 7094902(96.7642%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 144398\n",
      "reads failed due to low quality: 3864\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 7892\n",
      "reads with adapter trimmed: 145328\n",
      "bases trimmed due to adapters: 1536123\n",
      "reads corrected by overlap analysis: 11630\n",
      "bases corrected by overlap analysis: 19766\n",
      "\n",
      "Duplication rate: 20%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 71797\n",
      "% of original read pairs: 91.9555%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B21W_fastp.json\n",
      "HTML report: ./fastp_out/B21W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/B21W.R1.fastq -I 2_demultiplexed/BW1/B21W.R2.fastq -o 3_qc/BW1/B21W.R1.fastq -O 3_qc/BW1/B21W.R2.fastq --unpaired1 3_qc/BW1/B21W_unpaired.R1.fastq --unpaired2 3_qc/BW1/B21W_unpaired.R2.fastq --failed_out 3_qc/BW1/B21W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/B21W_merged.fastq -w 6 -h ./fastp_out/B21W_fastp.html -j ./fastp_out/B21W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 24844\n",
      "total bases: 2632881\n",
      "Q20 bases: 2565527(97.4418%)\n",
      "Q30 bases: 2469308(93.7873%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 24844\n",
      "total bases: 2619996\n",
      "Q20 bases: 2522167(96.2661%)\n",
      "Q30 bases: 2379641(90.8261%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 12469\n",
      "total bases: 1284219\n",
      "Q20 bases: 1275397(99.313%)\n",
      "Q30 bases: 1246399(97.055%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 24972\n",
      "reads failed due to low quality: 3378\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 21336\n",
      "reads with adapter trimmed: 25088\n",
      "bases trimmed due to adapters: 289421\n",
      "reads corrected by overlap analysis: 2140\n",
      "bases corrected by overlap analysis: 3674\n",
      "\n",
      "Duplication rate: 42.2222%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 12469\n",
      "% of original read pairs: 50.1892%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/BF3W_fastp.json\n",
      "HTML report: ./fastp_out/BF3W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/BF3W.R1.fastq -I 2_demultiplexed/BW1/BF3W.R2.fastq -o 3_qc/BW1/BF3W.R1.fastq -O 3_qc/BW1/BF3W.R2.fastq --unpaired1 3_qc/BW1/BF3W_unpaired.R1.fastq --unpaired2 3_qc/BW1/BF3W_unpaired.R2.fastq --failed_out 3_qc/BW1/BF3W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/BF3W_merged.fastq -w 6 -h ./fastp_out/BF3W_fastp.html -j ./fastp_out/BF3W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 118345\n",
      "total bases: 8537033\n",
      "Q20 bases: 7878277(92.2835%)\n",
      "Q30 bases: 7298300(85.4899%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 118345\n",
      "total bases: 7910972\n",
      "Q20 bases: 7583351(95.8587%)\n",
      "Q30 bases: 7179913(90.7589%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 3967\n",
      "total bases: 410191\n",
      "Q20 bases: 407433(99.3276%)\n",
      "Q30 bases: 398411(97.1282%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 8254\n",
      "reads failed due to low quality: 29118\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 199318\n",
      "reads with adapter trimmed: 7978\n",
      "bases trimmed due to adapters: 91801\n",
      "reads corrected by overlap analysis: 992\n",
      "bases corrected by overlap analysis: 1724\n",
      "\n",
      "Duplication rate: 28.125%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 3967\n",
      "% of original read pairs: 3.35206%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/BN1W_fastp.json\n",
      "HTML report: ./fastp_out/BN1W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/BN1W.R1.fastq -I 2_demultiplexed/BW1/BN1W.R2.fastq -o 3_qc/BW1/BN1W.R1.fastq -O 3_qc/BW1/BN1W.R2.fastq --unpaired1 3_qc/BW1/BN1W_unpaired.R1.fastq --unpaired2 3_qc/BW1/BN1W_unpaired.R2.fastq --failed_out 3_qc/BW1/BN1W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/BN1W_merged.fastq -w 6 -h ./fastp_out/BN1W_fastp.html -j ./fastp_out/BN1W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 288232\n",
      "total bases: 44485148\n",
      "Q20 bases: 43991310(98.8899%)\n",
      "Q30 bases: 42873117(96.3762%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 288232\n",
      "total bases: 44744753\n",
      "Q20 bases: 43352033(96.8874%)\n",
      "Q30 bases: 40941497(91.5001%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 283814\n",
      "total bases: 28945506\n",
      "Q20 bases: 28748684(99.32%)\n",
      "Q30 bases: 28107262(97.1041%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 568550\n",
      "reads failed due to low quality: 7260\n",
      "reads failed due to too many N: 8\n",
      "reads failed due to too short: 646\n",
      "reads with adapter trimmed: 570478\n",
      "bases trimmed due to adapters: 7218638\n",
      "reads corrected by overlap analysis: 54047\n",
      "bases corrected by overlap analysis: 92384\n",
      "\n",
      "Duplication rate: 81.2995%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 283814\n",
      "% of original read pairs: 98.4672%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/BP1W_fastp.json\n",
      "HTML report: ./fastp_out/BP1W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW1/BP1W.R1.fastq -I 2_demultiplexed/BW1/BP1W.R2.fastq -o 3_qc/BW1/BP1W.R1.fastq -O 3_qc/BW1/BP1W.R2.fastq --unpaired1 3_qc/BW1/BP1W_unpaired.R1.fastq --unpaired2 3_qc/BW1/BP1W_unpaired.R2.fastq --failed_out 3_qc/BW1/BP1W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW1/BP1W_merged.fastq -w 6 -h ./fastp_out/BP1W_fastp.html -j ./fastp_out/BP1W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 92947\n",
      "total bases: 12892586\n",
      "Q20 bases: 12645801(98.0858%)\n",
      "Q30 bases: 12231290(94.8707%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 92947\n",
      "total bases: 12944933\n",
      "Q20 bases: 12418140(95.9305%)\n",
      "Q30 bases: 11634948(89.8803%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 75948\n",
      "total bases: 7773905\n",
      "Q20 bases: 7714074(99.2304%)\n",
      "Q30 bases: 7531346(96.8798%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 154362\n",
      "reads failed due to low quality: 7442\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 24088\n",
      "reads with adapter trimmed: 154028\n",
      "bases trimmed due to adapters: 1939528\n",
      "reads corrected by overlap analysis: 14420\n",
      "bases corrected by overlap analysis: 26929\n",
      "\n",
      "Duplication rate: 23.2044%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 75948\n",
      "% of original read pairs: 81.7111%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B22W_fastp.json\n",
      "HTML report: ./fastp_out/B22W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B22W.R1.fastq -I 2_demultiplexed/BW2/B22W.R2.fastq -o 3_qc/BW2/B22W.R1.fastq -O 3_qc/BW2/B22W.R2.fastq --unpaired1 3_qc/BW2/B22W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B22W_unpaired.R2.fastq --failed_out 3_qc/BW2/B22W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B22W_merged.fastq -w 6 -h ./fastp_out/B22W_fastp.html -j ./fastp_out/B22W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 39596\n",
      "total bases: 4606379\n",
      "Q20 bases: 4493268(97.5445%)\n",
      "Q30 bases: 4323294(93.8545%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 39596\n",
      "total bases: 4619634\n",
      "Q20 bases: 4391474(95.0611%)\n",
      "Q30 bases: 4095149(88.6466%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 21340\n",
      "total bases: 2199427\n",
      "Q20 bases: 2182532(99.2318%)\n",
      "Q30 bases: 2129821(96.8353%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 48496\n",
      "reads failed due to low quality: 5312\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 25382\n",
      "reads with adapter trimmed: 45730\n",
      "bases trimmed due to adapters: 527019\n",
      "reads corrected by overlap analysis: 4500\n",
      "bases corrected by overlap analysis: 9443\n",
      "\n",
      "Duplication rate: 21.6216%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 21340\n",
      "% of original read pairs: 53.8943%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B23W_fastp.json\n",
      "HTML report: ./fastp_out/B23W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B23W.R1.fastq -I 2_demultiplexed/BW2/B23W.R2.fastq -o 3_qc/BW2/B23W.R1.fastq -O 3_qc/BW2/B23W.R2.fastq --unpaired1 3_qc/BW2/B23W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B23W_unpaired.R2.fastq --failed_out 3_qc/BW2/B23W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B23W_merged.fastq -w 6 -h ./fastp_out/B23W_fastp.html -j ./fastp_out/B23W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 36445\n",
      "total bases: 4922470\n",
      "Q20 bases: 4758765(96.6743%)\n",
      "Q30 bases: 4558790(92.6118%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 36445\n",
      "total bases: 4954431\n",
      "Q20 bases: 4696023(94.7843%)\n",
      "Q30 bases: 4350079(87.8018%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 24395\n",
      "total bases: 2510102\n",
      "Q20 bases: 2489832(99.1925%)\n",
      "Q30 bases: 2427251(96.6993%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 54582\n",
      "reads failed due to low quality: 5178\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 13130\n",
      "reads with adapter trimmed: 51554\n",
      "bases trimmed due to adapters: 548812\n",
      "reads corrected by overlap analysis: 4936\n",
      "bases corrected by overlap analysis: 11075\n",
      "\n",
      "Duplication rate: 8.33333%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 24395\n",
      "% of original read pairs: 66.9365%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B24W_fastp.json\n",
      "HTML report: ./fastp_out/B24W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B24W.R1.fastq -I 2_demultiplexed/BW2/B24W.R2.fastq -o 3_qc/BW2/B24W.R1.fastq -O 3_qc/BW2/B24W.R2.fastq --unpaired1 3_qc/BW2/B24W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B24W_unpaired.R2.fastq --failed_out 3_qc/BW2/B24W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B24W_merged.fastq -w 6 -h ./fastp_out/B24W_fastp.html -j ./fastp_out/B24W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 44601\n",
      "total bases: 5268295\n",
      "Q20 bases: 5136076(97.4903%)\n",
      "Q30 bases: 4936294(93.6981%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 44601\n",
      "total bases: 5284991\n",
      "Q20 bases: 5025538(95.0908%)\n",
      "Q30 bases: 4666399(88.2953%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 25615\n",
      "total bases: 2632324\n",
      "Q20 bases: 2613011(99.2663%)\n",
      "Q30 bases: 2551320(96.9227%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 56004\n",
      "reads failed due to low quality: 6252\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 26942\n",
      "reads with adapter trimmed: 53932\n",
      "bases trimmed due to adapters: 679519\n",
      "reads corrected by overlap analysis: 5182\n",
      "bases corrected by overlap analysis: 9165\n",
      "\n",
      "Duplication rate: 15.7407%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 25615\n",
      "% of original read pairs: 57.4314%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B25W_fastp.json\n",
      "HTML report: ./fastp_out/B25W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B25W.R1.fastq -I 2_demultiplexed/BW2/B25W.R2.fastq -o 3_qc/BW2/B25W.R1.fastq -O 3_qc/BW2/B25W.R2.fastq --unpaired1 3_qc/BW2/B25W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B25W_unpaired.R2.fastq --failed_out 3_qc/BW2/B25W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B25W_merged.fastq -w 6 -h ./fastp_out/B25W_fastp.html -j ./fastp_out/B25W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 61126\n",
      "total bases: 7582636\n",
      "Q20 bases: 7388804(97.4437%)\n",
      "Q30 bases: 7111444(93.7859%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 61126\n",
      "total bases: 7536673\n",
      "Q20 bases: 7234693(95.9932%)\n",
      "Q30 bases: 6794980(90.1589%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 35847\n",
      "total bases: 3679592\n",
      "Q20 bases: 3652682(99.2687%)\n",
      "Q30 bases: 3567268(96.9474%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 81770\n",
      "reads failed due to low quality: 7280\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 33202\n",
      "reads with adapter trimmed: 75342\n",
      "bases trimmed due to adapters: 870814\n",
      "reads corrected by overlap analysis: 6674\n",
      "bases corrected by overlap analysis: 11869\n",
      "\n",
      "Duplication rate: 13.253%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 35847\n",
      "% of original read pairs: 58.6444%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B26W_fastp.json\n",
      "HTML report: ./fastp_out/B26W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B26W.R1.fastq -I 2_demultiplexed/BW2/B26W.R2.fastq -o 3_qc/BW2/B26W.R1.fastq -O 3_qc/BW2/B26W.R2.fastq --unpaired1 3_qc/BW2/B26W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B26W_unpaired.R2.fastq --failed_out 3_qc/BW2/B26W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B26W_merged.fastq -w 6 -h ./fastp_out/B26W_fastp.html -j ./fastp_out/B26W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 100189\n",
      "total bases: 14118403\n",
      "Q20 bases: 13909262(98.5187%)\n",
      "Q30 bases: 13512787(95.7104%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 100189\n",
      "total bases: 14175557\n",
      "Q20 bases: 13642140(96.2371%)\n",
      "Q30 bases: 12811892(90.3802%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 82606\n",
      "total bases: 8497034\n",
      "Q20 bases: 8441812(99.3501%)\n",
      "Q30 bases: 8263020(97.2459%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 172408\n",
      "reads failed due to low quality: 5680\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 22286\n",
      "reads with adapter trimmed: 170472\n",
      "bases trimmed due to adapters: 1792114\n",
      "reads corrected by overlap analysis: 13707\n",
      "bases corrected by overlap analysis: 24775\n",
      "\n",
      "Duplication rate: 28.9256%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 82606\n",
      "% of original read pairs: 82.4502%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B27W_fastp.json\n",
      "HTML report: ./fastp_out/B27W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B27W.R1.fastq -I 2_demultiplexed/BW2/B27W.R2.fastq -o 3_qc/BW2/B27W.R1.fastq -O 3_qc/BW2/B27W.R2.fastq --unpaired1 3_qc/BW2/B27W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B27W_unpaired.R2.fastq --failed_out 3_qc/BW2/B27W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B27W_merged.fastq -w 6 -h ./fastp_out/B27W_fastp.html -j ./fastp_out/B27W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 76351\n",
      "total bases: 10247623\n",
      "Q20 bases: 10053669(98.1073%)\n",
      "Q30 bases: 9744411(95.0895%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 76351\n",
      "total bases: 10227878\n",
      "Q20 bases: 9896366(96.7587%)\n",
      "Q30 bases: 9372878(91.6405%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 57429\n",
      "total bases: 5873742\n",
      "Q20 bases: 5833527(99.3153%)\n",
      "Q30 bases: 5707567(97.1709%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 118512\n",
      "reads failed due to low quality: 5742\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 28448\n",
      "reads with adapter trimmed: 117214\n",
      "bases trimmed due to adapters: 1483318\n",
      "reads corrected by overlap analysis: 9075\n",
      "bases corrected by overlap analysis: 15867\n",
      "\n",
      "Duplication rate: 32.7103%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 57429\n",
      "% of original read pairs: 75.2171%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B28W_fastp.json\n",
      "HTML report: ./fastp_out/B28W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B28W.R1.fastq -I 2_demultiplexed/BW2/B28W.R2.fastq -o 3_qc/BW2/B28W.R1.fastq -O 3_qc/BW2/B28W.R2.fastq --unpaired1 3_qc/BW2/B28W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B28W_unpaired.R2.fastq --failed_out 3_qc/BW2/B28W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B28W_merged.fastq -w 6 -h ./fastp_out/B28W_fastp.html -j ./fastp_out/B28W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 123374\n",
      "total bases: 17448514\n",
      "Q20 bases: 17178887(98.4547%)\n",
      "Q30 bases: 16671623(95.5475%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 123374\n",
      "total bases: 17522242\n",
      "Q20 bases: 16842342(96.1198%)\n",
      "Q30 bases: 15784105(90.0804%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 106791\n",
      "total bases: 10910604\n",
      "Q20 bases: 10832476(99.2839%)\n",
      "Q30 bases: 10580358(96.9732%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 216348\n",
      "reads failed due to low quality: 6678\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 23722\n",
      "reads with adapter trimmed: 216224\n",
      "bases trimmed due to adapters: 2302595\n",
      "reads corrected by overlap analysis: 17920\n",
      "bases corrected by overlap analysis: 31384\n",
      "\n",
      "Duplication rate: 45.3125%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 106791\n",
      "% of original read pairs: 86.5588%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B29W_fastp.json\n",
      "HTML report: ./fastp_out/B29W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B29W.R1.fastq -I 2_demultiplexed/BW2/B29W.R2.fastq -o 3_qc/BW2/B29W.R1.fastq -O 3_qc/BW2/B29W.R2.fastq --unpaired1 3_qc/BW2/B29W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B29W_unpaired.R2.fastq --failed_out 3_qc/BW2/B29W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B29W_merged.fastq -w 6 -h ./fastp_out/B29W_fastp.html -j ./fastp_out/B29W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 112045\n",
      "total bases: 17140773\n",
      "Q20 bases: 16936349(98.8074%)\n",
      "Q30 bases: 16499441(96.2584%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 112045\n",
      "total bases: 17220652\n",
      "Q20 bases: 16637969(96.6164%)\n",
      "Q30 bases: 15684225(91.078%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 108066\n",
      "total bases: 11051942\n",
      "Q20 bases: 10981936(99.3666%)\n",
      "Q30 bases: 10753614(97.3007%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 216606\n",
      "reads failed due to low quality: 3444\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 4038\n",
      "reads with adapter trimmed: 217330\n",
      "bases trimmed due to adapters: 2735108\n",
      "reads corrected by overlap analysis: 17410\n",
      "bases corrected by overlap analysis: 30599\n",
      "\n",
      "Duplication rate: 73.5632%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 108066\n",
      "% of original read pairs: 96.4487%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B30W_fastp.json\n",
      "HTML report: ./fastp_out/B30W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B30W.R1.fastq -I 2_demultiplexed/BW2/B30W.R2.fastq -o 3_qc/BW2/B30W.R1.fastq -O 3_qc/BW2/B30W.R2.fastq --unpaired1 3_qc/BW2/B30W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B30W_unpaired.R2.fastq --failed_out 3_qc/BW2/B30W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B30W_merged.fastq -w 6 -h ./fastp_out/B30W_fastp.html -j ./fastp_out/B30W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 72883\n",
      "total bases: 8590428\n",
      "Q20 bases: 8438295(98.229%)\n",
      "Q30 bases: 8178861(95.209%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 72883\n",
      "total bases: 8634100\n",
      "Q20 bases: 8300922(96.1411%)\n",
      "Q30 bases: 7817757(90.5451%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 47182\n",
      "total bases: 4849896\n",
      "Q20 bases: 4817264(99.3272%)\n",
      "Q30 bases: 4714904(97.2166%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 96298\n",
      "reads failed due to low quality: 6668\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 42800\n",
      "reads with adapter trimmed: 95648\n",
      "bases trimmed due to adapters: 1115322\n",
      "reads corrected by overlap analysis: 7878\n",
      "bases corrected by overlap analysis: 13815\n",
      "\n",
      "Duplication rate: 21.875%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 47182\n",
      "% of original read pairs: 64.7366%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B31W_fastp.json\n",
      "HTML report: ./fastp_out/B31W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B31W.R1.fastq -I 2_demultiplexed/BW2/B31W.R2.fastq -o 3_qc/BW2/B31W.R1.fastq -O 3_qc/BW2/B31W.R2.fastq --unpaired1 3_qc/BW2/B31W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B31W_unpaired.R2.fastq --failed_out 3_qc/BW2/B31W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B31W_merged.fastq -w 6 -h ./fastp_out/B31W_fastp.html -j ./fastp_out/B31W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 84825\n",
      "total bases: 11791567\n",
      "Q20 bases: 11568165(98.1054%)\n",
      "Q30 bases: 11147210(94.5354%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 84825\n",
      "total bases: 11849145\n",
      "Q20 bases: 11353218(95.8147%)\n",
      "Q30 bases: 10570227(89.2067%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 71716\n",
      "total bases: 7332420\n",
      "Q20 bases: 7268173(99.1238%)\n",
      "Q30 bases: 7065845(96.3644%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 144902\n",
      "reads failed due to low quality: 5468\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 19280\n",
      "reads with adapter trimmed: 145620\n",
      "bases trimmed due to adapters: 1566932\n",
      "reads corrected by overlap analysis: 13738\n",
      "bases corrected by overlap analysis: 23758\n",
      "\n",
      "Duplication rate: 44.4444%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 71716\n",
      "% of original read pairs: 84.5458%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B32W_fastp.json\n",
      "HTML report: ./fastp_out/B32W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B32W.R1.fastq -I 2_demultiplexed/BW2/B32W.R2.fastq -o 3_qc/BW2/B32W.R1.fastq -O 3_qc/BW2/B32W.R2.fastq --unpaired1 3_qc/BW2/B32W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B32W_unpaired.R2.fastq --failed_out 3_qc/BW2/B32W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B32W_merged.fastq -w 6 -h ./fastp_out/B32W_fastp.html -j ./fastp_out/B32W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 130525\n",
      "total bases: 14457492\n",
      "Q20 bases: 14161289(97.9512%)\n",
      "Q30 bases: 13676960(94.6012%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 130525\n",
      "total bases: 14527756\n",
      "Q20 bases: 13914924(95.7816%)\n",
      "Q30 bases: 13044438(89.7898%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 69329\n",
      "total bases: 7080503\n",
      "Q20 bases: 7032600(99.3235%)\n",
      "Q30 bases: 6876204(97.1146%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 140604\n",
      "reads failed due to low quality: 13260\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 107186\n",
      "reads with adapter trimmed: 140240\n",
      "bases trimmed due to adapters: 1644043\n",
      "reads corrected by overlap analysis: 12565\n",
      "bases corrected by overlap analysis: 22920\n",
      "\n",
      "Duplication rate: 19.1489%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 69329\n",
      "% of original read pairs: 53.1155%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B33W_fastp.json\n",
      "HTML report: ./fastp_out/B33W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B33W.R1.fastq -I 2_demultiplexed/BW2/B33W.R2.fastq -o 3_qc/BW2/B33W.R1.fastq -O 3_qc/BW2/B33W.R2.fastq --unpaired1 3_qc/BW2/B33W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B33W_unpaired.R2.fastq --failed_out 3_qc/BW2/B33W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B33W_merged.fastq -w 6 -h ./fastp_out/B33W_fastp.html -j ./fastp_out/B33W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 79701\n",
      "total bases: 11344445\n",
      "Q20 bases: 11162998(98.4006%)\n",
      "Q30 bases: 10836782(95.525%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 79701\n",
      "total bases: 11335762\n",
      "Q20 bases: 10972404(96.7946%)\n",
      "Q30 bases: 10370389(91.4838%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 68697\n",
      "total bases: 7042754\n",
      "Q20 bases: 6993479(99.3003%)\n",
      "Q30 bases: 6836491(97.0713%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 140290\n",
      "reads failed due to low quality: 4628\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 14484\n",
      "reads with adapter trimmed: 139464\n",
      "bases trimmed due to adapters: 1475125\n",
      "reads corrected by overlap analysis: 10333\n",
      "bases corrected by overlap analysis: 17408\n",
      "\n",
      "Duplication rate: 30.3571%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 68697\n",
      "% of original read pairs: 86.1934%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B34W_fastp.json\n",
      "HTML report: ./fastp_out/B34W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B34W.R1.fastq -I 2_demultiplexed/BW2/B34W.R2.fastq -o 3_qc/BW2/B34W.R1.fastq -O 3_qc/BW2/B34W.R2.fastq --unpaired1 3_qc/BW2/B34W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B34W_unpaired.R2.fastq --failed_out 3_qc/BW2/B34W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B34W_merged.fastq -w 6 -h ./fastp_out/B34W_fastp.html -j ./fastp_out/B34W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 180700\n",
      "total bases: 24397230\n",
      "Q20 bases: 23922890(98.0558%)\n",
      "Q30 bases: 23150197(94.8886%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 180700\n",
      "total bases: 24440603\n",
      "Q20 bases: 23571306(96.4432%)\n",
      "Q30 bases: 22237948(90.9877%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 136559\n",
      "total bases: 14000925\n",
      "Q20 bases: 13903062(99.301%)\n",
      "Q30 bases: 13593591(97.0907%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 276848\n",
      "reads failed due to low quality: 14212\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 70334\n",
      "reads with adapter trimmed: 276636\n",
      "bases trimmed due to adapters: 3523914\n",
      "reads corrected by overlap analysis: 22106\n",
      "bases corrected by overlap analysis: 38760\n",
      "\n",
      "Duplication rate: 6.66667%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 136559\n",
      "% of original read pairs: 75.5722%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B35W_fastp.json\n",
      "HTML report: ./fastp_out/B35W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B35W.R1.fastq -I 2_demultiplexed/BW2/B35W.R2.fastq -o 3_qc/BW2/B35W.R1.fastq -O 3_qc/BW2/B35W.R2.fastq --unpaired1 3_qc/BW2/B35W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B35W_unpaired.R2.fastq --failed_out 3_qc/BW2/B35W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B35W_merged.fastq -w 6 -h ./fastp_out/B35W_fastp.html -j ./fastp_out/B35W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 61975\n",
      "total bases: 5317042\n",
      "Q20 bases: 5164815(97.137%)\n",
      "Q30 bases: 4958445(93.2557%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 61975\n",
      "total bases: 5272480\n",
      "Q20 bases: 5081416(96.3762%)\n",
      "Q30 bases: 4810362(91.2353%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 18691\n",
      "total bases: 1913847\n",
      "Q20 bases: 1900039(99.2785%)\n",
      "Q30 bases: 1856993(97.0293%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 37686\n",
      "reads failed due to low quality: 7210\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 79054\n",
      "reads with adapter trimmed: 37816\n",
      "bases trimmed due to adapters: 399972\n",
      "reads corrected by overlap analysis: 2786\n",
      "bases corrected by overlap analysis: 4808\n",
      "\n",
      "Duplication rate: 20.2128%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 18691\n",
      "% of original read pairs: 30.1589%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B36W_fastp.json\n",
      "HTML report: ./fastp_out/B36W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B36W.R1.fastq -I 2_demultiplexed/BW2/B36W.R2.fastq -o 3_qc/BW2/B36W.R1.fastq -O 3_qc/BW2/B36W.R2.fastq --unpaired1 3_qc/BW2/B36W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B36W_unpaired.R2.fastq --failed_out 3_qc/BW2/B36W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B36W_merged.fastq -w 6 -h ./fastp_out/B36W_fastp.html -j ./fastp_out/B36W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 216300\n",
      "total bases: 31397439\n",
      "Q20 bases: 30962651(98.6152%)\n",
      "Q30 bases: 30093684(95.8476%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 216300\n",
      "total bases: 31509599\n",
      "Q20 bases: 30437649(96.598%)\n",
      "Q30 bases: 28701308(91.0875%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 189949\n",
      "total bases: 19497030\n",
      "Q20 bases: 19363130(99.3132%)\n",
      "Q30 bases: 18935860(97.1218%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 382510\n",
      "reads failed due to low quality: 11340\n",
      "reads failed due to too many N: 8\n",
      "reads failed due to too short: 38742\n",
      "reads with adapter trimmed: 383388\n",
      "bases trimmed due to adapters: 4821312\n",
      "reads corrected by overlap analysis: 31001\n",
      "bases corrected by overlap analysis: 54424\n",
      "\n",
      "Duplication rate: 24.1935%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 189949\n",
      "% of original read pairs: 87.8174%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B37W_fastp.json\n",
      "HTML report: ./fastp_out/B37W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B37W.R1.fastq -I 2_demultiplexed/BW2/B37W.R2.fastq -o 3_qc/BW2/B37W.R1.fastq -O 3_qc/BW2/B37W.R2.fastq --unpaired1 3_qc/BW2/B37W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B37W_unpaired.R2.fastq --failed_out 3_qc/BW2/B37W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B37W_merged.fastq -w 6 -h ./fastp_out/B37W_fastp.html -j ./fastp_out/B37W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 353177\n",
      "total bases: 53901016\n",
      "Q20 bases: 53120916(98.5527%)\n",
      "Q30 bases: 51550208(95.6387%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 353177\n",
      "total bases: 54179863\n",
      "Q20 bases: 51979405(95.9386%)\n",
      "Q30 bases: 48544252(89.5983%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 340472\n",
      "total bases: 34771738\n",
      "Q20 bases: 34525843(99.2928%)\n",
      "Q30 bases: 33726288(96.9934%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 683930\n",
      "reads failed due to low quality: 14084\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 8340\n",
      "reads with adapter trimmed: 687052\n",
      "bases trimmed due to adapters: 8125891\n",
      "reads corrected by overlap analysis: 63567\n",
      "bases corrected by overlap analysis: 114239\n",
      "\n",
      "Duplication rate: 25.4777%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 340472\n",
      "% of original read pairs: 96.4027%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B38W_fastp.json\n",
      "HTML report: ./fastp_out/B38W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B38W.R1.fastq -I 2_demultiplexed/BW2/B38W.R2.fastq -o 3_qc/BW2/B38W.R1.fastq -O 3_qc/BW2/B38W.R2.fastq --unpaired1 3_qc/BW2/B38W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B38W_unpaired.R2.fastq --failed_out 3_qc/BW2/B38W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B38W_merged.fastq -w 6 -h ./fastp_out/B38W_fastp.html -j ./fastp_out/B38W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 265543\n",
      "total bases: 38175220\n",
      "Q20 bases: 37634651(98.584%)\n",
      "Q30 bases: 36580343(95.8222%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 265543\n",
      "total bases: 38300873\n",
      "Q20 bases: 36987811(96.5717%)\n",
      "Q30 bases: 34861301(91.0196%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 235022\n",
      "total bases: 24164392\n",
      "Q20 bases: 24000639(99.3223%)\n",
      "Q30 bases: 23475661(97.1498%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 470930\n",
      "reads failed due to low quality: 12702\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 47452\n",
      "reads with adapter trimmed: 473748\n",
      "bases trimmed due to adapters: 5056977\n",
      "reads corrected by overlap analysis: 36292\n",
      "bases corrected by overlap analysis: 64455\n",
      "\n",
      "Duplication rate: 19.0476%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 235022\n",
      "% of original read pairs: 88.5062%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B39W_fastp.json\n",
      "HTML report: ./fastp_out/B39W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B39W.R1.fastq -I 2_demultiplexed/BW2/B39W.R2.fastq -o 3_qc/BW2/B39W.R1.fastq -O 3_qc/BW2/B39W.R2.fastq --unpaired1 3_qc/BW2/B39W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B39W_unpaired.R2.fastq --failed_out 3_qc/BW2/B39W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B39W_merged.fastq -w 6 -h ./fastp_out/B39W_fastp.html -j ./fastp_out/B39W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 360224\n",
      "total bases: 53157882\n",
      "Q20 bases: 52349995(98.4802%)\n",
      "Q30 bases: 50852432(95.663%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 360224\n",
      "total bases: 53154034\n",
      "Q20 bases: 51376010(96.655%)\n",
      "Q30 bases: 48413467(91.0815%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 323261\n",
      "total bases: 33138966\n",
      "Q20 bases: 32908725(99.3052%)\n",
      "Q30 bases: 32167636(97.0689%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 649422\n",
      "reads failed due to low quality: 19618\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 51402\n",
      "reads with adapter trimmed: 651382\n",
      "bases trimmed due to adapters: 8124299\n",
      "reads corrected by overlap analysis: 56297\n",
      "bases corrected by overlap analysis: 99355\n",
      "\n",
      "Duplication rate: 18.3908%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 323261\n",
      "% of original read pairs: 89.7389%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B40W_fastp.json\n",
      "HTML report: ./fastp_out/B40W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/B40W.R1.fastq -I 2_demultiplexed/BW2/B40W.R2.fastq -o 3_qc/BW2/B40W.R1.fastq -O 3_qc/BW2/B40W.R2.fastq --unpaired1 3_qc/BW2/B40W_unpaired.R1.fastq --unpaired2 3_qc/BW2/B40W_unpaired.R2.fastq --failed_out 3_qc/BW2/B40W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/B40W_merged.fastq -w 6 -h ./fastp_out/B40W_fastp.html -j ./fastp_out/B40W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 17434\n",
      "total bases: 736778\n",
      "Q20 bases: 687862(93.3608%)\n",
      "Q30 bases: 637111(86.4726%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 17434\n",
      "total bases: 739508\n",
      "Q20 bases: 675532(91.3488%)\n",
      "Q30 bases: 615697(83.2577%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 116\n",
      "total bases: 12024\n",
      "Q20 bases: 11933(99.2432%)\n",
      "Q30 bases: 11598(96.4571%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 232\n",
      "reads failed due to low quality: 4338\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 30298\n",
      "reads with adapter trimmed: 236\n",
      "bases trimmed due to adapters: 2867\n",
      "reads corrected by overlap analysis: 28\n",
      "bases corrected by overlap analysis: 53\n",
      "\n",
      "Duplication rate: 35.5263%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 116\n",
      "% of original read pairs: 0.665367%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/BF2W_fastp.json\n",
      "HTML report: ./fastp_out/BF2W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/BF2W.R1.fastq -I 2_demultiplexed/BW2/BF2W.R2.fastq -o 3_qc/BW2/BF2W.R1.fastq -O 3_qc/BW2/BF2W.R2.fastq --unpaired1 3_qc/BW2/BF2W_unpaired.R1.fastq --unpaired2 3_qc/BW2/BF2W_unpaired.R2.fastq --failed_out 3_qc/BW2/BF2W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/BF2W_merged.fastq -w 6 -h ./fastp_out/BF2W_fastp.html -j ./fastp_out/BF2W_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 54827\n",
      "total bases: 3393719\n",
      "Q20 bases: 3123029(92.0238%)\n",
      "Q30 bases: 2885007(85.0102%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 54827\n",
      "total bases: 3152969\n",
      "Q20 bases: 2993554(94.944%)\n",
      "Q30 bases: 2813306(89.2272%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 0\n",
      "total bases: 0\n",
      "Q20 bases: 0(-nan%)\n",
      "Q30 bases: 0(-nan%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 64\n",
      "reads failed due to low quality: 13108\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 96482\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "reads corrected by overlap analysis: 169\n",
      "bases corrected by overlap analysis: 288\n",
      "\n",
      "Duplication rate: 24.2105%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 204\n",
      "\n",
      "Read pairs merged: 0\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/BN2W_fastp.json\n",
      "HTML report: ./fastp_out/BN2W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/BN2W.R1.fastq -I 2_demultiplexed/BW2/BN2W.R2.fastq -o 3_qc/BW2/BN2W.R1.fastq -O 3_qc/BW2/BN2W.R2.fastq --unpaired1 3_qc/BW2/BN2W_unpaired.R1.fastq --unpaired2 3_qc/BW2/BN2W_unpaired.R2.fastq --failed_out 3_qc/BW2/BN2W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/BN2W_merged.fastq -w 6 -h ./fastp_out/BN2W_fastp.html -j ./fastp_out/BN2W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 105088\n",
      "total bases: 16091287\n",
      "Q20 bases: 15891371(98.7576%)\n",
      "Q30 bases: 15453574(96.0369%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 105088\n",
      "total bases: 16205117\n",
      "Q20 bases: 15598231(96.255%)\n",
      "Q30 bases: 14627877(90.267%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 101926\n",
      "total bases: 10395156\n",
      "Q20 bases: 10319006(99.2674%)\n",
      "Q30 bases: 10074534(96.9157%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 204286\n",
      "reads failed due to low quality: 3656\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 2230\n",
      "reads with adapter trimmed: 204880\n",
      "bases trimmed due to adapters: 2614851\n",
      "reads corrected by overlap analysis: 21498\n",
      "bases corrected by overlap analysis: 38469\n",
      "\n",
      "Duplication rate: 77.0349%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 101926\n",
      "% of original read pairs: 96.9911%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/BP2W_fastp.json\n",
      "HTML report: ./fastp_out/BP2W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW2/BP2W.R1.fastq -I 2_demultiplexed/BW2/BP2W.R2.fastq -o 3_qc/BW2/BP2W.R1.fastq -O 3_qc/BW2/BP2W.R2.fastq --unpaired1 3_qc/BW2/BP2W_unpaired.R1.fastq --unpaired2 3_qc/BW2/BP2W_unpaired.R2.fastq --failed_out 3_qc/BW2/BP2W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW2/BP2W_merged.fastq -w 6 -h ./fastp_out/BP2W_fastp.html -j ./fastp_out/BP2W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 86204\n",
      "total bases: 11662137\n",
      "Q20 bases: 11418783(97.9133%)\n",
      "Q30 bases: 11014989(94.4509%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 86204\n",
      "total bases: 11663877\n",
      "Q20 bases: 11234839(96.3217%)\n",
      "Q30 bases: 10572746(90.6452%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 67559\n",
      "total bases: 6935736\n",
      "Q20 bases: 6877655(99.1626%)\n",
      "Q30 bases: 6701451(96.6221%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 136940\n",
      "reads failed due to low quality: 7614\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 27848\n",
      "reads with adapter trimmed: 137596\n",
      "bases trimmed due to adapters: 1714012\n",
      "reads corrected by overlap analysis: 12265\n",
      "bases corrected by overlap analysis: 21403\n",
      "\n",
      "Duplication rate: 25.8373%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 67559\n",
      "% of original read pairs: 78.3711%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B41W_fastp.json\n",
      "HTML report: ./fastp_out/B41W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B41W.R1.fastq -I 2_demultiplexed/BW3/B41W.R2.fastq -o 3_qc/BW3/B41W.R1.fastq -O 3_qc/BW3/B41W.R2.fastq --unpaired1 3_qc/BW3/B41W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B41W_unpaired.R2.fastq --failed_out 3_qc/BW3/B41W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B41W_merged.fastq -w 6 -h ./fastp_out/B41W_fastp.html -j ./fastp_out/B41W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 76607\n",
      "total bases: 10311445\n",
      "Q20 bases: 10133016(98.2696%)\n",
      "Q30 bases: 9804713(95.0857%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 76607\n",
      "total bases: 10325576\n",
      "Q20 bases: 9994747(96.796%)\n",
      "Q30 bases: 9455630(91.5748%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 59777\n",
      "total bases: 6160172\n",
      "Q20 bases: 6109117(99.1712%)\n",
      "Q30 bases: 5953605(96.6467%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 122618\n",
      "reads failed due to low quality: 5392\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 25204\n",
      "reads with adapter trimmed: 120736\n",
      "bases trimmed due to adapters: 1399880\n",
      "reads corrected by overlap analysis: 10922\n",
      "bases corrected by overlap analysis: 18962\n",
      "\n",
      "Duplication rate: 18.9474%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 59777\n",
      "% of original read pairs: 78.0307%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B42W_fastp.json\n",
      "HTML report: ./fastp_out/B42W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B42W.R1.fastq -I 2_demultiplexed/BW3/B42W.R2.fastq -o 3_qc/BW3/B42W.R1.fastq -O 3_qc/BW3/B42W.R2.fastq --unpaired1 3_qc/BW3/B42W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B42W_unpaired.R2.fastq --failed_out 3_qc/BW3/B42W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B42W_merged.fastq -w 6 -h ./fastp_out/B42W_fastp.html -j ./fastp_out/B42W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 41678\n",
      "total bases: 5605745\n",
      "Q20 bases: 5507037(98.2392%)\n",
      "Q30 bases: 5321715(94.9332%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 41678\n",
      "total bases: 5635763\n",
      "Q20 bases: 5428423(96.321%)\n",
      "Q30 bases: 5107017(90.618%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 32337\n",
      "total bases: 3325515\n",
      "Q20 bases: 3299275(99.2109%)\n",
      "Q30 bases: 3216916(96.7344%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 64988\n",
      "reads failed due to low quality: 3418\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 14950\n",
      "reads with adapter trimmed: 65338\n",
      "bases trimmed due to adapters: 691111\n",
      "reads corrected by overlap analysis: 5320\n",
      "bases corrected by overlap analysis: 9086\n",
      "\n",
      "Duplication rate: 32.6087%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 32337\n",
      "% of original read pairs: 77.5877%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B43W_fastp.json\n",
      "HTML report: ./fastp_out/B43W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B43W.R1.fastq -I 2_demultiplexed/BW3/B43W.R2.fastq -o 3_qc/BW3/B43W.R1.fastq -O 3_qc/BW3/B43W.R2.fastq --unpaired1 3_qc/BW3/B43W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B43W_unpaired.R2.fastq --failed_out 3_qc/BW3/B43W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B43W_merged.fastq -w 6 -h ./fastp_out/B43W_fastp.html -j ./fastp_out/B43W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 110294\n",
      "total bases: 16503492\n",
      "Q20 bases: 16234144(98.3679%)\n",
      "Q30 bases: 15717515(95.2375%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 110294\n",
      "total bases: 16559615\n",
      "Q20 bases: 16001374(96.6289%)\n",
      "Q30 bases: 15086026(91.1013%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 100610\n",
      "total bases: 10324718\n",
      "Q20 bases: 10245845(99.2361%)\n",
      "Q30 bases: 9999367(96.8488%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 203794\n",
      "reads failed due to low quality: 5714\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 11080\n",
      "reads with adapter trimmed: 204502\n",
      "bases trimmed due to adapters: 2592678\n",
      "reads corrected by overlap analysis: 16999\n",
      "bases corrected by overlap analysis: 29442\n",
      "\n",
      "Duplication rate: 31.1377%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 100610\n",
      "% of original read pairs: 91.2198%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B44W_fastp.json\n",
      "HTML report: ./fastp_out/B44W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B44W.R1.fastq -I 2_demultiplexed/BW3/B44W.R2.fastq -o 3_qc/BW3/B44W.R1.fastq -O 3_qc/BW3/B44W.R2.fastq --unpaired1 3_qc/BW3/B44W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B44W_unpaired.R2.fastq --failed_out 3_qc/BW3/B44W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B44W_merged.fastq -w 6 -h ./fastp_out/B44W_fastp.html -j ./fastp_out/B44W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 139820\n",
      "total bases: 20406694\n",
      "Q20 bases: 20072823(98.3639%)\n",
      "Q30 bases: 19460096(95.3613%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 139820\n",
      "total bases: 20423079\n",
      "Q20 bases: 19816149(97.0282%)\n",
      "Q30 bases: 18816171(92.1319%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 122761\n",
      "total bases: 12594394\n",
      "Q20 bases: 12495856(99.2176%)\n",
      "Q30 bases: 12194913(96.8281%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 248802\n",
      "reads failed due to low quality: 7344\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 23494\n",
      "reads with adapter trimmed: 248656\n",
      "bases trimmed due to adapters: 2895669\n",
      "reads corrected by overlap analysis: 19940\n",
      "bases corrected by overlap analysis: 33844\n",
      "\n",
      "Duplication rate: 36.3636%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 122761\n",
      "% of original read pairs: 87.7993%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B45W_fastp.json\n",
      "HTML report: ./fastp_out/B45W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B45W.R1.fastq -I 2_demultiplexed/BW3/B45W.R2.fastq -o 3_qc/BW3/B45W.R1.fastq -O 3_qc/BW3/B45W.R2.fastq --unpaired1 3_qc/BW3/B45W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B45W_unpaired.R2.fastq --failed_out 3_qc/BW3/B45W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B45W_merged.fastq -w 6 -h ./fastp_out/B45W_fastp.html -j ./fastp_out/B45W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 185603\n",
      "total bases: 28145042\n",
      "Q20 bases: 27789232(98.7358%)\n",
      "Q30 bases: 27051055(96.113%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 185603\n",
      "total bases: 28224047\n",
      "Q20 bases: 27355604(96.923%)\n",
      "Q30 bases: 25920823(91.8395%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 178523\n",
      "total bases: 18355274\n",
      "Q20 bases: 18231033(99.3231%)\n",
      "Q30 bases: 17835172(97.1665%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 358882\n",
      "reads failed due to low quality: 6160\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 6164\n",
      "reads with adapter trimmed: 360646\n",
      "bases trimmed due to adapters: 3809298\n",
      "reads corrected by overlap analysis: 25836\n",
      "bases corrected by overlap analysis: 45446\n",
      "\n",
      "Duplication rate: 24.8%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 178523\n",
      "% of original read pairs: 96.1854%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B46W_fastp.json\n",
      "HTML report: ./fastp_out/B46W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B46W.R1.fastq -I 2_demultiplexed/BW3/B46W.R2.fastq -o 3_qc/BW3/B46W.R1.fastq -O 3_qc/BW3/B46W.R2.fastq --unpaired1 3_qc/BW3/B46W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B46W_unpaired.R2.fastq --failed_out 3_qc/BW3/B46W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B46W_merged.fastq -w 6 -h ./fastp_out/B46W_fastp.html -j ./fastp_out/B46W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 125683\n",
      "total bases: 19196925\n",
      "Q20 bases: 18941761(98.6708%)\n",
      "Q30 bases: 18436020(96.0363%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 125683\n",
      "total bases: 19239095\n",
      "Q20 bases: 18735169(97.3807%)\n",
      "Q30 bases: 17860280(92.8333%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 120467\n",
      "total bases: 12321637\n",
      "Q20 bases: 12235364(99.2998%)\n",
      "Q30 bases: 11965720(97.1114%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 241984\n",
      "reads failed due to low quality: 4352\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 5026\n",
      "reads with adapter trimmed: 242744\n",
      "bases trimmed due to adapters: 3063123\n",
      "reads corrected by overlap analysis: 17081\n",
      "bases corrected by overlap analysis: 28641\n",
      "\n",
      "Duplication rate: 43.9189%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 120467\n",
      "% of original read pairs: 95.8499%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B47W_fastp.json\n",
      "HTML report: ./fastp_out/B47W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B47W.R1.fastq -I 2_demultiplexed/BW3/B47W.R2.fastq -o 3_qc/BW3/B47W.R1.fastq -O 3_qc/BW3/B47W.R2.fastq --unpaired1 3_qc/BW3/B47W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B47W_unpaired.R2.fastq --failed_out 3_qc/BW3/B47W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B47W_merged.fastq -w 6 -h ./fastp_out/B47W_fastp.html -j ./fastp_out/B47W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 183024\n",
      "total bases: 27862603\n",
      "Q20 bases: 27480030(98.6269%)\n",
      "Q30 bases: 26696715(95.8156%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 183024\n",
      "total bases: 27939452\n",
      "Q20 bases: 27098928(96.9916%)\n",
      "Q30 bases: 25668715(91.8727%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 177706\n",
      "total bases: 18230262\n",
      "Q20 bases: 18090957(99.2359%)\n",
      "Q30 bases: 17655824(96.849%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 356218\n",
      "reads failed due to low quality: 6124\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 3704\n",
      "reads with adapter trimmed: 358282\n",
      "bases trimmed due to adapters: 3788781\n",
      "reads corrected by overlap analysis: 27026\n",
      "bases corrected by overlap analysis: 45666\n",
      "\n",
      "Duplication rate: 37.9679%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 177706\n",
      "% of original read pairs: 97.0944%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B48W_fastp.json\n",
      "HTML report: ./fastp_out/B48W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B48W.R1.fastq -I 2_demultiplexed/BW3/B48W.R2.fastq -o 3_qc/BW3/B48W.R1.fastq -O 3_qc/BW3/B48W.R2.fastq --unpaired1 3_qc/BW3/B48W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B48W_unpaired.R2.fastq --failed_out 3_qc/BW3/B48W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B48W_merged.fastq -w 6 -h ./fastp_out/B48W_fastp.html -j ./fastp_out/B48W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 160982\n",
      "total bases: 24661807\n",
      "Q20 bases: 24347847(98.7269%)\n",
      "Q30 bases: 23688735(96.0543%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 160982\n",
      "total bases: 24740542\n",
      "Q20 bases: 24048753(97.2038%)\n",
      "Q30 bases: 22835458(92.2997%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 155376\n",
      "total bases: 15933612\n",
      "Q20 bases: 15821678(99.2975%)\n",
      "Q30 bases: 15469847(97.0894%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 311666\n",
      "reads failed due to low quality: 4822\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 5474\n",
      "reads with adapter trimmed: 312616\n",
      "bases trimmed due to adapters: 3916744\n",
      "reads corrected by overlap analysis: 23270\n",
      "bases corrected by overlap analysis: 38995\n",
      "\n",
      "Duplication rate: 33.5938%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 155376\n",
      "% of original read pairs: 96.5176%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B49W_fastp.json\n",
      "HTML report: ./fastp_out/B49W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B49W.R1.fastq -I 2_demultiplexed/BW3/B49W.R2.fastq -o 3_qc/BW3/B49W.R1.fastq -O 3_qc/BW3/B49W.R2.fastq --unpaired1 3_qc/BW3/B49W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B49W_unpaired.R2.fastq --failed_out 3_qc/BW3/B49W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B49W_merged.fastq -w 6 -h ./fastp_out/B49W_fastp.html -j ./fastp_out/B49W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 141360\n",
      "total bases: 21689656\n",
      "Q20 bases: 21385955(98.5998%)\n",
      "Q30 bases: 20774625(95.7813%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 141360\n",
      "total bases: 21746043\n",
      "Q20 bases: 21084362(96.9572%)\n",
      "Q30 bases: 19988320(91.917%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 137532\n",
      "total bases: 14078276\n",
      "Q20 bases: 13973197(99.2536%)\n",
      "Q30 bases: 13645826(96.9282%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 275614\n",
      "reads failed due to low quality: 5176\n",
      "reads failed due to too many N: 8\n",
      "reads failed due to too short: 1922\n",
      "reads with adapter trimmed: 277334\n",
      "bases trimmed due to adapters: 3235004\n",
      "reads corrected by overlap analysis: 21865\n",
      "bases corrected by overlap analysis: 37709\n",
      "\n",
      "Duplication rate: 42.4242%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 137532\n",
      "% of original read pairs: 97.292%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B50W_fastp.json\n",
      "HTML report: ./fastp_out/B50W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B50W.R1.fastq -I 2_demultiplexed/BW3/B50W.R2.fastq -o 3_qc/BW3/B50W.R1.fastq -O 3_qc/BW3/B50W.R2.fastq --unpaired1 3_qc/BW3/B50W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B50W_unpaired.R2.fastq --failed_out 3_qc/BW3/B50W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B50W_merged.fastq -w 6 -h ./fastp_out/B50W_fastp.html -j ./fastp_out/B50W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 156657\n",
      "total bases: 23958971\n",
      "Q20 bases: 23552790(98.3047%)\n",
      "Q30 bases: 22751844(94.9617%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 156657\n",
      "total bases: 24003791\n",
      "Q20 bases: 23223546(96.7495%)\n",
      "Q30 bases: 21885593(91.1756%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 152349\n",
      "total bases: 15614784\n",
      "Q20 bases: 15478205(99.1253%)\n",
      "Q30 bases: 15058179(96.4354%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 305440\n",
      "reads failed due to low quality: 6636\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 1236\n",
      "reads with adapter trimmed: 308226\n",
      "bases trimmed due to adapters: 3283269\n",
      "reads corrected by overlap analysis: 25434\n",
      "bases corrected by overlap analysis: 43253\n",
      "\n",
      "Duplication rate: 8.41121%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 152349\n",
      "% of original read pairs: 97.25%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B51W_fastp.json\n",
      "HTML report: ./fastp_out/B51W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B51W.R1.fastq -I 2_demultiplexed/BW3/B51W.R2.fastq -o 3_qc/BW3/B51W.R1.fastq -O 3_qc/BW3/B51W.R2.fastq --unpaired1 3_qc/BW3/B51W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B51W_unpaired.R2.fastq --failed_out 3_qc/BW3/B51W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B51W_merged.fastq -w 6 -h ./fastp_out/B51W_fastp.html -j ./fastp_out/B51W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 165948\n",
      "total bases: 21924907\n",
      "Q20 bases: 21558033(98.3267%)\n",
      "Q30 bases: 20894473(95.3002%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 165948\n",
      "total bases: 21943235\n",
      "Q20 bases: 21264831(96.9084%)\n",
      "Q30 bases: 20176787(91.9499%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 123982\n",
      "total bases: 12719470\n",
      "Q20 bases: 12626916(99.2723%)\n",
      "Q30 bases: 12338191(97.0024%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 248302\n",
      "reads failed due to low quality: 10674\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 72918\n",
      "reads with adapter trimmed: 249822\n",
      "bases trimmed due to adapters: 2898908\n",
      "reads corrected by overlap analysis: 19647\n",
      "bases corrected by overlap analysis: 33595\n",
      "\n",
      "Duplication rate: 15.0538%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 123982\n",
      "% of original read pairs: 74.7114%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B52W_fastp.json\n",
      "HTML report: ./fastp_out/B52W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B52W.R1.fastq -I 2_demultiplexed/BW3/B52W.R2.fastq -o 3_qc/BW3/B52W.R1.fastq -O 3_qc/BW3/B52W.R2.fastq --unpaired1 3_qc/BW3/B52W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B52W_unpaired.R2.fastq --failed_out 3_qc/BW3/B52W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B52W_merged.fastq -w 6 -h ./fastp_out/B52W_fastp.html -j ./fastp_out/B52W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 125442\n",
      "total bases: 19204457\n",
      "Q20 bases: 18921414(98.5262%)\n",
      "Q30 bases: 18351294(95.5575%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 125442\n",
      "total bases: 19249326\n",
      "Q20 bases: 18673793(97.0101%)\n",
      "Q30 bases: 17690919(91.9041%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 122715\n",
      "total bases: 12578557\n",
      "Q20 bases: 12473940(99.1683%)\n",
      "Q30 bases: 12156503(96.6447%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 245854\n",
      "reads failed due to low quality: 4614\n",
      "reads failed due to too many N: 8\n",
      "reads failed due to too short: 408\n",
      "reads with adapter trimmed: 247954\n",
      "bases trimmed due to adapters: 2632899\n",
      "reads corrected by overlap analysis: 18777\n",
      "bases corrected by overlap analysis: 31939\n",
      "\n",
      "Duplication rate: 36.1111%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 122715\n",
      "% of original read pairs: 97.8261%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B53W_fastp.json\n",
      "HTML report: ./fastp_out/B53W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B53W.R1.fastq -I 2_demultiplexed/BW3/B53W.R2.fastq -o 3_qc/BW3/B53W.R1.fastq -O 3_qc/BW3/B53W.R2.fastq --unpaired1 3_qc/BW3/B53W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B53W_unpaired.R2.fastq --failed_out 3_qc/BW3/B53W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B53W_merged.fastq -w 6 -h ./fastp_out/B53W_fastp.html -j ./fastp_out/B53W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 83362\n",
      "total bases: 11161423\n",
      "Q20 bases: 10951676(98.1208%)\n",
      "Q30 bases: 10583352(94.8208%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 83362\n",
      "total bases: 11190792\n",
      "Q20 bases: 10853693(96.9877%)\n",
      "Q30 bases: 10300191(92.0417%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 63117\n",
      "total bases: 6460929\n",
      "Q20 bases: 6409728(99.2075%)\n",
      "Q30 bases: 6251817(96.7634%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 126490\n",
      "reads failed due to low quality: 5522\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 34710\n",
      "reads with adapter trimmed: 127286\n",
      "bases trimmed due to adapters: 1619833\n",
      "reads corrected by overlap analysis: 9775\n",
      "bases corrected by overlap analysis: 16589\n",
      "\n",
      "Duplication rate: 24.1379%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 63117\n",
      "% of original read pairs: 75.7144%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B54W_fastp.json\n",
      "HTML report: ./fastp_out/B54W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B54W.R1.fastq -I 2_demultiplexed/BW3/B54W.R2.fastq -o 3_qc/BW3/B54W.R1.fastq -O 3_qc/BW3/B54W.R2.fastq --unpaired1 3_qc/BW3/B54W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B54W_unpaired.R2.fastq --failed_out 3_qc/BW3/B54W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B54W_merged.fastq -w 6 -h ./fastp_out/B54W_fastp.html -j ./fastp_out/B54W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 164210\n",
      "total bases: 25083280\n",
      "Q20 bases: 24695309(98.4533%)\n",
      "Q30 bases: 23935559(95.4244%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 164210\n",
      "total bases: 25111498\n",
      "Q20 bases: 24374221(97.064%)\n",
      "Q30 bases: 23114245(92.0465%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 159784\n",
      "total bases: 16369697\n",
      "Q20 bases: 16233015(99.165%)\n",
      "Q30 bases: 15816443(96.6203%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 320310\n",
      "reads failed due to low quality: 6328\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 1780\n",
      "reads with adapter trimmed: 322860\n",
      "bases trimmed due to adapters: 3419438\n",
      "reads corrected by overlap analysis: 24329\n",
      "bases corrected by overlap analysis: 40760\n",
      "\n",
      "Duplication rate: 40.4145%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 159784\n",
      "% of original read pairs: 97.3047%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/B55W_fastp.json\n",
      "HTML report: ./fastp_out/B55W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/B55W.R1.fastq -I 2_demultiplexed/BW3/B55W.R2.fastq -o 3_qc/BW3/B55W.R1.fastq -O 3_qc/BW3/B55W.R2.fastq --unpaired1 3_qc/BW3/B55W_unpaired.R1.fastq --unpaired2 3_qc/BW3/B55W_unpaired.R2.fastq --failed_out 3_qc/BW3/B55W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/B55W_merged.fastq -w 6 -h ./fastp_out/B55W_fastp.html -j ./fastp_out/B55W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 15568\n",
      "total bases: 1259231\n",
      "Q20 bases: 1206609(95.8211%)\n",
      "Q30 bases: 1143230(90.7879%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 15568\n",
      "total bases: 1230247\n",
      "Q20 bases: 1175612(95.559%)\n",
      "Q30 bases: 1104441(89.7739%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 4715\n",
      "total bases: 482944\n",
      "Q20 bases: 478692(99.1196%)\n",
      "Q30 bases: 465552(96.3988%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 9444\n",
      "reads failed due to low quality: 3044\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 18648\n",
      "reads with adapter trimmed: 9510\n",
      "bases trimmed due to adapters: 110957\n",
      "reads corrected by overlap analysis: 949\n",
      "bases corrected by overlap analysis: 1543\n",
      "\n",
      "Duplication rate: 15.942%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 4715\n",
      "% of original read pairs: 30.2865%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/BF1W_fastp.json\n",
      "HTML report: ./fastp_out/BF1W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/BF1W.R1.fastq -I 2_demultiplexed/BW3/BF1W.R2.fastq -o 3_qc/BW3/BF1W.R1.fastq -O 3_qc/BW3/BF1W.R2.fastq --unpaired1 3_qc/BW3/BF1W_unpaired.R1.fastq --unpaired2 3_qc/BW3/BF1W_unpaired.R2.fastq --failed_out 3_qc/BW3/BF1W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/BF1W_merged.fastq -w 6 -h ./fastp_out/BF1W_fastp.html -j ./fastp_out/BF1W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 7316\n",
      "total bases: 650166\n",
      "Q20 bases: 636782(97.9414%)\n",
      "Q30 bases: 614806(94.5614%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7316\n",
      "total bases: 653258\n",
      "Q20 bases: 630633(96.5366%)\n",
      "Q30 bases: 598389(91.6007%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 2936\n",
      "total bases: 302304\n",
      "Q20 bases: 300495(99.4016%)\n",
      "Q30 bases: 294269(97.3421%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5884\n",
      "reads failed due to low quality: 810\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 7938\n",
      "reads with adapter trimmed: 5894\n",
      "bases trimmed due to adapters: 67650\n",
      "reads corrected by overlap analysis: 463\n",
      "bases corrected by overlap analysis: 737\n",
      "\n",
      "Duplication rate: 5.12821%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 2936\n",
      "% of original read pairs: 40.1312%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/BN3W_fastp.json\n",
      "HTML report: ./fastp_out/BN3W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/BN3W.R1.fastq -I 2_demultiplexed/BW3/BN3W.R2.fastq -o 3_qc/BW3/BN3W.R1.fastq -O 3_qc/BW3/BN3W.R2.fastq --unpaired1 3_qc/BW3/BN3W_unpaired.R1.fastq --unpaired2 3_qc/BW3/BN3W_unpaired.R2.fastq --failed_out 3_qc/BW3/BN3W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/BN3W_merged.fastq -w 6 -h ./fastp_out/BN3W_fastp.html -j ./fastp_out/BN3W_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 116481\n",
      "total bases: 17970519\n",
      "Q20 bases: 17741955(98.7281%)\n",
      "Q30 bases: 17241025(95.9406%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 116481\n",
      "total bases: 18065339\n",
      "Q20 bases: 17523143(96.9987%)\n",
      "Q30 bases: 16585157(91.8065%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 114424\n",
      "total bases: 11669972\n",
      "Q20 bases: 11578820(99.2189%)\n",
      "Q30 bases: 11292708(96.7672%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 229242\n",
      "reads failed due to low quality: 3246\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 470\n",
      "reads with adapter trimmed: 230232\n",
      "bases trimmed due to adapters: 2914742\n",
      "reads corrected by overlap analysis: 21594\n",
      "bases corrected by overlap analysis: 36363\n",
      "\n",
      "Duplication rate: 68.6411%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 114424\n",
      "% of original read pairs: 98.234%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/BP3W_fastp.json\n",
      "HTML report: ./fastp_out/BP3W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/BW3/BP3W.R1.fastq -I 2_demultiplexed/BW3/BP3W.R2.fastq -o 3_qc/BW3/BP3W.R1.fastq -O 3_qc/BW3/BP3W.R2.fastq --unpaired1 3_qc/BW3/BP3W_unpaired.R1.fastq --unpaired2 3_qc/BW3/BP3W_unpaired.R2.fastq --failed_out 3_qc/BW3/BP3W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/BW3/BP3W_merged.fastq -w 6 -h ./fastp_out/BP3W_fastp.html -j ./fastp_out/BP3W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 175286\n",
      "total bases: 26928992\n",
      "Q20 bases: 26490117(98.3703%)\n",
      "Q30 bases: 25644606(95.2305%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 175286\n",
      "total bases: 26986895\n",
      "Q20 bases: 26238420(97.2265%)\n",
      "Q30 bases: 24958151(92.4825%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 168376\n",
      "total bases: 17269583\n",
      "Q20 bases: 17118980(99.1279%)\n",
      "Q30 bases: 16668439(96.5191%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 338680\n",
      "reads failed due to low quality: 7586\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 4300\n",
      "reads with adapter trimmed: 340952\n",
      "bases trimmed due to adapters: 4246490\n",
      "reads corrected by overlap analysis: 27610\n",
      "bases corrected by overlap analysis: 46854\n",
      "\n",
      "Duplication rate: 24.1107%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 168376\n",
      "% of original read pairs: 96.0579%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C01W_fastp.json\n",
      "HTML report: ./fastp_out/C01W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C01W.R1.fastq -I 2_demultiplexed/CW1/C01W.R2.fastq -o 3_qc/CW1/C01W.R1.fastq -O 3_qc/CW1/C01W.R2.fastq --unpaired1 3_qc/CW1/C01W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C01W_unpaired.R2.fastq --failed_out 3_qc/CW1/C01W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C01W_merged.fastq -w 6 -h ./fastp_out/C01W_fastp.html -j ./fastp_out/C01W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 204214\n",
      "total bases: 31397473\n",
      "Q20 bases: 30950264(98.5757%)\n",
      "Q30 bases: 30036335(95.6648%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 204214\n",
      "total bases: 31452217\n",
      "Q20 bases: 30612171(97.3291%)\n",
      "Q30 bases: 29156247(92.7001%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 198217\n",
      "total bases: 20367358\n",
      "Q20 bases: 20200063(99.1786%)\n",
      "Q30 bases: 19696324(96.7053%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 398336\n",
      "reads failed due to low quality: 7434\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2658\n",
      "reads with adapter trimmed: 401072\n",
      "bases trimmed due to adapters: 4627378\n",
      "reads corrected by overlap analysis: 31534\n",
      "bases corrected by overlap analysis: 53486\n",
      "\n",
      "Duplication rate: 40%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 198217\n",
      "% of original read pairs: 97.0634%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C02W_fastp.json\n",
      "HTML report: ./fastp_out/C02W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C02W.R1.fastq -I 2_demultiplexed/CW1/C02W.R2.fastq -o 3_qc/CW1/C02W.R1.fastq -O 3_qc/CW1/C02W.R2.fastq --unpaired1 3_qc/CW1/C02W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C02W_unpaired.R2.fastq --failed_out 3_qc/CW1/C02W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C02W_merged.fastq -w 6 -h ./fastp_out/C02W_fastp.html -j ./fastp_out/C02W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 84194\n",
      "total bases: 12755496\n",
      "Q20 bases: 12548886(98.3802%)\n",
      "Q30 bases: 12126854(95.0716%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 84194\n",
      "total bases: 12785135\n",
      "Q20 bases: 12412598(97.0862%)\n",
      "Q30 bases: 11766937(92.0361%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 80558\n",
      "total bases: 8256948\n",
      "Q20 bases: 8185178(99.1308%)\n",
      "Q30 bases: 7962976(96.4397%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 162222\n",
      "reads failed due to low quality: 3432\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2734\n",
      "reads with adapter trimmed: 162922\n",
      "bases trimmed due to adapters: 1727620\n",
      "reads corrected by overlap analysis: 12988\n",
      "bases corrected by overlap analysis: 21754\n",
      "\n",
      "Duplication rate: 53.3333%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 80558\n",
      "% of original read pairs: 95.6814%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C03W_fastp.json\n",
      "HTML report: ./fastp_out/C03W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C03W.R1.fastq -I 2_demultiplexed/CW1/C03W.R2.fastq -o 3_qc/CW1/C03W.R1.fastq -O 3_qc/CW1/C03W.R2.fastq --unpaired1 3_qc/CW1/C03W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C03W_unpaired.R2.fastq --failed_out 3_qc/CW1/C03W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C03W_merged.fastq -w 6 -h ./fastp_out/C03W_fastp.html -j ./fastp_out/C03W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 159895\n",
      "total bases: 24649907\n",
      "Q20 bases: 24260875(98.4218%)\n",
      "Q30 bases: 23471392(95.219%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 159895\n",
      "total bases: 24734333\n",
      "Q20 bases: 24021314(97.1173%)\n",
      "Q30 bases: 22795171(92.16%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 153910\n",
      "total bases: 15758908\n",
      "Q20 bases: 15625441(99.1531%)\n",
      "Q30 bases: 15216562(96.5585%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 310284\n",
      "reads failed due to low quality: 6340\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 3166\n",
      "reads with adapter trimmed: 312170\n",
      "bases trimmed due to adapters: 3959915\n",
      "reads corrected by overlap analysis: 25700\n",
      "bases corrected by overlap analysis: 44039\n",
      "\n",
      "Duplication rate: 49.0909%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 153910\n",
      "% of original read pairs: 96.2569%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C04W_fastp.json\n",
      "HTML report: ./fastp_out/C04W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C04W.R1.fastq -I 2_demultiplexed/CW1/C04W.R2.fastq -o 3_qc/CW1/C04W.R1.fastq -O 3_qc/CW1/C04W.R2.fastq --unpaired1 3_qc/CW1/C04W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C04W_unpaired.R2.fastq --failed_out 3_qc/CW1/C04W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C04W_merged.fastq -w 6 -h ./fastp_out/C04W_fastp.html -j ./fastp_out/C04W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 175560\n",
      "total bases: 26805235\n",
      "Q20 bases: 26378430(98.4078%)\n",
      "Q30 bases: 25538053(95.2726%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 175560\n",
      "total bases: 26850652\n",
      "Q20 bases: 26158803(97.4233%)\n",
      "Q30 bases: 24969466(92.9939%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 169147\n",
      "total bases: 17278384\n",
      "Q20 bases: 17125723(99.1165%)\n",
      "Q30 bases: 16664178(96.4452%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 339706\n",
      "reads failed due to low quality: 6812\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 4602\n",
      "reads with adapter trimmed: 341980\n",
      "bases trimmed due to adapters: 3985315\n",
      "reads corrected by overlap analysis: 27655\n",
      "bases corrected by overlap analysis: 46215\n",
      "\n",
      "Duplication rate: 68.3274%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 169147\n",
      "% of original read pairs: 96.3471%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C05W_fastp.json\n",
      "HTML report: ./fastp_out/C05W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C05W.R1.fastq -I 2_demultiplexed/CW1/C05W.R2.fastq -o 3_qc/CW1/C05W.R1.fastq -O 3_qc/CW1/C05W.R2.fastq --unpaired1 3_qc/CW1/C05W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C05W_unpaired.R2.fastq --failed_out 3_qc/CW1/C05W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C05W_merged.fastq -w 6 -h ./fastp_out/C05W_fastp.html -j ./fastp_out/C05W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 117464\n",
      "total bases: 17811847\n",
      "Q20 bases: 17536727(98.4554%)\n",
      "Q30 bases: 16992009(95.3972%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 117464\n",
      "total bases: 17854390\n",
      "Q20 bases: 17339995(97.1189%)\n",
      "Q30 bases: 16470148(92.247%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 113277\n",
      "total bases: 11592447\n",
      "Q20 bases: 11496608(99.1733%)\n",
      "Q30 bases: 11200232(96.6166%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 227370\n",
      "reads failed due to low quality: 4620\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2938\n",
      "reads with adapter trimmed: 228906\n",
      "bases trimmed due to adapters: 2419505\n",
      "reads corrected by overlap analysis: 17925\n",
      "bases corrected by overlap analysis: 30271\n",
      "\n",
      "Duplication rate: 46.8966%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 113277\n",
      "% of original read pairs: 96.4355%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C06W_fastp.json\n",
      "HTML report: ./fastp_out/C06W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C06W.R1.fastq -I 2_demultiplexed/CW1/C06W.R2.fastq -o 3_qc/CW1/C06W.R1.fastq -O 3_qc/CW1/C06W.R2.fastq --unpaired1 3_qc/CW1/C06W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C06W_unpaired.R2.fastq --failed_out 3_qc/CW1/C06W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C06W_merged.fastq -w 6 -h ./fastp_out/C06W_fastp.html -j ./fastp_out/C06W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 148252\n",
      "total bases: 22636131\n",
      "Q20 bases: 22316988(98.5901%)\n",
      "Q30 bases: 21670659(95.7348%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 148252\n",
      "total bases: 22677492\n",
      "Q20 bases: 22159523(97.7159%)\n",
      "Q30 bases: 21238582(93.6549%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 140696\n",
      "total bases: 14404431\n",
      "Q20 bases: 14287854(99.1907%)\n",
      "Q30 bases: 13935977(96.7478%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 284156\n",
      "reads failed due to low quality: 5284\n",
      "reads failed due to too many N: 10\n",
      "reads failed due to too short: 7054\n",
      "reads with adapter trimmed: 285750\n",
      "bases trimmed due to adapters: 3591349\n",
      "reads corrected by overlap analysis: 20710\n",
      "bases corrected by overlap analysis: 34557\n",
      "\n",
      "Duplication rate: 55.6098%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 140696\n",
      "% of original read pairs: 94.9033%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C07W_fastp.json\n",
      "HTML report: ./fastp_out/C07W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C07W.R1.fastq -I 2_demultiplexed/CW1/C07W.R2.fastq -o 3_qc/CW1/C07W.R1.fastq -O 3_qc/CW1/C07W.R2.fastq --unpaired1 3_qc/CW1/C07W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C07W_unpaired.R2.fastq --failed_out 3_qc/CW1/C07W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C07W_merged.fastq -w 6 -h ./fastp_out/C07W_fastp.html -j ./fastp_out/C07W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 192662\n",
      "total bases: 29407059\n",
      "Q20 bases: 28963366(98.4912%)\n",
      "Q30 bases: 28069830(95.4527%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 192662\n",
      "total bases: 29469836\n",
      "Q20 bases: 28635112(97.1675%)\n",
      "Q30 bases: 27198096(92.2913%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 187313\n",
      "total bases: 19200137\n",
      "Q20 bases: 19040830(99.1703%)\n",
      "Q30 bases: 18547948(96.6032%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 375540\n",
      "reads failed due to low quality: 7256\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 2524\n",
      "reads with adapter trimmed: 378638\n",
      "bases trimmed due to adapters: 4009256\n",
      "reads corrected by overlap analysis: 29448\n",
      "bases corrected by overlap analysis: 48908\n",
      "\n",
      "Duplication rate: 51.4523%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 187313\n",
      "% of original read pairs: 97.2236%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C08W_fastp.json\n",
      "HTML report: ./fastp_out/C08W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C08W.R1.fastq -I 2_demultiplexed/CW1/C08W.R2.fastq -o 3_qc/CW1/C08W.R1.fastq -O 3_qc/CW1/C08W.R2.fastq --unpaired1 3_qc/CW1/C08W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C08W_unpaired.R2.fastq --failed_out 3_qc/CW1/C08W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C08W_merged.fastq -w 6 -h ./fastp_out/C08W_fastp.html -j ./fastp_out/C08W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 162708\n",
      "total bases: 25158252\n",
      "Q20 bases: 24819018(98.6516%)\n",
      "Q30 bases: 24099192(95.7904%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 162708\n",
      "total bases: 25224275\n",
      "Q20 bases: 24591764(97.4925%)\n",
      "Q30 bases: 23449417(92.9637%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 159247\n",
      "total bases: 16328959\n",
      "Q20 bases: 16202256(99.2241%)\n",
      "Q30 bases: 15810568(96.8253%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 318958\n",
      "reads failed due to low quality: 4870\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 1586\n",
      "reads with adapter trimmed: 320916\n",
      "bases trimmed due to adapters: 4017676\n",
      "reads corrected by overlap analysis: 23729\n",
      "bases corrected by overlap analysis: 39578\n",
      "\n",
      "Duplication rate: 66.0944%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 159247\n",
      "% of original read pairs: 97.8729%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C09W_fastp.json\n",
      "HTML report: ./fastp_out/C09W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C09W.R1.fastq -I 2_demultiplexed/CW1/C09W.R2.fastq -o 3_qc/CW1/C09W.R1.fastq -O 3_qc/CW1/C09W.R2.fastq --unpaired1 3_qc/CW1/C09W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C09W_unpaired.R2.fastq --failed_out 3_qc/CW1/C09W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C09W_merged.fastq -w 6 -h ./fastp_out/C09W_fastp.html -j ./fastp_out/C09W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 230202\n",
      "total bases: 35336060\n",
      "Q20 bases: 34853118(98.6333%)\n",
      "Q30 bases: 33829684(95.737%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 230202\n",
      "total bases: 35429566\n",
      "Q20 bases: 34475403(97.3069%)\n",
      "Q30 bases: 32789814(92.5493%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 221860\n",
      "total bases: 22833558\n",
      "Q20 bases: 22640060(99.1526%)\n",
      "Q30 bases: 22062113(96.6214%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 450472\n",
      "reads failed due to low quality: 7444\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2488\n",
      "reads with adapter trimmed: 446926\n",
      "bases trimmed due to adapters: 5178150\n",
      "reads corrected by overlap analysis: 41287\n",
      "bases corrected by overlap analysis: 70663\n",
      "\n",
      "Duplication rate: 22.9008%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 221860\n",
      "% of original read pairs: 96.3762%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C10W_fastp.json\n",
      "HTML report: ./fastp_out/C10W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C10W.R1.fastq -I 2_demultiplexed/CW1/C10W.R2.fastq -o 3_qc/CW1/C10W.R1.fastq -O 3_qc/CW1/C10W.R2.fastq --unpaired1 3_qc/CW1/C10W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C10W_unpaired.R2.fastq --failed_out 3_qc/CW1/C10W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C10W_merged.fastq -w 6 -h ./fastp_out/C10W_fastp.html -j ./fastp_out/C10W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 111081\n",
      "total bases: 16698210\n",
      "Q20 bases: 16393335(98.1742%)\n",
      "Q30 bases: 15798295(94.6107%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 111081\n",
      "total bases: 16721068\n",
      "Q20 bases: 16205271(96.9153%)\n",
      "Q30 bases: 15327040(91.663%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 105290\n",
      "total bases: 10783885\n",
      "Q20 bases: 10682472(99.0596%)\n",
      "Q30 bases: 10373178(96.1915%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 211612\n",
      "reads failed due to low quality: 5300\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 5248\n",
      "reads with adapter trimmed: 213170\n",
      "bases trimmed due to adapters: 2271838\n",
      "reads corrected by overlap analysis: 18415\n",
      "bases corrected by overlap analysis: 31367\n",
      "\n",
      "Duplication rate: 45.4545%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 105290\n",
      "% of original read pairs: 94.7867%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C11W_fastp.json\n",
      "HTML report: ./fastp_out/C11W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C11W.R1.fastq -I 2_demultiplexed/CW1/C11W.R2.fastq -o 3_qc/CW1/C11W.R1.fastq -O 3_qc/CW1/C11W.R2.fastq --unpaired1 3_qc/CW1/C11W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C11W_unpaired.R2.fastq --failed_out 3_qc/CW1/C11W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C11W_merged.fastq -w 6 -h ./fastp_out/C11W_fastp.html -j ./fastp_out/C11W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 96543\n",
      "total bases: 11497579\n",
      "Q20 bases: 11248829(97.8365%)\n",
      "Q30 bases: 10838371(94.2665%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 96543\n",
      "total bases: 11475141\n",
      "Q20 bases: 11130046(96.9927%)\n",
      "Q30 bases: 10590431(92.2902%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 58661\n",
      "total bases: 6007640\n",
      "Q20 bases: 5957488(99.1652%)\n",
      "Q30 bases: 5803810(96.6072%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 117706\n",
      "reads failed due to low quality: 8616\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 66760\n",
      "reads with adapter trimmed: 118446\n",
      "bases trimmed due to adapters: 1371233\n",
      "reads corrected by overlap analysis: 9625\n",
      "bases corrected by overlap analysis: 16229\n",
      "\n",
      "Duplication rate: 10.3448%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 58661\n",
      "% of original read pairs: 60.7615%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C12W_fastp.json\n",
      "HTML report: ./fastp_out/C12W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C12W.R1.fastq -I 2_demultiplexed/CW1/C12W.R2.fastq -o 3_qc/CW1/C12W.R1.fastq -O 3_qc/CW1/C12W.R2.fastq --unpaired1 3_qc/CW1/C12W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C12W_unpaired.R2.fastq --failed_out 3_qc/CW1/C12W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C12W_merged.fastq -w 6 -h ./fastp_out/C12W_fastp.html -j ./fastp_out/C12W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 55865\n",
      "total bases: 8341897\n",
      "Q20 bases: 8202891(98.3336%)\n",
      "Q30 bases: 7930597(95.0695%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 55865\n",
      "total bases: 8345995\n",
      "Q20 bases: 8125922(97.3631%)\n",
      "Q30 bases: 7742424(92.7681%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 52125\n",
      "total bases: 5320664\n",
      "Q20 bases: 5273869(99.1205%)\n",
      "Q30 bases: 5128220(96.3831%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 105582\n",
      "reads failed due to low quality: 2664\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 3482\n",
      "reads with adapter trimmed: 105798\n",
      "bases trimmed due to adapters: 1116629\n",
      "reads corrected by overlap analysis: 8063\n",
      "bases corrected by overlap analysis: 13539\n",
      "\n",
      "Duplication rate: 38.2812%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 52125\n",
      "% of original read pairs: 93.3053%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C13W_fastp.json\n",
      "HTML report: ./fastp_out/C13W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C13W.R1.fastq -I 2_demultiplexed/CW1/C13W.R2.fastq -o 3_qc/CW1/C13W.R1.fastq -O 3_qc/CW1/C13W.R2.fastq --unpaired1 3_qc/CW1/C13W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C13W_unpaired.R2.fastq --failed_out 3_qc/CW1/C13W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C13W_merged.fastq -w 6 -h ./fastp_out/C13W_fastp.html -j ./fastp_out/C13W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 64180\n",
      "total bases: 6186025\n",
      "Q20 bases: 5969808(96.5048%)\n",
      "Q30 bases: 5684664(91.8953%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 64180\n",
      "total bases: 6125787\n",
      "Q20 bases: 5913485(96.5343%)\n",
      "Q30 bases: 5609895(91.5784%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 20514\n",
      "total bases: 2102780\n",
      "Q20 bases: 2084154(99.1142%)\n",
      "Q30 bases: 2027888(96.4384%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 45076\n",
      "reads failed due to low quality: 8956\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 74328\n",
      "reads with adapter trimmed: 43514\n",
      "bases trimmed due to adapters: 536034\n",
      "reads corrected by overlap analysis: 3716\n",
      "bases corrected by overlap analysis: 6430\n",
      "\n",
      "Duplication rate: 12.987%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 20514\n",
      "% of original read pairs: 31.9632%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C14W_fastp.json\n",
      "HTML report: ./fastp_out/C14W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C14W.R1.fastq -I 2_demultiplexed/CW1/C14W.R2.fastq -o 3_qc/CW1/C14W.R1.fastq -O 3_qc/CW1/C14W.R2.fastq --unpaired1 3_qc/CW1/C14W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C14W_unpaired.R2.fastq --failed_out 3_qc/CW1/C14W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C14W_merged.fastq -w 6 -h ./fastp_out/C14W_fastp.html -j ./fastp_out/C14W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 117568\n",
      "total bases: 11948673\n",
      "Q20 bases: 11481629(96.0912%)\n",
      "Q30 bases: 10948974(91.6334%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 117568\n",
      "total bases: 11614898\n",
      "Q20 bases: 11283757(97.149%)\n",
      "Q30 bases: 10775656(92.7744%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 48400\n",
      "total bases: 4966401\n",
      "Q20 bases: 4925104(99.1685%)\n",
      "Q30 bases: 4797854(96.6063%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 97844\n",
      "reads failed due to low quality: 17324\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 119968\n",
      "reads with adapter trimmed: 98254\n",
      "bases trimmed due to adapters: 1039768\n",
      "reads corrected by overlap analysis: 7017\n",
      "bases corrected by overlap analysis: 11450\n",
      "\n",
      "Duplication rate: 23.301%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 48400\n",
      "% of original read pairs: 41.1677%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C15W_fastp.json\n",
      "HTML report: ./fastp_out/C15W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C15W.R1.fastq -I 2_demultiplexed/CW1/C15W.R2.fastq -o 3_qc/CW1/C15W.R1.fastq -O 3_qc/CW1/C15W.R2.fastq --unpaired1 3_qc/CW1/C15W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C15W_unpaired.R2.fastq --failed_out 3_qc/CW1/C15W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C15W_merged.fastq -w 6 -h ./fastp_out/C15W_fastp.html -j ./fastp_out/C15W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 162806\n",
      "total bases: 24784815\n",
      "Q20 bases: 24414303(98.5051%)\n",
      "Q30 bases: 23659426(95.4594%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 162806\n",
      "total bases: 24830108\n",
      "Q20 bases: 24181540(97.388%)\n",
      "Q30 bases: 23046135(92.8153%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 154993\n",
      "total bases: 15864666\n",
      "Q20 bases: 15730101(99.1518%)\n",
      "Q30 bases: 15322503(96.5826%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 310686\n",
      "reads failed due to low quality: 6200\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 8724\n",
      "reads with adapter trimmed: 312788\n",
      "bases trimmed due to adapters: 3925768\n",
      "reads corrected by overlap analysis: 24964\n",
      "bases corrected by overlap analysis: 42132\n",
      "\n",
      "Duplication rate: 70.5674%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 154993\n",
      "% of original read pairs: 95.201%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C16W_fastp.json\n",
      "HTML report: ./fastp_out/C16W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C16W.R1.fastq -I 2_demultiplexed/CW1/C16W.R2.fastq -o 3_qc/CW1/C16W.R1.fastq -O 3_qc/CW1/C16W.R2.fastq --unpaired1 3_qc/CW1/C16W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C16W_unpaired.R2.fastq --failed_out 3_qc/CW1/C16W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C16W_merged.fastq -w 6 -h ./fastp_out/C16W_fastp.html -j ./fastp_out/C16W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 78537\n",
      "total bases: 11907619\n",
      "Q20 bases: 11701215(98.2666%)\n",
      "Q30 bases: 11300670(94.9029%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 78537\n",
      "total bases: 11925394\n",
      "Q20 bases: 11571656(97.0337%)\n",
      "Q30 bases: 10984094(92.1068%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 74467\n",
      "total bases: 7611483\n",
      "Q20 bases: 7543565(99.1077%)\n",
      "Q30 bases: 7336283(96.3844%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 150028\n",
      "reads failed due to low quality: 3658\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 3384\n",
      "reads with adapter trimmed: 150894\n",
      "bases trimmed due to adapters: 1763349\n",
      "reads corrected by overlap analysis: 13090\n",
      "bases corrected by overlap analysis: 22486\n",
      "\n",
      "Duplication rate: 61.3065%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 74467\n",
      "% of original read pairs: 94.8177%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C17W_fastp.json\n",
      "HTML report: ./fastp_out/C17W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C17W.R1.fastq -I 2_demultiplexed/CW1/C17W.R2.fastq -o 3_qc/CW1/C17W.R1.fastq -O 3_qc/CW1/C17W.R2.fastq --unpaired1 3_qc/CW1/C17W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C17W_unpaired.R2.fastq --failed_out 3_qc/CW1/C17W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C17W_merged.fastq -w 6 -h ./fastp_out/C17W_fastp.html -j ./fastp_out/C17W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 75099\n",
      "total bases: 8848665\n",
      "Q20 bases: 8626583(97.4902%)\n",
      "Q30 bases: 8284065(93.6194%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 75099\n",
      "total bases: 8823477\n",
      "Q20 bases: 8529586(96.6692%)\n",
      "Q30 bases: 8086370(91.6461%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 45391\n",
      "total bases: 4645567\n",
      "Q20 bases: 4604096(99.1073%)\n",
      "Q30 bases: 4479189(96.4186%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 91204\n",
      "reads failed due to low quality: 7396\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 51598\n",
      "reads with adapter trimmed: 92092\n",
      "bases trimmed due to adapters: 982458\n",
      "reads corrected by overlap analysis: 7709\n",
      "bases corrected by overlap analysis: 13252\n",
      "\n",
      "Duplication rate: 22.6415%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 45391\n",
      "% of original read pairs: 60.4416%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C18W_fastp.json\n",
      "HTML report: ./fastp_out/C18W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C18W.R1.fastq -I 2_demultiplexed/CW1/C18W.R2.fastq -o 3_qc/CW1/C18W.R1.fastq -O 3_qc/CW1/C18W.R2.fastq --unpaired1 3_qc/CW1/C18W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C18W_unpaired.R2.fastq --failed_out 3_qc/CW1/C18W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C18W_merged.fastq -w 6 -h ./fastp_out/C18W_fastp.html -j ./fastp_out/C18W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 39131\n",
      "total bases: 4916283\n",
      "Q20 bases: 4776941(97.1657%)\n",
      "Q30 bases: 4595905(93.4833%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 39131\n",
      "total bases: 4867520\n",
      "Q20 bases: 4735217(97.2819%)\n",
      "Q30 bases: 4520416(92.869%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 25129\n",
      "total bases: 2571333\n",
      "Q20 bases: 2550717(99.1982%)\n",
      "Q30 bases: 2486805(96.7127%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 50840\n",
      "reads failed due to low quality: 4160\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 23258\n",
      "reads with adapter trimmed: 51134\n",
      "bases trimmed due to adapters: 632075\n",
      "reads corrected by overlap analysis: 3794\n",
      "bases corrected by overlap analysis: 6470\n",
      "\n",
      "Duplication rate: 21.3483%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 25129\n",
      "% of original read pairs: 64.2176%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C19W_fastp.json\n",
      "HTML report: ./fastp_out/C19W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C19W.R1.fastq -I 2_demultiplexed/CW1/C19W.R2.fastq -o 3_qc/CW1/C19W.R1.fastq -O 3_qc/CW1/C19W.R2.fastq --unpaired1 3_qc/CW1/C19W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C19W_unpaired.R2.fastq --failed_out 3_qc/CW1/C19W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C19W_merged.fastq -w 6 -h ./fastp_out/C19W_fastp.html -j ./fastp_out/C19W_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 50325\n",
      "total bases: 7509743\n",
      "Q20 bases: 7409062(98.6593%)\n",
      "Q30 bases: 7189145(95.7309%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 50325\n",
      "total bases: 7529601\n",
      "Q20 bases: 7340851(97.4932%)\n",
      "Q30 bases: 7009668(93.0948%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 46598\n",
      "total bases: 4767573\n",
      "Q20 bases: 4731216(99.2374%)\n",
      "Q30 bases: 4613531(96.769%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 94558\n",
      "reads failed due to low quality: 1808\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 4282\n",
      "reads with adapter trimmed: 94602\n",
      "bases trimmed due to adapters: 1089352\n",
      "reads corrected by overlap analysis: 7064\n",
      "bases corrected by overlap analysis: 11774\n",
      "\n",
      "Duplication rate: 39.4495%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 46598\n",
      "% of original read pairs: 92.5941%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C20W_fastp.json\n",
      "HTML report: ./fastp_out/C20W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C20W.R1.fastq -I 2_demultiplexed/CW1/C20W.R2.fastq -o 3_qc/CW1/C20W.R1.fastq -O 3_qc/CW1/C20W.R2.fastq --unpaired1 3_qc/CW1/C20W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C20W_unpaired.R2.fastq --failed_out 3_qc/CW1/C20W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C20W_merged.fastq -w 6 -h ./fastp_out/C20W_fastp.html -j ./fastp_out/C20W_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 20392\n",
      "total bases: 2505046\n",
      "Q20 bases: 2409327(96.179%)\n",
      "Q30 bases: 2294663(91.6016%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 20392\n",
      "total bases: 2451893\n",
      "Q20 bases: 2369244(96.6292%)\n",
      "Q30 bases: 2249718(91.7543%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 11257\n",
      "total bases: 1151217\n",
      "Q20 bases: 1140107(99.0349%)\n",
      "Q30 bases: 1106945(96.1543%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 25704\n",
      "reads failed due to low quality: 2992\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 12086\n",
      "reads with adapter trimmed: 23400\n",
      "bases trimmed due to adapters: 242467\n",
      "reads corrected by overlap analysis: 2001\n",
      "bases corrected by overlap analysis: 3415\n",
      "\n",
      "Duplication rate: 11.2903%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 11257\n",
      "% of original read pairs: 55.203%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C21W_fastp.json\n",
      "HTML report: ./fastp_out/C21W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/C21W.R1.fastq -I 2_demultiplexed/CW1/C21W.R2.fastq -o 3_qc/CW1/C21W.R1.fastq -O 3_qc/CW1/C21W.R2.fastq --unpaired1 3_qc/CW1/C21W_unpaired.R1.fastq --unpaired2 3_qc/CW1/C21W_unpaired.R2.fastq --failed_out 3_qc/CW1/C21W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/C21W_merged.fastq -w 6 -h ./fastp_out/C21W_fastp.html -j ./fastp_out/C21W_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 9779\n",
      "total bases: 655791\n",
      "Q20 bases: 618856(94.3679%)\n",
      "Q30 bases: 579735(88.4024%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 9779\n",
      "total bases: 636328\n",
      "Q20 bases: 602041(94.6117%)\n",
      "Q30 bases: 561662(88.2661%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 1671\n",
      "total bases: 172096\n",
      "Q20 bases: 170667(99.1696%)\n",
      "Q30 bases: 166261(96.6095%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3344\n",
      "reads failed due to low quality: 2188\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 14026\n",
      "reads with adapter trimmed: 3388\n",
      "bases trimmed due to adapters: 39668\n",
      "reads corrected by overlap analysis: 397\n",
      "bases corrected by overlap analysis: 682\n",
      "\n",
      "Duplication rate: 15.6863%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 1671\n",
      "% of original read pairs: 17.0876%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/CF3W_fastp.json\n",
      "HTML report: ./fastp_out/CF3W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/CF3W.R1.fastq -I 2_demultiplexed/CW1/CF3W.R2.fastq -o 3_qc/CW1/CF3W.R1.fastq -O 3_qc/CW1/CF3W.R2.fastq --unpaired1 3_qc/CW1/CF3W_unpaired.R1.fastq --unpaired2 3_qc/CW1/CF3W_unpaired.R2.fastq --failed_out 3_qc/CW1/CF3W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/CF3W_merged.fastq -w 6 -h ./fastp_out/CF3W_fastp.html -j ./fastp_out/CF3W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 47779\n",
      "total bases: 3322706\n",
      "Q20 bases: 3111581(93.646%)\n",
      "Q30 bases: 2906668(87.4789%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 47779\n",
      "total bases: 3156217\n",
      "Q20 bases: 3027740(95.9294%)\n",
      "Q30 bases: 2867818(90.8625%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 1289\n",
      "total bases: 132745\n",
      "Q20 bases: 131621(99.1533%)\n",
      "Q30 bases: 128410(96.7343%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2644\n",
      "reads failed due to low quality: 9926\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 82988\n",
      "reads with adapter trimmed: 2602\n",
      "bases trimmed due to adapters: 30043\n",
      "reads corrected by overlap analysis: 345\n",
      "bases corrected by overlap analysis: 605\n",
      "\n",
      "Duplication rate: 13.6986%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 1289\n",
      "% of original read pairs: 2.69784%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/CN1W_fastp.json\n",
      "HTML report: ./fastp_out/CN1W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/CN1W.R1.fastq -I 2_demultiplexed/CW1/CN1W.R2.fastq -o 3_qc/CW1/CN1W.R1.fastq -O 3_qc/CW1/CN1W.R2.fastq --unpaired1 3_qc/CW1/CN1W_unpaired.R1.fastq --unpaired2 3_qc/CW1/CN1W_unpaired.R2.fastq --failed_out 3_qc/CW1/CN1W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/CN1W_merged.fastq -w 6 -h ./fastp_out/CN1W_fastp.html -j ./fastp_out/CN1W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 82515\n",
      "total bases: 12733855\n",
      "Q20 bases: 12550761(98.5621%)\n",
      "Q30 bases: 12161064(95.5018%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 82515\n",
      "total bases: 12790758\n",
      "Q20 bases: 12408579(97.0121%)\n",
      "Q30 bases: 11737750(91.7674%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 80897\n",
      "total bases: 8250541\n",
      "Q20 bases: 8177610(99.116%)\n",
      "Q30 bases: 7955627(96.4255%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 162092\n",
      "reads failed due to low quality: 2568\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 370\n",
      "reads with adapter trimmed: 163042\n",
      "bases trimmed due to adapters: 2065947\n",
      "reads corrected by overlap analysis: 16228\n",
      "bases corrected by overlap analysis: 26565\n",
      "\n",
      "Duplication rate: 81.4118%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 80897\n",
      "% of original read pairs: 98.0391%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/CP1W_fastp.json\n",
      "HTML report: ./fastp_out/CP1W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW1/CP1W.R1.fastq -I 2_demultiplexed/CW1/CP1W.R2.fastq -o 3_qc/CW1/CP1W.R1.fastq -O 3_qc/CW1/CP1W.R2.fastq --unpaired1 3_qc/CW1/CP1W_unpaired.R1.fastq --unpaired2 3_qc/CW1/CP1W_unpaired.R2.fastq --failed_out 3_qc/CW1/CP1W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW1/CP1W_merged.fastq -w 6 -h ./fastp_out/CP1W_fastp.html -j ./fastp_out/CP1W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 188447\n",
      "total bases: 29041461\n",
      "Q20 bases: 28557415(98.3333%)\n",
      "Q30 bases: 27610975(95.0743%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 188447\n",
      "total bases: 29071340\n",
      "Q20 bases: 28432442(97.8023%)\n",
      "Q30 bases: 27245132(93.7182%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 181871\n",
      "total bases: 18680097\n",
      "Q20 bases: 18503442(99.0543%)\n",
      "Q30 bases: 17984899(96.2784%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 365536\n",
      "reads failed due to low quality: 7716\n",
      "reads failed due to too many N: 10\n",
      "reads failed due to too short: 3632\n",
      "reads with adapter trimmed: 368416\n",
      "bases trimmed due to adapters: 4566891\n",
      "reads corrected by overlap analysis: 27176\n",
      "bases corrected by overlap analysis: 43996\n",
      "\n",
      "Duplication rate: 24.1107%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 181871\n",
      "% of original read pairs: 96.5104%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C22W_fastp.json\n",
      "HTML report: ./fastp_out/C22W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C22W.R1.fastq -I 2_demultiplexed/CW2/C22W.R2.fastq -o 3_qc/CW2/C22W.R1.fastq -O 3_qc/CW2/C22W.R2.fastq --unpaired1 3_qc/CW2/C22W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C22W_unpaired.R2.fastq --failed_out 3_qc/CW2/C22W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C22W_merged.fastq -w 6 -h ./fastp_out/C22W_fastp.html -j ./fastp_out/C22W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 79388\n",
      "total bases: 11751110\n",
      "Q20 bases: 11535410(98.1644%)\n",
      "Q30 bases: 11127020(94.6891%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 79388\n",
      "total bases: 11754007\n",
      "Q20 bases: 11462453(97.5195%)\n",
      "Q30 bases: 10954982(93.2021%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 63987\n",
      "total bases: 6574096\n",
      "Q20 bases: 6512542(99.0637%)\n",
      "Q30 bases: 6331024(96.3026%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 142684\n",
      "reads failed due to low quality: 4964\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 11128\n",
      "reads with adapter trimmed: 138124\n",
      "bases trimmed due to adapters: 1553840\n",
      "reads corrected by overlap analysis: 10997\n",
      "bases corrected by overlap analysis: 18590\n",
      "\n",
      "Duplication rate: 33.0275%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 63987\n",
      "% of original read pairs: 80.6003%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C23W_fastp.json\n",
      "HTML report: ./fastp_out/C23W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C23W.R1.fastq -I 2_demultiplexed/CW2/C23W.R2.fastq -o 3_qc/CW2/C23W.R1.fastq -O 3_qc/CW2/C23W.R2.fastq --unpaired1 3_qc/CW2/C23W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C23W_unpaired.R2.fastq --failed_out 3_qc/CW2/C23W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C23W_merged.fastq -w 6 -h ./fastp_out/C23W_fastp.html -j ./fastp_out/C23W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 78555\n",
      "total bases: 11775964\n",
      "Q20 bases: 11571281(98.2619%)\n",
      "Q30 bases: 11159514(94.7652%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 78555\n",
      "total bases: 11788432\n",
      "Q20 bases: 11510482(97.6422%)\n",
      "Q30 bases: 10994210(93.2627%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 72730\n",
      "total bases: 7468416\n",
      "Q20 bases: 7397619(99.052%)\n",
      "Q30 bases: 7183221(96.1813%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 148506\n",
      "reads failed due to low quality: 3476\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 5126\n",
      "reads with adapter trimmed: 147596\n",
      "bases trimmed due to adapters: 1558425\n",
      "reads corrected by overlap analysis: 10714\n",
      "bases corrected by overlap analysis: 17236\n",
      "\n",
      "Duplication rate: 33.0508%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 72730\n",
      "% of original read pairs: 92.5848%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C24W_fastp.json\n",
      "HTML report: ./fastp_out/C24W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C24W.R1.fastq -I 2_demultiplexed/CW2/C24W.R2.fastq -o 3_qc/CW2/C24W.R1.fastq -O 3_qc/CW2/C24W.R2.fastq --unpaired1 3_qc/CW2/C24W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C24W_unpaired.R2.fastq --failed_out 3_qc/CW2/C24W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C24W_merged.fastq -w 6 -h ./fastp_out/C24W_fastp.html -j ./fastp_out/C24W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 62174\n",
      "total bases: 9255429\n",
      "Q20 bases: 9068871(97.9843%)\n",
      "Q30 bases: 8715246(94.1636%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 62174\n",
      "total bases: 9262767\n",
      "Q20 bases: 9011842(97.291%)\n",
      "Q30 bases: 8574672(92.5714%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 51282\n",
      "total bases: 5275441\n",
      "Q20 bases: 5223074(99.0073%)\n",
      "Q30 bases: 5065445(96.0194%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 112418\n",
      "reads failed due to low quality: 3904\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 8020\n",
      "reads with adapter trimmed: 107896\n",
      "bases trimmed due to adapters: 1337680\n",
      "reads corrected by overlap analysis: 9042\n",
      "bases corrected by overlap analysis: 14702\n",
      "\n",
      "Duplication rate: 43.5583%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 51282\n",
      "% of original read pairs: 82.4814%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C25W_fastp.json\n",
      "HTML report: ./fastp_out/C25W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C25W.R1.fastq -I 2_demultiplexed/CW2/C25W.R2.fastq -o 3_qc/CW2/C25W.R1.fastq -O 3_qc/CW2/C25W.R2.fastq --unpaired1 3_qc/CW2/C25W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C25W_unpaired.R2.fastq --failed_out 3_qc/CW2/C25W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C25W_merged.fastq -w 6 -h ./fastp_out/C25W_fastp.html -j ./fastp_out/C25W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 57417\n",
      "total bases: 7989478\n",
      "Q20 bases: 7823757(97.9258%)\n",
      "Q30 bases: 7528158(94.2259%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 57417\n",
      "total bases: 7975445\n",
      "Q20 bases: 7792499(97.7061%)\n",
      "Q30 bases: 7467661(93.6332%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 40045\n",
      "total bases: 4099074\n",
      "Q20 bases: 4059867(99.0435%)\n",
      "Q30 bases: 3942615(96.1831%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 91808\n",
      "reads failed due to low quality: 4106\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 18920\n",
      "reads with adapter trimmed: 86536\n",
      "bases trimmed due to adapters: 969287\n",
      "reads corrected by overlap analysis: 6859\n",
      "bases corrected by overlap analysis: 11421\n",
      "\n",
      "Duplication rate: 23.5955%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 40045\n",
      "% of original read pairs: 69.7442%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C26W_fastp.json\n",
      "HTML report: ./fastp_out/C26W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C26W.R1.fastq -I 2_demultiplexed/CW2/C26W.R2.fastq -o 3_qc/CW2/C26W.R1.fastq -O 3_qc/CW2/C26W.R2.fastq --unpaired1 3_qc/CW2/C26W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C26W_unpaired.R2.fastq --failed_out 3_qc/CW2/C26W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C26W_merged.fastq -w 6 -h ./fastp_out/C26W_fastp.html -j ./fastp_out/C26W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 79493\n",
      "total bases: 12021295\n",
      "Q20 bases: 11819263(98.3194%)\n",
      "Q30 bases: 11424779(95.0378%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 79493\n",
      "total bases: 12033080\n",
      "Q20 bases: 11734278(97.5168%)\n",
      "Q30 bases: 11202589(93.0983%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 74298\n",
      "total bases: 7600332\n",
      "Q20 bases: 7533099(99.1154%)\n",
      "Q30 bases: 7326573(96.3981%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 152164\n",
      "reads failed due to low quality: 3476\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 3346\n",
      "reads with adapter trimmed: 152890\n",
      "bases trimmed due to adapters: 1603651\n",
      "reads corrected by overlap analysis: 11267\n",
      "bases corrected by overlap analysis: 18499\n",
      "\n",
      "Duplication rate: 57.6471%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 74298\n",
      "% of original read pairs: 93.4648%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C27W_fastp.json\n",
      "HTML report: ./fastp_out/C27W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C27W.R1.fastq -I 2_demultiplexed/CW2/C27W.R2.fastq -o 3_qc/CW2/C27W.R1.fastq -O 3_qc/CW2/C27W.R2.fastq --unpaired1 3_qc/CW2/C27W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C27W_unpaired.R2.fastq --failed_out 3_qc/CW2/C27W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C27W_merged.fastq -w 6 -h ./fastp_out/C27W_fastp.html -j ./fastp_out/C27W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 71304\n",
      "total bases: 10454290\n",
      "Q20 bases: 10269306(98.2305%)\n",
      "Q30 bases: 9927545(94.9614%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 71304\n",
      "total bases: 10427000\n",
      "Q20 bases: 10233260(98.1419%)\n",
      "Q30 bases: 9866330(94.6229%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 61709\n",
      "total bases: 6311980\n",
      "Q20 bases: 6254422(99.0881%)\n",
      "Q30 bases: 6083336(96.3776%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 127416\n",
      "reads failed due to low quality: 3628\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 11562\n",
      "reads with adapter trimmed: 126680\n",
      "bases trimmed due to adapters: 1581775\n",
      "reads corrected by overlap analysis: 8615\n",
      "bases corrected by overlap analysis: 13815\n",
      "\n",
      "Duplication rate: 47.9167%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 61709\n",
      "% of original read pairs: 86.5435%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C28W_fastp.json\n",
      "HTML report: ./fastp_out/C28W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C28W.R1.fastq -I 2_demultiplexed/CW2/C28W.R2.fastq -o 3_qc/CW2/C28W.R1.fastq -O 3_qc/CW2/C28W.R2.fastq --unpaired1 3_qc/CW2/C28W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C28W_unpaired.R2.fastq --failed_out 3_qc/CW2/C28W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C28W_merged.fastq -w 6 -h ./fastp_out/C28W_fastp.html -j ./fastp_out/C28W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 101064\n",
      "total bases: 15357563\n",
      "Q20 bases: 15094330(98.286%)\n",
      "Q30 bases: 14582714(94.9546%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 101064\n",
      "total bases: 15367087\n",
      "Q20 bases: 15003089(97.6313%)\n",
      "Q30 bases: 14342584(93.3331%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 95950\n",
      "total bases: 9840668\n",
      "Q20 bases: 9749291(99.0714%)\n",
      "Q30 bases: 9471360(96.2471%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 194546\n",
      "reads failed due to low quality: 4292\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 3290\n",
      "reads with adapter trimmed: 195912\n",
      "bases trimmed due to adapters: 2064534\n",
      "reads corrected by overlap analysis: 14274\n",
      "bases corrected by overlap analysis: 23137\n",
      "\n",
      "Duplication rate: 40.625%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 95950\n",
      "% of original read pairs: 94.9398%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C29W_fastp.json\n",
      "HTML report: ./fastp_out/C29W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C29W.R1.fastq -I 2_demultiplexed/CW2/C29W.R2.fastq -o 3_qc/CW2/C29W.R1.fastq -O 3_qc/CW2/C29W.R2.fastq --unpaired1 3_qc/CW2/C29W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C29W_unpaired.R2.fastq --failed_out 3_qc/CW2/C29W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C29W_merged.fastq -w 6 -h ./fastp_out/C29W_fastp.html -j ./fastp_out/C29W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 99149\n",
      "total bases: 15204294\n",
      "Q20 bases: 14971933(98.4717%)\n",
      "Q30 bases: 14495648(95.3392%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 99149\n",
      "total bases: 15223662\n",
      "Q20 bases: 14908592(97.9304%)\n",
      "Q30 bases: 14303929(93.9585%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 94581\n",
      "total bases: 9690113\n",
      "Q20 bases: 9606595(99.1381%)\n",
      "Q30 bases: 9352383(96.5147%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 191424\n",
      "reads failed due to low quality: 3308\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 3562\n",
      "reads with adapter trimmed: 192122\n",
      "bases trimmed due to adapters: 2395670\n",
      "reads corrected by overlap analysis: 13417\n",
      "bases corrected by overlap analysis: 21505\n",
      "\n",
      "Duplication rate: 57.0621%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 94581\n",
      "% of original read pairs: 95.3928%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C30W_fastp.json\n",
      "HTML report: ./fastp_out/C30W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C30W.R1.fastq -I 2_demultiplexed/CW2/C30W.R2.fastq -o 3_qc/CW2/C30W.R1.fastq -O 3_qc/CW2/C30W.R2.fastq --unpaired1 3_qc/CW2/C30W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C30W_unpaired.R2.fastq --failed_out 3_qc/CW2/C30W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C30W_merged.fastq -w 6 -h ./fastp_out/C30W_fastp.html -j ./fastp_out/C30W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 67054\n",
      "total bases: 9712134\n",
      "Q20 bases: 9538986(98.2172%)\n",
      "Q30 bases: 9209592(94.8256%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 67054\n",
      "total bases: 9713058\n",
      "Q20 bases: 9493742(97.742%)\n",
      "Q30 bases: 9101901(93.7079%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 58145\n",
      "total bases: 5955410\n",
      "Q20 bases: 5897856(99.0336%)\n",
      "Q30 bases: 5730253(96.2193%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 120556\n",
      "reads failed due to low quality: 3396\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 10156\n",
      "reads with adapter trimmed: 119000\n",
      "bases trimmed due to adapters: 1370946\n",
      "reads corrected by overlap analysis: 9060\n",
      "bases corrected by overlap analysis: 14774\n",
      "\n",
      "Duplication rate: 17.9775%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 58145\n",
      "% of original read pairs: 86.7137%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C31W_fastp.json\n",
      "HTML report: ./fastp_out/C31W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C31W.R1.fastq -I 2_demultiplexed/CW2/C31W.R2.fastq -o 3_qc/CW2/C31W.R1.fastq -O 3_qc/CW2/C31W.R2.fastq --unpaired1 3_qc/CW2/C31W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C31W_unpaired.R2.fastq --failed_out 3_qc/CW2/C31W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C31W_merged.fastq -w 6 -h ./fastp_out/C31W_fastp.html -j ./fastp_out/C31W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 72763\n",
      "total bases: 10517490\n",
      "Q20 bases: 10279933(97.7413%)\n",
      "Q30 bases: 9845915(93.6147%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 72763\n",
      "total bases: 10504072\n",
      "Q20 bases: 10233959(97.4285%)\n",
      "Q30 bases: 9747349(92.7959%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 60706\n",
      "total bases: 6218871\n",
      "Q20 bases: 6151571(98.9178%)\n",
      "Q30 bases: 5949554(95.6694%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 128440\n",
      "reads failed due to low quality: 4936\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 12150\n",
      "reads with adapter trimmed: 127446\n",
      "bases trimmed due to adapters: 1336025\n",
      "reads corrected by overlap analysis: 10693\n",
      "bases corrected by overlap analysis: 17300\n",
      "\n",
      "Duplication rate: 36.9231%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 60706\n",
      "% of original read pairs: 83.4298%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C32W_fastp.json\n",
      "HTML report: ./fastp_out/C32W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C32W.R1.fastq -I 2_demultiplexed/CW2/C32W.R2.fastq -o 3_qc/CW2/C32W.R1.fastq -O 3_qc/CW2/C32W.R2.fastq --unpaired1 3_qc/CW2/C32W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C32W_unpaired.R2.fastq --failed_out 3_qc/CW2/C32W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C32W_merged.fastq -w 6 -h ./fastp_out/C32W_fastp.html -j ./fastp_out/C32W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 76555\n",
      "total bases: 6865385\n",
      "Q20 bases: 6641418(96.7377%)\n",
      "Q30 bases: 6323696(92.1099%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 76555\n",
      "total bases: 6778040\n",
      "Q20 bases: 6592961(97.2694%)\n",
      "Q30 bases: 6315977(93.1829%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 22031\n",
      "total bases: 2247383\n",
      "Q20 bases: 2225238(99.0146%)\n",
      "Q30 bases: 2158628(96.0507%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 45306\n",
      "reads failed due to low quality: 9574\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 98230\n",
      "reads with adapter trimmed: 45216\n",
      "bases trimmed due to adapters: 522055\n",
      "reads corrected by overlap analysis: 3655\n",
      "bases corrected by overlap analysis: 5967\n",
      "\n",
      "Duplication rate: 40.566%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 22031\n",
      "% of original read pairs: 28.778%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C33W_fastp.json\n",
      "HTML report: ./fastp_out/C33W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C33W.R1.fastq -I 2_demultiplexed/CW2/C33W.R2.fastq -o 3_qc/CW2/C33W.R1.fastq -O 3_qc/CW2/C33W.R2.fastq --unpaired1 3_qc/CW2/C33W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C33W_unpaired.R2.fastq --failed_out 3_qc/CW2/C33W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C33W_merged.fastq -w 6 -h ./fastp_out/C33W_fastp.html -j ./fastp_out/C33W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 125739\n",
      "total bases: 18762013\n",
      "Q20 bases: 18408673(98.1167%)\n",
      "Q30 bases: 17736758(94.5355%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 125739\n",
      "total bases: 18738471\n",
      "Q20 bases: 18331857(97.8301%)\n",
      "Q30 bases: 17577702(93.8054%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 113468\n",
      "total bases: 11592627\n",
      "Q20 bases: 11477736(99.0089%)\n",
      "Q30 bases: 11132804(96.0335%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 235040\n",
      "reads failed due to low quality: 6706\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 9730\n",
      "reads with adapter trimmed: 235018\n",
      "bases trimmed due to adapters: 2453240\n",
      "reads corrected by overlap analysis: 16973\n",
      "bases corrected by overlap analysis: 27541\n",
      "\n",
      "Duplication rate: 33.5878%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 113468\n",
      "% of original read pairs: 90.2409%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C34W_fastp.json\n",
      "HTML report: ./fastp_out/C34W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C34W.R1.fastq -I 2_demultiplexed/CW2/C34W.R2.fastq -o 3_qc/CW2/C34W.R1.fastq -O 3_qc/CW2/C34W.R2.fastq --unpaired1 3_qc/CW2/C34W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C34W_unpaired.R2.fastq --failed_out 3_qc/CW2/C34W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C34W_merged.fastq -w 6 -h ./fastp_out/C34W_fastp.html -j ./fastp_out/C34W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 112364\n",
      "total bases: 15220293\n",
      "Q20 bases: 14895033(97.863%)\n",
      "Q30 bases: 14337442(94.1995%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 112364\n",
      "total bases: 15180883\n",
      "Q20 bases: 14845413(97.7902%)\n",
      "Q30 bases: 14252055(93.8816%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 85679\n",
      "total bases: 8777203\n",
      "Q20 bases: 8694311(99.0556%)\n",
      "Q30 bases: 8448821(96.2587%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 172896\n",
      "reads failed due to low quality: 7806\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 44022\n",
      "reads with adapter trimmed: 173934\n",
      "bases trimmed due to adapters: 2198802\n",
      "reads corrected by overlap analysis: 12441\n",
      "bases corrected by overlap analysis: 19925\n",
      "\n",
      "Duplication rate: 18.8119%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 85679\n",
      "% of original read pairs: 76.2513%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C35W_fastp.json\n",
      "HTML report: ./fastp_out/C35W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C35W.R1.fastq -I 2_demultiplexed/CW2/C35W.R2.fastq -o 3_qc/CW2/C35W.R1.fastq -O 3_qc/CW2/C35W.R2.fastq --unpaired1 3_qc/CW2/C35W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C35W_unpaired.R2.fastq --failed_out 3_qc/CW2/C35W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C35W_merged.fastq -w 6 -h ./fastp_out/C35W_fastp.html -j ./fastp_out/C35W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 117278\n",
      "total bases: 13706077\n",
      "Q20 bases: 13281990(96.9058%)\n",
      "Q30 bases: 12712594(92.7515%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 117278\n",
      "total bases: 13435954\n",
      "Q20 bases: 13141991(97.8121%)\n",
      "Q30 bases: 12635637(94.0435%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 67616\n",
      "total bases: 6924550\n",
      "Q20 bases: 6859009(99.0535%)\n",
      "Q30 bases: 6662612(96.2173%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 136804\n",
      "reads failed due to low quality: 13070\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 84682\n",
      "reads with adapter trimmed: 137622\n",
      "bases trimmed due to adapters: 1454970\n",
      "reads corrected by overlap analysis: 9284\n",
      "bases corrected by overlap analysis: 14877\n",
      "\n",
      "Duplication rate: 11.2245%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 67616\n",
      "% of original read pairs: 57.6545%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C36W_fastp.json\n",
      "HTML report: ./fastp_out/C36W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C36W.R1.fastq -I 2_demultiplexed/CW2/C36W.R2.fastq -o 3_qc/CW2/C36W.R1.fastq -O 3_qc/CW2/C36W.R2.fastq --unpaired1 3_qc/CW2/C36W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C36W_unpaired.R2.fastq --failed_out 3_qc/CW2/C36W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C36W_merged.fastq -w 6 -h ./fastp_out/C36W_fastp.html -j ./fastp_out/C36W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 241549\n",
      "total bases: 36348507\n",
      "Q20 bases: 35734314(98.3103%)\n",
      "Q30 bases: 34521531(94.9737%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 241549\n",
      "total bases: 36367038\n",
      "Q20 bases: 35588436(97.859%)\n",
      "Q30 bases: 34132476(93.8555%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 224298\n",
      "total bases: 22960110\n",
      "Q20 bases: 22742464(99.0521%)\n",
      "Q30 bases: 22095090(96.2325%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 451696\n",
      "reads failed due to low quality: 10376\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 21024\n",
      "reads with adapter trimmed: 454302\n",
      "bases trimmed due to adapters: 5683849\n",
      "reads corrected by overlap analysis: 33398\n",
      "bases corrected by overlap analysis: 54160\n",
      "\n",
      "Duplication rate: 46.2428%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 224298\n",
      "% of original read pairs: 92.8582%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C37W_fastp.json\n",
      "HTML report: ./fastp_out/C37W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C37W.R1.fastq -I 2_demultiplexed/CW2/C37W.R2.fastq -o 3_qc/CW2/C37W.R1.fastq -O 3_qc/CW2/C37W.R2.fastq --unpaired1 3_qc/CW2/C37W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C37W_unpaired.R2.fastq --failed_out 3_qc/CW2/C37W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C37W_merged.fastq -w 6 -h ./fastp_out/C37W_fastp.html -j ./fastp_out/C37W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 160526\n",
      "total bases: 24446185\n",
      "Q20 bases: 23997670(98.1653%)\n",
      "Q30 bases: 23131168(94.6208%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 160526\n",
      "total bases: 24446991\n",
      "Q20 bases: 23873423(97.6538%)\n",
      "Q30 bases: 22830318(93.387%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 147345\n",
      "total bases: 15141755\n",
      "Q20 bases: 14999654(99.0615%)\n",
      "Q30 bases: 14570932(96.2301%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 305738\n",
      "reads failed due to low quality: 7754\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 7558\n",
      "reads with adapter trimmed: 302464\n",
      "bases trimmed due to adapters: 3480004\n",
      "reads corrected by overlap analysis: 23137\n",
      "bases corrected by overlap analysis: 38423\n",
      "\n",
      "Duplication rate: 29.0323%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 147345\n",
      "% of original read pairs: 91.7889%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C38W_fastp.json\n",
      "HTML report: ./fastp_out/C38W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C38W.R1.fastq -I 2_demultiplexed/CW2/C38W.R2.fastq -o 3_qc/CW2/C38W.R1.fastq -O 3_qc/CW2/C38W.R2.fastq --unpaired1 3_qc/CW2/C38W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C38W_unpaired.R2.fastq --failed_out 3_qc/CW2/C38W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C38W_merged.fastq -w 6 -h ./fastp_out/C38W_fastp.html -j ./fastp_out/C38W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 162973\n",
      "total bases: 18764254\n",
      "Q20 bases: 18275173(97.3935%)\n",
      "Q30 bases: 17513810(93.336%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 162973\n",
      "total bases: 18651520\n",
      "Q20 bases: 18140266(97.2589%)\n",
      "Q30 bases: 17318149(92.8511%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 92644\n",
      "total bases: 9498829\n",
      "Q20 bases: 9407090(99.0342%)\n",
      "Q30 bases: 9133159(96.1504%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 188434\n",
      "reads failed due to low quality: 16028\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 121480\n",
      "reads with adapter trimmed: 189202\n",
      "bases trimmed due to adapters: 2008444\n",
      "reads corrected by overlap analysis: 14181\n",
      "bases corrected by overlap analysis: 23244\n",
      "\n",
      "Duplication rate: 24.2038%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 92644\n",
      "% of original read pairs: 56.8462%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C39W_fastp.json\n",
      "HTML report: ./fastp_out/C39W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C39W.R1.fastq -I 2_demultiplexed/CW2/C39W.R2.fastq -o 3_qc/CW2/C39W.R1.fastq -O 3_qc/CW2/C39W.R2.fastq --unpaired1 3_qc/CW2/C39W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C39W_unpaired.R2.fastq --failed_out 3_qc/CW2/C39W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C39W_merged.fastq -w 6 -h ./fastp_out/C39W_fastp.html -j ./fastp_out/C39W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 87900\n",
      "total bases: 10862277\n",
      "Q20 bases: 10473037(96.4166%)\n",
      "Q30 bases: 9973943(91.8218%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 87900\n",
      "total bases: 10564319\n",
      "Q20 bases: 10300731(97.5049%)\n",
      "Q30 bases: 9828516(93.035%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 52901\n",
      "total bases: 5476384\n",
      "Q20 bases: 5419563(98.9624%)\n",
      "Q30 bases: 5253120(95.9231%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 107552\n",
      "reads failed due to low quality: 11844\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 56402\n",
      "reads with adapter trimmed: 107928\n",
      "bases trimmed due to adapters: 1330642\n",
      "reads corrected by overlap analysis: 8903\n",
      "bases corrected by overlap analysis: 14549\n",
      "\n",
      "Duplication rate: 18.0952%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 52901\n",
      "% of original read pairs: 60.1832%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C40W_fastp.json\n",
      "HTML report: ./fastp_out/C40W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/C40W.R1.fastq -I 2_demultiplexed/CW2/C40W.R2.fastq -o 3_qc/CW2/C40W.R1.fastq -O 3_qc/CW2/C40W.R2.fastq --unpaired1 3_qc/CW2/C40W_unpaired.R1.fastq --unpaired2 3_qc/CW2/C40W_unpaired.R2.fastq --failed_out 3_qc/CW2/C40W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/C40W_merged.fastq -w 6 -h ./fastp_out/C40W_fastp.html -j ./fastp_out/C40W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 7934\n",
      "total bases: 369908\n",
      "Q20 bases: 337881(91.3419%)\n",
      "Q30 bases: 308252(83.3321%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7934\n",
      "total bases: 345497\n",
      "Q20 bases: 322702(93.4023%)\n",
      "Q30 bases: 300775(87.0557%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 0\n",
      "total bases: 0\n",
      "Q20 bases: 0(-nan%)\n",
      "Q30 bases: 0(-nan%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 0\n",
      "reads failed due to low quality: 1906\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 13962\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "reads corrected by overlap analysis: 8\n",
      "bases corrected by overlap analysis: 14\n",
      "\n",
      "Duplication rate: 43.1034%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 0\n",
      "\n",
      "Read pairs merged: 0\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/CF2W_fastp.json\n",
      "HTML report: ./fastp_out/CF2W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/CF2W.R1.fastq -I 2_demultiplexed/CW2/CF2W.R2.fastq -o 3_qc/CW2/CF2W.R1.fastq -O 3_qc/CW2/CF2W.R2.fastq --unpaired1 3_qc/CW2/CF2W_unpaired.R1.fastq --unpaired2 3_qc/CW2/CF2W_unpaired.R2.fastq --failed_out 3_qc/CW2/CF2W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/CF2W_merged.fastq -w 6 -h ./fastp_out/CF2W_fastp.html -j ./fastp_out/CF2W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 47569\n",
      "total bases: 3219015\n",
      "Q20 bases: 2993145(92.9833%)\n",
      "Q30 bases: 2774080(86.1779%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 47569\n",
      "total bases: 2990875\n",
      "Q20 bases: 2888586(96.58%)\n",
      "Q30 bases: 2755266(92.1224%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 382\n",
      "total bases: 39345\n",
      "Q20 bases: 38891(98.8461%)\n",
      "Q30 bases: 37648(95.6869%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 838\n",
      "reads failed due to low quality: 9990\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 84310\n",
      "reads with adapter trimmed: 778\n",
      "bases trimmed due to adapters: 9250\n",
      "reads corrected by overlap analysis: 183\n",
      "bases corrected by overlap analysis: 355\n",
      "\n",
      "Duplication rate: 29.4737%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 382\n",
      "% of original read pairs: 0.803044%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/CN2W_fastp.json\n",
      "HTML report: ./fastp_out/CN2W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/CN2W.R1.fastq -I 2_demultiplexed/CW2/CN2W.R2.fastq -o 3_qc/CW2/CN2W.R1.fastq -O 3_qc/CW2/CN2W.R2.fastq --unpaired1 3_qc/CW2/CN2W_unpaired.R1.fastq --unpaired2 3_qc/CW2/CN2W_unpaired.R2.fastq --failed_out 3_qc/CW2/CN2W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/CN2W_merged.fastq -w 6 -h ./fastp_out/CN2W_fastp.html -j ./fastp_out/CN2W_fastp.json \n",
      "fastp v0.20.0, time used: 0 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 109412\n",
      "total bases: 16892283\n",
      "Q20 bases: 16634730(98.4753%)\n",
      "Q30 bases: 16091695(95.2606%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 109412\n",
      "total bases: 16942861\n",
      "Q20 bases: 16529297(97.5591%)\n",
      "Q30 bases: 15759724(93.0169%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 107418\n",
      "total bases: 10955320\n",
      "Q20 bases: 10850767(99.0456%)\n",
      "Q30 bases: 10537587(96.1869%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 215172\n",
      "reads failed due to low quality: 3240\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 412\n",
      "reads with adapter trimmed: 216686\n",
      "bases trimmed due to adapters: 2730707\n",
      "reads corrected by overlap analysis: 19193\n",
      "bases corrected by overlap analysis: 30435\n",
      "\n",
      "Duplication rate: 83.2258%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 107418\n",
      "% of original read pairs: 98.1775%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/CP2W_fastp.json\n",
      "HTML report: ./fastp_out/CP2W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW2/CP2W.R1.fastq -I 2_demultiplexed/CW2/CP2W.R2.fastq -o 3_qc/CW2/CP2W.R1.fastq -O 3_qc/CW2/CP2W.R2.fastq --unpaired1 3_qc/CW2/CP2W_unpaired.R1.fastq --unpaired2 3_qc/CW2/CP2W_unpaired.R2.fastq --failed_out 3_qc/CW2/CP2W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW2/CP2W_merged.fastq -w 6 -h ./fastp_out/CP2W_fastp.html -j ./fastp_out/CP2W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 87393\n",
      "total bases: 12383287\n",
      "Q20 bases: 12126419(97.9257%)\n",
      "Q30 bases: 11682678(94.3423%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 87393\n",
      "total bases: 12366483\n",
      "Q20 bases: 12017702(97.1796%)\n",
      "Q30 bases: 11441779(92.5225%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 71278\n",
      "total bases: 7321501\n",
      "Q20 bases: 7252315(99.055%)\n",
      "Q30 bases: 7047223(96.2538%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 149570\n",
      "reads failed due to low quality: 6744\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 18470\n",
      "reads with adapter trimmed: 146512\n",
      "bases trimmed due to adapters: 1823724\n",
      "reads corrected by overlap analysis: 12280\n",
      "bases corrected by overlap analysis: 20685\n",
      "\n",
      "Duplication rate: 14.0351%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 71278\n",
      "% of original read pairs: 81.5603%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C41W_fastp.json\n",
      "HTML report: ./fastp_out/C41W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C41W.R1.fastq -I 2_demultiplexed/CW3/C41W.R2.fastq -o 3_qc/CW3/C41W.R1.fastq -O 3_qc/CW3/C41W.R2.fastq --unpaired1 3_qc/CW3/C41W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C41W_unpaired.R2.fastq --failed_out 3_qc/CW3/C41W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C41W_merged.fastq -w 6 -h ./fastp_out/C41W_fastp.html -j ./fastp_out/C41W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 84252\n",
      "total bases: 12062408\n",
      "Q20 bases: 11863941(98.3547%)\n",
      "Q30 bases: 11476129(95.1396%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 84252\n",
      "total bases: 12072519\n",
      "Q20 bases: 11748160(97.3132%)\n",
      "Q30 bases: 11186848(92.6637%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 65866\n",
      "total bases: 6776725\n",
      "Q20 bases: 6718097(99.1349%)\n",
      "Q30 bases: 6541644(96.5311%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 149412\n",
      "reads failed due to low quality: 4838\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 14250\n",
      "reads with adapter trimmed: 138740\n",
      "bases trimmed due to adapters: 1599121\n",
      "reads corrected by overlap analysis: 12097\n",
      "bases corrected by overlap analysis: 20362\n",
      "\n",
      "Duplication rate: 31.3131%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 65866\n",
      "% of original read pairs: 78.1774%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C42W_fastp.json\n",
      "HTML report: ./fastp_out/C42W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C42W.R1.fastq -I 2_demultiplexed/CW3/C42W.R2.fastq -o 3_qc/CW3/C42W.R1.fastq -O 3_qc/CW3/C42W.R2.fastq --unpaired1 3_qc/CW3/C42W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C42W_unpaired.R2.fastq --failed_out 3_qc/CW3/C42W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C42W_merged.fastq -w 6 -h ./fastp_out/C42W_fastp.html -j ./fastp_out/C42W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 37229\n",
      "total bases: 5055829\n",
      "Q20 bases: 4951202(97.9306%)\n",
      "Q30 bases: 4759984(94.1484%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 37229\n",
      "total bases: 5069817\n",
      "Q20 bases: 4903900(96.7274%)\n",
      "Q30 bases: 4636268(91.4484%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 28621\n",
      "total bases: 2932036\n",
      "Q20 bases: 2904504(99.061%)\n",
      "Q30 bases: 2819600(96.1653%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 58956\n",
      "reads failed due to low quality: 3186\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 12316\n",
      "reads with adapter trimmed: 58516\n",
      "bases trimmed due to adapters: 615287\n",
      "reads corrected by overlap analysis: 5137\n",
      "bases corrected by overlap analysis: 8444\n",
      "\n",
      "Duplication rate: 28.125%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 28621\n",
      "% of original read pairs: 76.8782%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C43W_fastp.json\n",
      "HTML report: ./fastp_out/C43W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C43W.R1.fastq -I 2_demultiplexed/CW3/C43W.R2.fastq -o 3_qc/CW3/C43W.R1.fastq -O 3_qc/CW3/C43W.R2.fastq --unpaired1 3_qc/CW3/C43W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C43W_unpaired.R2.fastq --failed_out 3_qc/CW3/C43W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C43W_merged.fastq -w 6 -h ./fastp_out/C43W_fastp.html -j ./fastp_out/C43W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 68242\n",
      "total bases: 10208483\n",
      "Q20 bases: 10031786(98.2691%)\n",
      "Q30 bases: 9686191(94.8837%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 68242\n",
      "total bases: 10226109\n",
      "Q20 bases: 9944888(97.25%)\n",
      "Q30 bases: 9456106(92.4702%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 62303\n",
      "total bases: 6387701\n",
      "Q20 bases: 6330972(99.1119%)\n",
      "Q30 bases: 6156799(96.3852%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 126212\n",
      "reads failed due to low quality: 3396\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 6874\n",
      "reads with adapter trimmed: 126708\n",
      "bases trimmed due to adapters: 1600254\n",
      "reads corrected by overlap analysis: 10408\n",
      "bases corrected by overlap analysis: 17696\n",
      "\n",
      "Duplication rate: 30.3704%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 62303\n",
      "% of original read pairs: 91.2971%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C44W_fastp.json\n",
      "HTML report: ./fastp_out/C44W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C44W.R1.fastq -I 2_demultiplexed/CW3/C44W.R2.fastq -o 3_qc/CW3/C44W.R1.fastq -O 3_qc/CW3/C44W.R2.fastq --unpaired1 3_qc/CW3/C44W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C44W_unpaired.R2.fastq --failed_out 3_qc/CW3/C44W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C44W_merged.fastq -w 6 -h ./fastp_out/C44W_fastp.html -j ./fastp_out/C44W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 100311\n",
      "total bases: 14722029\n",
      "Q20 bases: 14436145(98.0581%)\n",
      "Q30 bases: 13935254(94.6558%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 100311\n",
      "total bases: 14738253\n",
      "Q20 bases: 14277236(96.872%)\n",
      "Q30 bases: 13611207(92.3529%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 86500\n",
      "total bases: 8882953\n",
      "Q20 bases: 8802617(99.0956%)\n",
      "Q30 bases: 8562216(96.3893%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 178038\n",
      "reads failed due to low quality: 7068\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 15516\n",
      "reads with adapter trimmed: 175930\n",
      "bases trimmed due to adapters: 2033174\n",
      "reads corrected by overlap analysis: 13908\n",
      "bases corrected by overlap analysis: 22959\n",
      "\n",
      "Duplication rate: 25.2427%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 86500\n",
      "% of original read pairs: 86.2318%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C45W_fastp.json\n",
      "HTML report: ./fastp_out/C45W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C45W.R1.fastq -I 2_demultiplexed/CW3/C45W.R2.fastq -o 3_qc/CW3/C45W.R1.fastq -O 3_qc/CW3/C45W.R2.fastq --unpaired1 3_qc/CW3/C45W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C45W_unpaired.R2.fastq --failed_out 3_qc/CW3/C45W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C45W_merged.fastq -w 6 -h ./fastp_out/C45W_fastp.html -j ./fastp_out/C45W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 59689\n",
      "total bases: 8925224\n",
      "Q20 bases: 8782387(98.3996%)\n",
      "Q30 bases: 8500123(95.2371%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 59689\n",
      "total bases: 8939057\n",
      "Q20 bases: 8697010(97.2923%)\n",
      "Q30 bases: 8274878(92.5699%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 55658\n",
      "total bases: 5710913\n",
      "Q20 bases: 5662435(99.1511%)\n",
      "Q30 bases: 5511926(96.5157%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 113100\n",
      "reads failed due to low quality: 2582\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 3694\n",
      "reads with adapter trimmed: 112672\n",
      "bases trimmed due to adapters: 1189257\n",
      "reads corrected by overlap analysis: 8616\n",
      "bases corrected by overlap analysis: 14357\n",
      "\n",
      "Duplication rate: 39.6694%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 55658\n",
      "% of original read pairs: 93.2467%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C46W_fastp.json\n",
      "HTML report: ./fastp_out/C46W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C46W.R1.fastq -I 2_demultiplexed/CW3/C46W.R2.fastq -o 3_qc/CW3/C46W.R1.fastq -O 3_qc/CW3/C46W.R2.fastq --unpaired1 3_qc/CW3/C46W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C46W_unpaired.R2.fastq --failed_out 3_qc/CW3/C46W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C46W_merged.fastq -w 6 -h ./fastp_out/C46W_fastp.html -j ./fastp_out/C46W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 143729\n",
      "total bases: 21943337\n",
      "Q20 bases: 21626311(98.5553%)\n",
      "Q30 bases: 20992140(95.6652%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 143729\n",
      "total bases: 21959298\n",
      "Q20 bases: 21493020(97.8766%)\n",
      "Q30 bases: 20633943(93.9645%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 136718\n",
      "total bases: 14014298\n",
      "Q20 bases: 13901065(99.192%)\n",
      "Q30 bases: 13554917(96.7221%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 275282\n",
      "reads failed due to low quality: 5342\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 6832\n",
      "reads with adapter trimmed: 276734\n",
      "bases trimmed due to adapters: 3469459\n",
      "reads corrected by overlap analysis: 18956\n",
      "bases corrected by overlap analysis: 30923\n",
      "\n",
      "Duplication rate: 53.7634%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 136718\n",
      "% of original read pairs: 95.1221%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C47W_fastp.json\n",
      "HTML report: ./fastp_out/C47W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C47W.R1.fastq -I 2_demultiplexed/CW3/C47W.R2.fastq -o 3_qc/CW3/C47W.R1.fastq -O 3_qc/CW3/C47W.R2.fastq --unpaired1 3_qc/CW3/C47W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C47W_unpaired.R2.fastq --failed_out 3_qc/CW3/C47W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C47W_merged.fastq -w 6 -h ./fastp_out/C47W_fastp.html -j ./fastp_out/C47W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 130947\n",
      "total bases: 19955130\n",
      "Q20 bases: 19626011(98.3507%)\n",
      "Q30 bases: 18980200(95.1144%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 130947\n",
      "total bases: 19979694\n",
      "Q20 bases: 19458930(97.3935%)\n",
      "Q30 bases: 18540667(92.7976%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 126912\n",
      "total bases: 13002273\n",
      "Q20 bases: 12884647(99.0953%)\n",
      "Q30 bases: 12525496(96.3331%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 254412\n",
      "reads failed due to low quality: 5270\n",
      "reads failed due to too many N: 2\n",
      "reads failed due to too short: 2210\n",
      "reads with adapter trimmed: 256858\n",
      "bases trimmed due to adapters: 2709674\n",
      "reads corrected by overlap analysis: 19894\n",
      "bases corrected by overlap analysis: 32471\n",
      "\n",
      "Duplication rate: 23.8095%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 126912\n",
      "% of original read pairs: 96.9186%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C48W_fastp.json\n",
      "HTML report: ./fastp_out/C48W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C48W.R1.fastq -I 2_demultiplexed/CW3/C48W.R2.fastq -o 3_qc/CW3/C48W.R1.fastq -O 3_qc/CW3/C48W.R2.fastq --unpaired1 3_qc/CW3/C48W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C48W_unpaired.R2.fastq --failed_out 3_qc/CW3/C48W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C48W_merged.fastq -w 6 -h ./fastp_out/C48W_fastp.html -j ./fastp_out/C48W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 130003\n",
      "total bases: 19968564\n",
      "Q20 bases: 19675768(98.5337%)\n",
      "Q30 bases: 19064366(95.4719%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 130003\n",
      "total bases: 20010400\n",
      "Q20 bases: 19527114(97.5848%)\n",
      "Q30 bases: 18636329(93.1332%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 124965\n",
      "total bases: 12815249\n",
      "Q20 bases: 12706591(99.1521%)\n",
      "Q30 bases: 12377070(96.5808%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 251508\n",
      "reads failed due to low quality: 4430\n",
      "reads failed due to too many N: 8\n",
      "reads failed due to too short: 4060\n",
      "reads with adapter trimmed: 253308\n",
      "bases trimmed due to adapters: 3159988\n",
      "reads corrected by overlap analysis: 19243\n",
      "bases corrected by overlap analysis: 31311\n",
      "\n",
      "Duplication rate: 39.6825%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 124965\n",
      "% of original read pairs: 96.1247%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C49W_fastp.json\n",
      "HTML report: ./fastp_out/C49W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C49W.R1.fastq -I 2_demultiplexed/CW3/C49W.R2.fastq -o 3_qc/CW3/C49W.R1.fastq -O 3_qc/CW3/C49W.R2.fastq --unpaired1 3_qc/CW3/C49W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C49W_unpaired.R2.fastq --failed_out 3_qc/CW3/C49W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C49W_merged.fastq -w 6 -h ./fastp_out/C49W_fastp.html -j ./fastp_out/C49W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 251530\n",
      "total bases: 38479663\n",
      "Q20 bases: 37862806(98.3969%)\n",
      "Q30 bases: 36634014(95.2036%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 251530\n",
      "total bases: 38535801\n",
      "Q20 bases: 37584212(97.5306%)\n",
      "Q30 bases: 35901505(93.164%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 242197\n",
      "total bases: 24810051\n",
      "Q20 bases: 24585478(99.0948%)\n",
      "Q30 bases: 23912501(96.3823%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 488512\n",
      "reads failed due to low quality: 9854\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 4690\n",
      "reads with adapter trimmed: 490328\n",
      "bases trimmed due to adapters: 5684871\n",
      "reads corrected by overlap analysis: 39105\n",
      "bases corrected by overlap analysis: 64340\n",
      "\n",
      "Duplication rate: 28.0576%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 242197\n",
      "% of original read pairs: 96.2895%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C50W_fastp.json\n",
      "HTML report: ./fastp_out/C50W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C50W.R1.fastq -I 2_demultiplexed/CW3/C50W.R2.fastq -o 3_qc/CW3/C50W.R1.fastq -O 3_qc/CW3/C50W.R2.fastq --unpaired1 3_qc/CW3/C50W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C50W_unpaired.R2.fastq --failed_out 3_qc/CW3/C50W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C50W_merged.fastq -w 6 -h ./fastp_out/C50W_fastp.html -j ./fastp_out/C50W_fastp.json \n",
      "fastp v0.20.0, time used: 3 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 183500\n",
      "total bases: 27887650\n",
      "Q20 bases: 27366629(98.1317%)\n",
      "Q30 bases: 26346191(94.4726%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 183500\n",
      "total bases: 27911057\n",
      "Q20 bases: 27117879(97.1582%)\n",
      "Q30 bases: 25723165(92.1612%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 176455\n",
      "total bases: 18079668\n",
      "Q20 bases: 17898419(98.9975%)\n",
      "Q30 bases: 17354941(95.9915%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 354454\n",
      "reads failed due to low quality: 8340\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 4202\n",
      "reads with adapter trimmed: 357936\n",
      "bases trimmed due to adapters: 3799010\n",
      "reads corrected by overlap analysis: 30485\n",
      "bases corrected by overlap analysis: 50621\n",
      "\n",
      "Duplication rate: 50.7692%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 176455\n",
      "% of original read pairs: 96.1608%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C51W_fastp.json\n",
      "HTML report: ./fastp_out/C51W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C51W.R1.fastq -I 2_demultiplexed/CW3/C51W.R2.fastq -o 3_qc/CW3/C51W.R1.fastq -O 3_qc/CW3/C51W.R2.fastq --unpaired1 3_qc/CW3/C51W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C51W_unpaired.R2.fastq --failed_out 3_qc/CW3/C51W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C51W_merged.fastq -w 6 -h ./fastp_out/C51W_fastp.html -j ./fastp_out/C51W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 117782\n",
      "total bases: 15776616\n",
      "Q20 bases: 15477759(98.1057%)\n",
      "Q30 bases: 14940555(94.7006%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 117782\n",
      "total bases: 15758872\n",
      "Q20 bases: 15344252(97.369%)\n",
      "Q30 bases: 14647347(92.9467%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 89607\n",
      "total bases: 9187842\n",
      "Q20 bases: 9106967(99.1198%)\n",
      "Q30 bases: 8863581(96.4708%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 180214\n",
      "reads failed due to low quality: 8062\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 47288\n",
      "reads with adapter trimmed: 181378\n",
      "bases trimmed due to adapters: 2099058\n",
      "reads corrected by overlap analysis: 14200\n",
      "bases corrected by overlap analysis: 23565\n",
      "\n",
      "Duplication rate: 33.0357%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 89607\n",
      "% of original read pairs: 76.0787%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C52W_fastp.json\n",
      "HTML report: ./fastp_out/C52W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C52W.R1.fastq -I 2_demultiplexed/CW3/C52W.R2.fastq -o 3_qc/CW3/C52W.R1.fastq -O 3_qc/CW3/C52W.R2.fastq --unpaired1 3_qc/CW3/C52W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C52W_unpaired.R2.fastq --failed_out 3_qc/CW3/C52W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C52W_merged.fastq -w 6 -h ./fastp_out/C52W_fastp.html -j ./fastp_out/C52W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 149901\n",
      "total bases: 22916414\n",
      "Q20 bases: 22543324(98.372%)\n",
      "Q30 bases: 21797056(95.1155%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 149901\n",
      "total bases: 22933361\n",
      "Q20 bases: 22381853(97.5952%)\n",
      "Q30 bases: 21390130(93.2708%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 145921\n",
      "total bases: 14948079\n",
      "Q20 bases: 14810516(99.0797%)\n",
      "Q30 bases: 14390852(96.2723%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 292878\n",
      "reads failed due to low quality: 5922\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 996\n",
      "reads with adapter trimmed: 295566\n",
      "bases trimmed due to adapters: 3116667\n",
      "reads corrected by overlap analysis: 21903\n",
      "bases corrected by overlap analysis: 35618\n",
      "\n",
      "Duplication rate: 61.6034%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 145921\n",
      "% of original read pairs: 97.3449%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C53W_fastp.json\n",
      "HTML report: ./fastp_out/C53W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C53W.R1.fastq -I 2_demultiplexed/CW3/C53W.R2.fastq -o 3_qc/CW3/C53W.R1.fastq -O 3_qc/CW3/C53W.R2.fastq --unpaired1 3_qc/CW3/C53W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C53W_unpaired.R2.fastq --failed_out 3_qc/CW3/C53W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C53W_merged.fastq -w 6 -h ./fastp_out/C53W_fastp.html -j ./fastp_out/C53W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 149364\n",
      "total bases: 20952855\n",
      "Q20 bases: 20530488(97.9842%)\n",
      "Q30 bases: 19793273(94.4658%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 149364\n",
      "total bases: 20943292\n",
      "Q20 bases: 20397198(97.3925%)\n",
      "Q30 bases: 19468233(92.9569%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 122080\n",
      "total bases: 12510811\n",
      "Q20 bases: 12397114(99.0912%)\n",
      "Q30 bases: 12058762(96.3867%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 244804\n",
      "reads failed due to low quality: 9864\n",
      "reads failed due to too many N: 4\n",
      "reads failed due to too short: 44056\n",
      "reads with adapter trimmed: 246908\n",
      "bases trimmed due to adapters: 3141177\n",
      "reads corrected by overlap analysis: 19074\n",
      "bases corrected by overlap analysis: 31654\n",
      "\n",
      "Duplication rate: 15.0538%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 122080\n",
      "% of original read pairs: 81.7332%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C54W_fastp.json\n",
      "HTML report: ./fastp_out/C54W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C54W.R1.fastq -I 2_demultiplexed/CW3/C54W.R2.fastq -o 3_qc/CW3/C54W.R1.fastq -O 3_qc/CW3/C54W.R2.fastq --unpaired1 3_qc/CW3/C54W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C54W_unpaired.R2.fastq --failed_out 3_qc/CW3/C54W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C54W_merged.fastq -w 6 -h ./fastp_out/C54W_fastp.html -j ./fastp_out/C54W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 139013\n",
      "total bases: 21207456\n",
      "Q20 bases: 20852944(98.3284%)\n",
      "Q30 bases: 20163649(95.0781%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 139013\n",
      "total bases: 21199591\n",
      "Q20 bases: 20702015(97.6529%)\n",
      "Q30 bases: 19804396(93.4188%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 134573\n",
      "total bases: 13796496\n",
      "Q20 bases: 13671167(99.0916%)\n",
      "Q30 bases: 13292913(96.3499%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 270070\n",
      "reads failed due to low quality: 5838\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 2118\n",
      "reads with adapter trimmed: 272524\n",
      "bases trimmed due to adapters: 2869034\n",
      "reads corrected by overlap analysis: 20139\n",
      "bases corrected by overlap analysis: 32605\n",
      "\n",
      "Duplication rate: 58.2996%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 134573\n",
      "% of original read pairs: 96.8061%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/C55W_fastp.json\n",
      "HTML report: ./fastp_out/C55W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/C55W.R1.fastq -I 2_demultiplexed/CW3/C55W.R2.fastq -o 3_qc/CW3/C55W.R1.fastq -O 3_qc/CW3/C55W.R2.fastq --unpaired1 3_qc/CW3/C55W_unpaired.R1.fastq --unpaired2 3_qc/CW3/C55W_unpaired.R2.fastq --failed_out 3_qc/CW3/C55W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/C55W_merged.fastq -w 6 -h ./fastp_out/C55W_fastp.html -j ./fastp_out/C55W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 29374\n",
      "total bases: 3796249\n",
      "Q20 bases: 3726692(98.1677%)\n",
      "Q30 bases: 3604263(94.9427%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 29374\n",
      "total bases: 3785179\n",
      "Q20 bases: 3672166(97.0143%)\n",
      "Q30 bases: 3484009(92.0434%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 22022\n",
      "total bases: 2268140\n",
      "Q20 bases: 2250795(99.2353%)\n",
      "Q30 bases: 2196142(96.8257%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 44120\n",
      "reads failed due to low quality: 2388\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 12240\n",
      "reads with adapter trimmed: 44326\n",
      "bases trimmed due to adapters: 512519\n",
      "reads corrected by overlap analysis: 3992\n",
      "bases corrected by overlap analysis: 6779\n",
      "\n",
      "Duplication rate: 48.3051%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 22022\n",
      "% of original read pairs: 74.9711%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/CF1W_fastp.json\n",
      "HTML report: ./fastp_out/CF1W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/CF1W.R1.fastq -I 2_demultiplexed/CW3/CF1W.R2.fastq -o 3_qc/CW3/CF1W.R1.fastq -O 3_qc/CW3/CF1W.R2.fastq --unpaired1 3_qc/CW3/CF1W_unpaired.R1.fastq --unpaired2 3_qc/CW3/CF1W_unpaired.R2.fastq --failed_out 3_qc/CW3/CF1W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/CF1W_merged.fastq -w 6 -h ./fastp_out/CF1W_fastp.html -j ./fastp_out/CF1W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 7613\n",
      "total bases: 891596\n",
      "Q20 bases: 876598(98.3178%)\n",
      "Q30 bases: 848261(95.1396%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 7613\n",
      "total bases: 889445\n",
      "Q20 bases: 866809(97.455%)\n",
      "Q30 bases: 828076(93.1003%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 4883\n",
      "total bases: 502767\n",
      "Q20 bases: 499044(99.2595%)\n",
      "Q30 bases: 487283(96.9202%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 9784\n",
      "reads failed due to low quality: 526\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 4916\n",
      "reads with adapter trimmed: 9794\n",
      "bases trimmed due to adapters: 111981\n",
      "reads corrected by overlap analysis: 734\n",
      "bases corrected by overlap analysis: 1146\n",
      "\n",
      "Duplication rate: 9.09091%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 143\n",
      "\n",
      "Read pairs merged: 4883\n",
      "% of original read pairs: 64.1403%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/CN3W_fastp.json\n",
      "HTML report: ./fastp_out/CN3W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/CN3W.R1.fastq -I 2_demultiplexed/CW3/CN3W.R2.fastq -o 3_qc/CW3/CN3W.R1.fastq -O 3_qc/CW3/CN3W.R2.fastq --unpaired1 3_qc/CW3/CN3W_unpaired.R1.fastq --unpaired2 3_qc/CW3/CN3W_unpaired.R2.fastq --failed_out 3_qc/CW3/CN3W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/CN3W_merged.fastq -w 6 -h ./fastp_out/CN3W_fastp.html -j ./fastp_out/CN3W_fastp.json \n",
      "fastp v0.20.0, time used: 1 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 250717\n",
      "total bases: 38763019\n",
      "Q20 bases: 38198210(98.5429%)\n",
      "Q30 bases: 37000704(95.4536%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 250717\n",
      "total bases: 38908932\n",
      "Q20 bases: 37837482(97.2463%)\n",
      "Q30 bases: 35930193(92.3443%)\n",
      "\n",
      "Merged and filtered:\n",
      "total reads: 246464\n",
      "total bases: 25136055\n",
      "Q20 bases: 24910894(99.1042%)\n",
      "Q30 bases: 24225655(96.3781%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 493770\n",
      "reads failed due to low quality: 7626\n",
      "reads failed due to too many N: 6\n",
      "reads failed due to too short: 32\n",
      "reads with adapter trimmed: 496790\n",
      "bases trimmed due to adapters: 6277416\n",
      "reads corrected by overlap analysis: 46484\n",
      "bases corrected by overlap analysis: 75673\n",
      "\n",
      "Duplication rate: 84.225%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 142\n",
      "\n",
      "Read pairs merged: 246464\n",
      "% of original read pairs: 98.3037%\n",
      "% in reads after filtering: 100%\n",
      "\n",
      "\n",
      "JSON report: ./fastp_out/CP3W_fastp.json\n",
      "HTML report: ./fastp_out/CP3W_fastp.html\n",
      "\n",
      "fastp -i 2_demultiplexed/CW3/CP3W.R1.fastq -I 2_demultiplexed/CW3/CP3W.R2.fastq -o 3_qc/CW3/CP3W.R1.fastq -O 3_qc/CW3/CP3W.R2.fastq --unpaired1 3_qc/CW3/CP3W_unpaired.R1.fastq --unpaired2 3_qc/CW3/CP3W_unpaired.R2.fastq --failed_out 3_qc/CW3/CP3W_failed.fastq -q 30 --cut_tail --trim_front1 20 --trim_front2 20 --max_len1 106 --max_len2 106 -l 90 --merge --overlap_len_require 90 --correction --merged_out 4_merged_fastq/CW3/CP3W_merged.fastq -w 6 -h ./fastp_out/CP3W_fastp.html -j ./fastp_out/CP3W_fastp.json \n",
      "fastp v0.20.0, time used: 2 seconds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#PAIRED READS QC\n",
    "\n",
    "mkdir 3_qc\n",
    "mkdir 4_merged_fastq\n",
    "mkdir 5_merged_fasta\n",
    "\n",
    "libraries=$(ls -1 2_demultiplexed)\n",
    "for library in $libraries\n",
    "do\n",
    "    mkdir 3_qc/${library}/\n",
    "    mkdir 4_merged_fastq/${library}/\n",
    "    mkdir 5_merged_fasta/${library}/\n",
    "    files=$(ls -1 2_demultiplexed/${library} | grep -v invalid | grep R1.fastq | sed 's/.R1.fastq.*//')\n",
    "    for sample in $files\n",
    "    do\n",
    "        fastp -i 2_demultiplexed/${library}/${sample}.R1.fastq -I 2_demultiplexed/${library}/${sample}.R2.fastq \\\n",
    "        -o 3_qc/${library}/${sample}.R1.fastq -O 3_qc/${library}/${sample}.R2.fastq \\\n",
    "        --unpaired1 3_qc/${library}/${sample}_unpaired.R1.fastq \\\n",
    "        --unpaired2 3_qc/${library}/${sample}_unpaired.R2.fastq \\\n",
    "        --failed_out 3_qc/${library}/${sample}_failed.fastq \\\n",
    "        -q 30 \\\n",
    "        --cut_tail \\\n",
    "        --trim_front1 20 \\\n",
    "        --trim_front2 20 \\\n",
    "        --max_len1 106 \\\n",
    "        --max_len2 106 \\\n",
    "        -l 90 \\\n",
    "        --merge \\\n",
    "        --overlap_len_require 90 \\\n",
    "        --correction \\\n",
    "        --merged_out 4_merged_fastq/${library}/${sample}_merged.fastq \\\n",
    "        -w 6 \\\n",
    "        -h ./fastp_out/${sample}_fastp.html \\\n",
    "        -j ./fastp_out/${sample}_fastp.json\n",
    "        cat 3_qc/${library}/${sample}.R1.fastq >> 4_merged_fastq/${library}/${sample}_merged.fastq\n",
    "        cat 3_qc/${library}/${sample}_unpaired.R1.fastq >> 4_merged_fastq/${library}/${sample}_merged.fastq\n",
    "        seqkit fq2fa 4_merged_fastq/${library}/${sample}_merged.fastq -o 5_merged_fasta/${library}/${sample}_merged.fasta\n",
    "    done\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# derep - denoise - chimera - rerep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL01_merged.fasta 100%\n",
      "13272433 nt in 128542 seqs, min 90, max 121, avg 103\n",
      "Sorting 100%\n",
      "4993 unique sequences, avg cluster 25.7, median 1, max 38228\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "1060 uniques written, 3933 clusters discarded (78.8%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL01_derep.fasta 100%\n",
      "109929 nt in 1060 seqs, min 91, max 120, avg 104\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 182 Size min 3, max 46204, avg 5.8\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 14 (7.7%) chimeras, 168 (92.3%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 182 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "201 (0.2%) chimeras, 123945 (99.8%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 124146 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 123945 reads from 168 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL02_merged.fasta 100%\n",
      "6913639 nt in 67159 seqs, min 90, max 121, avg 103\n",
      "Sorting 100%\n",
      "2486 unique sequences, avg cluster 27.0, median 1, max 25440\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "567 uniques written, 1919 clusters discarded (77.2%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL02_derep.fasta 100%\n",
      "58661 nt in 567 seqs, min 91, max 121, avg 103\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 149 Size min 3, max 28437, avg 3.8\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 8 (5.4%) chimeras, 141 (94.6%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 149 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "471 (0.7%) chimeras, 64492 (99.3%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 64963 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 64492 reads from 141 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL03_merged.fasta 100%\n",
      "9697552 nt in 94159 seqs, min 90, max 121, avg 103\n",
      "Sorting 100%\n",
      "3896 unique sequences, avg cluster 24.2, median 1, max 33460\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "866 uniques written, 3030 clusters discarded (77.8%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL03_derep.fasta 100%\n",
      "89930 nt in 866 seqs, min 91, max 121, avg 104\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 357 Size min 3, max 37549, avg 2.4\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 7 (2.0%) chimeras, 350 (98.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 357 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "636 (0.7%) chimeras, 90062 (99.3%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 90698 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 90062 reads from 350 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL04_merged.fasta 100%\n",
      "6806951 nt in 66339 seqs, min 91, max 121, avg 103\n",
      "Sorting 100%\n",
      "2813 unique sequences, avg cluster 23.6, median 1, max 21106\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "573 uniques written, 2240 clusters discarded (79.6%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL04_derep.fasta 100%\n",
      "59185 nt in 573 seqs, min 91, max 121, avg 103\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 194 Size min 3, max 24458, avg 3.0\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 5 (2.6%) chimeras, 189 (97.4%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 194 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "370 (0.6%) chimeras, 63355 (99.4%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 63725 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 63355 reads from 189 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL05_merged.fasta 100%\n",
      "11510683 nt in 112073 seqs, min 90, max 121, avg 103\n",
      "Sorting 100%\n",
      "3428 unique sequences, avg cluster 32.7, median 1, max 40971\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "768 uniques written, 2660 clusters discarded (77.6%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL05_derep.fasta 100%\n",
      "79424 nt in 768 seqs, min 91, max 120, avg 103\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 209 Size min 3, max 45848, avg 3.7\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 8 (3.8%) chimeras, 201 (96.2%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 209 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "784 (0.7%) chimeras, 108311 (99.3%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 109095 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 108311 reads from 201 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL06_merged.fasta 100%\n",
      "10956681 nt in 106565 seqs, min 90, max 121, avg 103\n",
      "Sorting 100%\n",
      "3482 unique sequences, avg cluster 30.6, median 1, max 36506\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "782 uniques written, 2700 clusters discarded (77.5%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL06_derep.fasta 100%\n",
      "80889 nt in 782 seqs, min 91, max 121, avg 103\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 249 Size min 3, max 40696, avg 3.1\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 6 (2.4%) chimeras, 243 (97.6%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 249 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "595 (0.6%) chimeras, 102893 (99.4%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 103488 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 102893 reads from 243 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL07_merged.fasta 100%\n",
      "8795039 nt in 85709 seqs, min 90, max 121, avg 103\n",
      "Sorting 100%\n",
      "2859 unique sequences, avg cluster 30.0, median 1, max 24789\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "626 uniques written, 2233 clusters discarded (78.1%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL07_derep.fasta 100%\n",
      "64726 nt in 626 seqs, min 91, max 121, avg 103\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 147 Size min 3, max 27816, avg 4.3\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 1 (0.7%) chimeras, 146 (99.3%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 147 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "3 (0.0%) chimeras, 83133 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 83136 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 83133 reads from 146 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL08_merged.fasta 100%\n",
      "7551246 nt in 73203 seqs, min 90, max 120, avg 103\n",
      "Sorting 100%\n",
      "2841 unique sequences, avg cluster 25.8, median 1, max 21036\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "563 uniques written, 2278 clusters discarded (80.2%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL08_derep.fasta 100%\n",
      "58472 nt in 563 seqs, min 92, max 120, avg 104\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 185 Size min 3, max 23645, avg 3.0\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 0 (0.0%) chimeras, 185 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 185 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "0 (0.0%) chimeras, 70595 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 70595 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 70595 reads from 185 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL09_merged.fasta 100%\n",
      "6124577 nt in 59351 seqs, min 91, max 121, avg 103\n",
      "Sorting 100%\n",
      "2492 unique sequences, avg cluster 23.8, median 1, max 20220\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "461 uniques written, 2031 clusters discarded (81.5%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL09_derep.fasta 100%\n",
      "48022 nt in 461 seqs, min 91, max 121, avg 104\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 191 Size min 3, max 22383, avg 2.4\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 0 (0.0%) chimeras, 191 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 191 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "0 (0.0%) chimeras, 57045 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 57045 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 57045 reads from 191 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL10_merged.fasta 100%\n",
      "5551555 nt in 53512 seqs, min 90, max 121, avg 104\n",
      "Sorting 100%\n",
      "2885 unique sequences, avg cluster 18.5, median 1, max 16554\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "548 uniques written, 2337 clusters discarded (81.0%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL10_derep.fasta 100%\n",
      "57257 nt in 548 seqs, min 90, max 121, avg 104\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 253 Size min 3, max 18889, avg 2.2\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 1 (0.4%) chimeras, 252 (99.6%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 253 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "83 (0.2%) chimeras, 50746 (99.8%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 50829 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 50746 reads from 252 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL11_merged.fasta 100%\n",
      "8678632 nt in 84461 seqs, min 91, max 121, avg 103\n",
      "Sorting 100%\n",
      "3169 unique sequences, avg cluster 26.7, median 1, max 19086\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "552 uniques written, 2617 clusters discarded (82.6%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL11_derep.fasta 100%\n",
      "56997 nt in 552 seqs, min 91, max 120, avg 103\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 135 Size min 3, max 20075, avg 4.1\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 1 (0.7%) chimeras, 134 (99.3%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 135 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "4 (0.0%) chimeras, 81422 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 81426 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 81422 reads from 134 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL12_merged.fasta 100%\n",
      "5499897 nt in 53577 seqs, min 90, max 121, avg 103\n",
      "Sorting 100%\n",
      "2056 unique sequences, avg cluster 26.1, median 1, max 16132\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "447 uniques written, 1609 clusters discarded (78.3%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL12_derep.fasta 100%\n",
      "46129 nt in 447 seqs, min 91, max 121, avg 103\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 117 Size min 3, max 18264, avg 3.8\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 6 (5.1%) chimeras, 111 (94.9%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 117 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "398 (0.8%) chimeras, 51331 (99.2%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 51729 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 51331 reads from 111 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL13_merged.fasta 100%\n",
      "4833646 nt in 46929 seqs, min 91, max 121, avg 103\n",
      "Sorting 100%\n",
      "1977 unique sequences, avg cluster 23.7, median 1, max 17445\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "457 uniques written, 1520 clusters discarded (76.9%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL13_derep.fasta 100%\n",
      "47315 nt in 457 seqs, min 92, max 121, avg 104\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 151 Size min 3, max 18352, avg 3.0\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 16 (10.6%) chimeras, 135 (89.4%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 151 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "571 (1.3%) chimeras, 44603 (98.7%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 45174 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 44603 reads from 135 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL14_merged.fasta 100%\n",
      "1137529 nt in 11020 seqs, min 92, max 119, avg 103\n",
      "Sorting 100%\n",
      "919 unique sequences, avg cluster 12.0, median 1, max 3672\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "132 uniques written, 787 clusters discarded (85.6%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL14_derep.fasta 100%\n",
      "13771 nt in 132 seqs, min 92, max 119, avg 104\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 93 Size min 3, max 3925, avg 1.4\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 3 (3.2%) chimeras, 90 (96.8%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 93 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "41 (0.4%) chimeras, 10093 (99.6%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 10134 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 10093 reads from 90 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL15_merged.fasta 100%\n",
      "5669486 nt in 55227 seqs, min 90, max 119, avg 103\n",
      "Sorting 100%\n",
      "2348 unique sequences, avg cluster 23.5, median 1, max 13042\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "486 uniques written, 1862 clusters discarded (79.3%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL15_derep.fasta 100%\n",
      "50175 nt in 486 seqs, min 91, max 119, avg 103\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 130 Size min 3, max 14550, avg 3.7\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 6 (4.6%) chimeras, 124 (95.4%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 130 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "323 (0.6%) chimeras, 52713 (99.4%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 53036 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 52713 reads from 124 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL16_merged.fasta 100%\n",
      "5291572 nt in 51277 seqs, min 90, max 121, avg 103\n",
      "Sorting 100%\n",
      "2558 unique sequences, avg cluster 20.0, median 1, max 17148\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "518 uniques written, 2040 clusters discarded (79.7%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL16_derep.fasta 100%\n",
      "54086 nt in 518 seqs, min 91, max 121, avg 104\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 266 Size min 3, max 18648, avg 1.9\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 1 (0.4%) chimeras, 265 (99.6%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 266 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "148 (0.3%) chimeras, 48820 (99.7%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 48968 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 48820 reads from 265 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL17_merged.fasta 100%\n",
      "1565810 nt in 15125 seqs, min 91, max 121, avg 104\n",
      "Sorting 100%\n",
      "1074 unique sequences, avg cluster 14.1, median 1, max 4334\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "140 uniques written, 934 clusters discarded (87.0%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL17_derep.fasta 100%\n",
      "14640 nt in 140 seqs, min 92, max 121, avg 105\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 78 Size min 3, max 4522, avg 1.8\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 1 (1.3%) chimeras, 77 (98.7%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 78 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "40 (0.3%) chimeras, 14013 (99.7%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 14053 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 14013 reads from 77 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL-EB_merged.fasta 100%\n",
      "212 nt in 2 seqs, min 106, max 106, avg 106\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 1.0, median 1, max 1\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "0 uniques written, 2 clusters discarded (100.0%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "\n",
      "\n",
      "Fatal error: Unable to read from file (6_denoise_uc/derep/EA01/BLEL-EB_derep.fasta)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "\n",
      "\n",
      "Fatal error: Unable to read from file (6_denoise_uc/denoise/EA01/BLEL-EB_denoise.fasta)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "\n",
      "\n",
      "Fatal error: Unable to read from file (6_denoise_uc/uchime/EA01/BLEL-EB_nc.fasta)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL-FB_merged.fasta 100%\n",
      "5395 nt in 48 seqs, min 105, max 117, avg 112\n",
      "Sorting 100%\n",
      "36 unique sequences, avg cluster 1.3, median 1, max 3\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "3 uniques written, 33 clusters discarded (91.7%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL-FB_derep.fasta 100%\n",
      "351 nt in 3 seqs, min 117, max 117, avg 117\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 3 Size min 3, max 3, avg 1.0\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 0 (0.0%) chimeras, 3 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 3 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "0 (0.0%) chimeras, 9 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 9 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 9 reads from 3 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL-Neg_merged.fasta 100%\n",
      "13084 nt in 127 seqs, min 90, max 106, avg 103\n",
      "Sorting 100%\n",
      "81 unique sequences, avg cluster 1.6, median 1, max 45\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "2 uniques written, 79 clusters discarded (97.5%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL-Neg_derep.fasta 100%\n",
      "205 nt in 2 seqs, min 102, max 103, avg 102\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 1 Size min 48, max 48, avg 2.0\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 0 (0.0%) chimeras, 1 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 1 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "0 (0.0%) chimeras, 48 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 48 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 48 reads from 1 amplicons\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file 5_merged_fasta/EA01/BLEL-Pos_merged.fasta 100%\n",
      "1368857 nt in 13424 seqs, min 99, max 106, avg 102\n",
      "Sorting 100%\n",
      "344 unique sequences, avg cluster 39.0, median 1, max 12050\n",
      "Writing output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "103 uniques written, 241 clusters discarded (70.1%)\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file 6_denoise_uc/derep/EA01/BLEL-Pos_derep.fasta 100%\n",
      "10503 nt in 103 seqs, min 100, max 106, avg 102\n",
      "Masking 100%\n",
      "Sorting by abundance 100%\n",
      "Counting k-mers 100%\n",
      "Clustering 100%\n",
      "Sorting clusters 100%\n",
      "Writing clusters 100%\n",
      "Clusters: 1 Size min 13139, max 13139, avg 103.0\n",
      "Singletons: 0, 0.0% of seqs, 0.0% of clusters\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Reading file blast_db/12s_full.fasta 100%\n",
      "3656075 nt in 2259 seqs, min 63, max 18999, avg 1618\n",
      "Masking 100%\n",
      "Counting k-mers 100%\n",
      "Creating k-mer index 100%\n",
      "Detecting chimeras 100%\n",
      "Found 0 (0.0%) chimeras, 1 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 1 unique sequences.\n",
      "Taking abundance information into account, this corresponds to\n",
      "0 (0.0%) chimeras, 13139 (100.0%) non-chimeras,\n",
      "and 0 (0.0%) borderline sequences in 13139 total sequences.\n",
      "vsearch v2.14.2_linux_x86_64, 31.3GB RAM, 12 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rereplicating 100%\n",
      "Rereplicated 13139 reads from 1 amplicons\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#DEREPLICATE, DENOISE, CHIMERA DETECTION AND REREPLICATE\n",
    "\n",
    "mkdir 6_denoise_uc/\n",
    "mkdir 6_denoise_uc/derep/\n",
    "mkdir 6_denoise_uc/denoise/\n",
    "mkdir 6_denoise_uc/uchime/\n",
    "mkdir 6_denoise_uc/rerep/\n",
    "mkdir 6_denoise_uc/clusters/\n",
    "\n",
    "libraries=$(ls -1 5_merged_fasta/)\n",
    "for library in $libraries\n",
    "do\n",
    "    mkdir 6_denoise_uc/derep/${library}/\n",
    "    mkdir 6_denoise_uc/denoise/${library}/\n",
    "    mkdir 6_denoise_uc/uchime/${library}/\n",
    "    mkdir 6_denoise_uc/rerep/${library}/\n",
    "    mkdir 6_denoise_uc/clusters/${library}/\n",
    "    \n",
    "    files=$(ls -1 5_merged_fasta/${library} | grep _merged.fasta | sed 's/_merged.fasta*//')\n",
    "    for sample in $files\n",
    "    do\n",
    "        \n",
    "        vsearch --derep_fulllength 5_merged_fasta/${library}/${sample}_merged.fasta \\\n",
    "        --sizeout \\\n",
    "        --output 6_denoise_uc/derep/${library}/${sample}_derep.fasta --minuniquesize 3 \\\n",
    "        --uc 6_denoise_uc/clusters/${library}/${sample}_derep.txt\n",
    "        \n",
    "        vsearch --cluster_unoise 6_denoise_uc/derep/${library}/${sample}_derep.fasta \\\n",
    "        --sizein --sizeout --minsize 1 --unoise_alpha 2.0 \\\n",
    "        --id 0.99 \\\n",
    "        --centroids 6_denoise_uc/denoise/${library}/${sample}_denoise.fasta \\\n",
    "        --uc 6_denoise_uc/clusters/${library}/${sample}_denoise.txt\n",
    "        \n",
    "        vsearch --uchime_ref 6_denoise_uc/denoise/${library}/${sample}_denoise.fasta --db blast_db/12s_full.fasta \\\n",
    "        --chimeras 6_denoise_uc/uchime/${library}/${sample}_chimera.fasta \\\n",
    "        --borderline 6_denoise_uc/uchime/${library}/${sample}_chimera.fasta \\\n",
    "        --mindiffs 1 --mindiv 0.8 --nonchimeras 6_denoise_uc/uchime/${library}/${sample}_nc.fasta \\\n",
    "        \n",
    "        vsearch --rereplicate 6_denoise_uc/uchime/${library}/${sample}_nc.fasta \\\n",
    "        --output 6_denoise_uc/rerep/${library}/${sample}_rerep.fasta #--uc stuff.txt\n",
    "        \n",
    "    done\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Blast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: [blastn] Query is Empty!\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir blast_out\n",
    "\n",
    "libraries=$(ls -1 6_denoise_uc/uchime/)\n",
    "for library in $libraries\n",
    "do\n",
    "    mkdir blast_out/${library}/\n",
    "    files=$(ls -1 6_denoise_uc/uchime/${library} | grep _nc.fasta | sed 's/_nc.fasta*//')\n",
    "    for sample in $files\n",
    "    do\n",
    "        blastn -query 6_denoise_uc/uchime/${library}/${sample}_nc.fasta \\\n",
    "        -db blast_db/12s_full \\\n",
    "        -out blast_out/${library}/${sample}_blast.tsv \\\n",
    "        -perc_identity 98 \\\n",
    "        -outfmt '6 qseqid stitle sacc staxids pident qcovs evalue bitscore' \\\n",
    "        -num_alignments 20 \\\n",
    "        -num_threads 6\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Taxonomy to Blast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir blast_tax\n",
    "\n",
    "python tax_to_blast.py -i blast_out -o blast_tax -lin new_taxdump/rankedlineage.dmp\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir blast_lca\n",
    "\n",
    "libraries=$(ls -1 blast_tax/)\n",
    "for library in $libraries\n",
    "do\n",
    "    mkdir blast_lca/${library}/\n",
    "    files=$(ls -1 blast_tax/${library} | grep _tax.tsv | sed 's/_tax.tsv*//')\n",
    "    for sample in $files\n",
    "    do\n",
    "        python mlca.py -i blast_tax/${library}/${sample}_tax.tsv -o blast_lca/${library}/${sample}_lca.tsv \\\n",
    "        -b 10 -id 100 -cov 60 -m 100 -hits 1\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LCA to TSV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "!python mlca-tsv.py -i blast_lca/ -r 6_denoise_uc/rerep/  -o final_out.tsv"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
